<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="css/style.css">
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<nav id="TOC">
<ul>
<li><a href="#allele-and-genotype-frequencies">Allele and genotype frequencies</a><ul>
<li><a href="#allele-frequencies">Allele frequencies</a></li>
<li><a href="#hardyweinberg-proportions">Hardy–Weinberg proportions</a></li>
<li><a href="#coefficient-of-kinship">Coefficient of kinship</a></li>
<li><a href="#inbreeding">Inbreeding</a></li>
<li><a href="#calculating-inbreeding-coefficients-from-data">Calculating inbreeding coefficients from data</a></li>
<li><a href="#summarizing-population-structure">Summarizing population structure</a></li>
<li><a href="#other-approaches-to-population-structure">Other approaches to population structure</a><ul>
<li><a href="#assignment-methods">Assignment Methods</a></li>
<li><a href="#principal-components-analysis">Principal components analysis</a></li>
</ul></li>
<li><a href="#correlations-between-loci-linkage-disequilibrium-and-recombination">Correlations between loci, linkage disequilibrium, and recombination</a></li>
</ul></li>
</ul>
</nav>
<h1 id="allele-and-genotype-frequencies">Allele and genotype frequencies</h1>
<h2 id="allele-frequencies">Allele frequencies</h2>
<p>Consider a diploid autosomal locus segregating at two alleles (<span class="math inline">\(A_1\)</span> and <span class="math inline">\(A_2\)</span>). Let <span class="math inline">\(N_{11}\)</span> and <span class="math inline">\(N_{12}\)</span> be the number of <span class="math inline">\(A_1A_1\)</span> homozygotes and <span class="math inline">\(A_1A_2\)</span> heterozygotes, respectively. Moreover, let <span class="math inline">\(N\)</span> be the total number of diploid individuals in the population. We can then define the relative frequencies of <span class="math inline">\(A_1A_1\)</span> and <span class="math inline">\(A_1A_2\)</span> genotypes as <span class="math inline">\(f_{11} = N_{11}/N\)</span> and <span class="math inline">\(f_{12} = N_{12}/N\)</span>, respectively. The frequency of allele <span class="math inline">\(A_1\)</span> in the population is then given by <span class="math display">\[p = \frac{2 N_{11} + N_{12}}{2N} = f_{11} + \frac{1}{2} f_{12}. \]</span> Note that this holds independently of Hardy–Weinberg proportions and equilibrium [see below]. The frequency of the alternate allele (<span class="math inline">\(A_2\)</span>) is then just <span class="math inline">\(q=1-p\)</span>.</p>
<h2 id="hardyweinberg-proportions">Hardy–Weinberg proportions</h2>
<p>Imagine a population mating at random with respect to genotypes, i.e.no inbreeding, no population structure, no sex differences in allele frequencies. The frequency of allele <span class="math inline">\(A_1\)</span> in the population at the time of reproduction is <span class="math inline">\(p\)</span>. An <span class="math inline">\(A_1A_1\)</span> genotype is made by reaching out into our population and independently drawing two <span class="math inline">\(A_1\)</span> alelle gametes to form a zygote. Therefore, the probability that an individual is an <span class="math inline">\(A_1A_1\)</span> homozygote is <span class="math inline">\(p^2\)</span>. This probability is also the expected frequency of the <span class="math inline">\(A_1A_1\)</span> homozygote in the population. The expected frequency of the three possible genotypes is</p>
<table>
<thead>
<tr class="header">
<th style="text-align: center;"><span class="math inline">\(f_{11}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(f_{12}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(f_{22}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\(p^2\)</span></td>
<td style="text-align: center;"><span class="math inline">\(2pq\)</span></td>
<td style="text-align: center;"><span class="math inline">\(q^2\)</span></td>
</tr>
</tbody>
</table>
<p>.</p>
<p>Note that we only need to assume random mating with respect to our alleles in order for these expected frequencies to hold, as long at <span class="math inline">\(p\)</span> is the frequency of the <span class="math inline">\(A_1\)</span> allele in the population at the time when gametes fuse.</p>
<p>Suppose the following genotype frequencies were observed for at an esterase locus in a population of Drosophila (A denotes the “fast” allele and B denotes the “slow” allele):</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: center;">AA</td>
<td style="text-align: center;">AB</td>
<td style="text-align: center;">BB</td>
</tr>
<tr class="even">
<td style="text-align: center;">0.6</td>
<td style="text-align: center;">0.2</td>
<td style="text-align: center;">0.2</td>
</tr>
</tbody>
</table>
<p>.</p>
<p>What genotype frequencies would you expect under Hardy-Weinberg expectations?</p>
<figure>
<img src="figures/CEU_YRI_separately_HWE.png" alt="Demonstrating Hardy-Weinberg proportions using 10,000 SNPs from the HapMap CEU European and YRI African populations. Within each of these populations I plot the allele frequency against the frequency of the 3 genotypes. Each SNP is represented by 3 different coloured points. The solid lines show the mean genotype frequency (calculated using a loess smoothing). The dashed line shows the predicted genotype frequency from Hardy Weinberg equilibrium. I find it really pretty that the analytical predictions work as well as they do. " /><figcaption>Demonstrating Hardy-Weinberg proportions using 10,000 SNPs from the HapMap CEU European and YRI African populations. Within each of these populations I plot the allele frequency against the frequency of the 3 genotypes. Each SNP is represented by 3 different coloured points. The solid lines show the mean genotype frequency (calculated using a loess smoothing). The dashed line shows the predicted genotype frequency from Hardy Weinberg equilibrium. I find it really pretty that the analytical predictions work as well as they do. </figcaption>
</figure>
<h2 id="coefficient-of-kinship">Coefficient of kinship</h2>
<p>We will define two alleles to be identical by descent if they are identical due to a common ancestor in the past few generations. For the moment, we ignore mutation, and we will be more precise about what we mean by ‘past few generations’ later on. For example, parent and child share exactly one allele identical by descent at a locus, assuming that the two parents of the child are randomly mated individuals from the population.<br />
A key quantity is the probability that our pair of individuals share 0, 1, or 2 alleles identical by descent. We denote these probabilities by <span class="math inline">\(r_0\)</span>, <span class="math inline">\(r_1\)</span>, and <span class="math inline">\(r_2\)</span> respectively. See Table [table:IBDprobs] for some examples.<br />
One summary of relatedness that will be important is the probability that two alleles picked at random, one from each of the two different individuals <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>, are identical by descent. We call this quantity the coefficient of kinship of individuals <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>, and denote it by <span class="math inline">\(F_{ij}\)</span>. It is calculated as <span class="math display">\[F_{ij}= 0 \times r_0 + \frac{1}{4} r_1  + \frac{1}{2} r_2. 
\label{eqn:coeffkinship}\]</span></p>
<p>The coefficient of kinship will appear multiple times, in both our discussion of inbreeding and in the context of phenotypic resemblance between relatives.<br />
</p>
<table>
<caption>Probability that two individuals of a given relationship share 0, 1, or 2 alleles identical by descent. <span class="math inline">\(^{*}\)</span> assuming this is the only relationship the pair of individuals share (above that expected from randomly sampling individuals from the population). <span data-label="table:IBDprobs"></span></caption>
<thead>
<tr class="header">
<th style="text-align: left;">Relationship (i,j)<span class="math inline">\(^{*}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(r_0\)</span></th>
<th style="text-align: center;"><span class="math inline">\(r_1\)</span></th>
<th style="text-align: center;"><span class="math inline">\(r_2\)</span></th>
<th style="text-align: center;"><span class="math inline">\(F_{ij}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">parent–child</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1/4</td>
</tr>
<tr class="even">
<td style="text-align: left;">full siblings</td>
<td style="text-align: center;">1/4</td>
<td style="text-align: center;">1/2</td>
<td style="text-align: center;">1/4</td>
<td style="text-align: center;">1/4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">identical (monzygotic) twins</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">1/2</td>
</tr>
<tr class="even">
<td style="text-align: left;"><span class="math inline">\(1^{st}\)</span> cousins</td>
<td style="text-align: center;">3/4</td>
<td style="text-align: center;">1/4</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1/16</td>
</tr>
</tbody>
</table>
<p>Consider a biallelic locus where allele 1 is at frequency <span class="math inline">\(p\)</span>, and two individuals who have IBD allele sharing probabilities <span class="math inline">\(r_0\)</span>, <span class="math inline">\(r_1\)</span>, <span class="math inline">\(r_2\)</span>.<br />
<span><strong>A)</strong></span> What is the probability that these two individuals are both homozygous for allele 1?<br />
<span><strong></strong></span> What is the probability that both individuals heterozygotes?</p>
<h2 id="inbreeding">Inbreeding</h2>
<p>We can define an inbred individual as an individual whose parents are more closely related to each other than two random individuals drawn from some reference population.<br />
When two related individuals produce an offspring, that individual can receive two alleles that are identical by descent, i.e. they can be homozygous by descent (sometimes termed autozygous), due to the fact that they have two copies of an allele through different paths through the pedigree. This increased likelihood of being homozygous relative to an outbred individual is the most obvious effect of inbreeding. It is also the one that will be of most interest to us, as it underlies a lot of our ideas about inbreeding depression and population structure.<br />
As the offspring receives a random allele from each parent (<span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>), the probability that those two alleles are identical by descent is equal to the kinship coefficient <span class="math inline">\(F_{ij}\)</span> of the two parents (Eqn. [eqn:coeffkinship]). This follows from the fact that the genotype of the offspring is made by sampling an allele at random from each of our parents. We will use IBD for identical by descent.<br />
The only way the offspring can be heterozygous (<span class="math inline">\(A_1 A_2\)</span>) is if their two alleles at a locus are not IBD (otherwise they would necessarily be homozygous). Therefore, the probability that they are heterozygous is <span class="math display">\[(1-F) 2p q,
\label{eq:hetGenHW}\]</span> where we have dropped the indices <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> for simplicity. The offspring can be homozygous for the <span class="math inline">\(A_1\)</span> allele in two different ways. They can have two non-IBD alleles that are not IBD but happen to be of the allelic type <span class="math inline">\(A_1\)</span>, or their two alleles can be IBD, such that they inherited allele <span class="math inline">\(A_1\)</span> by two different routes from the same ancestor. Thus, the probability that an offspring is homozygous is <span class="math display">\[(1-F) p^2 + F p.\]</span> Therefore, the frequencies of the three possible genotypes can be written as given in Table [table:GeneralizedHWE], which provides a generalization of the Hardy–Weinberg proportions.<br />
</p>
<table>
<caption><strong>Generalized Hardy–Weinberg</strong><span data-label="table:GeneralizedHWE"></span></caption>
<thead>
<tr class="header">
<th style="text-align: center;"><span class="math inline">\(f_{11}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(f_{12}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(f_{22}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\((1-F) p^2 + F p\)</span></td>
<td style="text-align: center;"><span class="math inline">\((1-F) 2pq\)</span></td>
<td style="text-align: center;"><span class="math inline">\((1-F) q^2 + F q\)</span></td>
</tr>
</tbody>
</table>
<p>Note that the generalized Hardy–Weinberg proportions completely specify the genotype probabilities, as there are two parameters (<span class="math inline">\(p\)</span> and <span class="math inline">\(F\)</span>) and two degrees of freedom (as <span class="math inline">\(p\)</span> and <span class="math inline">\(q\)</span> have to sum to one). Therefore, any combination of genotype frequencies at a biallelic site can be specified by a combination of <span class="math inline">\(p\)</span> and <span class="math inline">\(F\)</span>.<br />
</p>
<h2 id="calculating-inbreeding-coefficients-from-data">Calculating inbreeding coefficients from data</h2>
<p>If the observed heterozygosity in a population is <span class="math inline">\(H_O\)</span>, and we assume that the generalized Hardy–Weinberg proportions hold, we can set <span class="math inline">\(H_O\)</span> equal to <span class="math inline">\(f_{12}\)</span>, and solve Eq.  for <span class="math inline">\(F\)</span> to obtain an estimate of the inbreeding coefficient as <span class="math display">\[\hat{F} = 1-\frac{f_{12}}{2pq} = \frac{2pq - f_{12}}{2pq}.
\label{eqn:Fhat}\]</span> As before, <span class="math inline">\(p\)</span> is the frequency of allele <span class="math inline">\(A_{1}\)</span> in the population. This can be rewritten in terms of the observed heterozygosity (<span class="math inline">\(H_O\)</span>) and the heterozygosity expected in the absence of inbreeding, <span class="math inline">\(H_E=2pq\)</span>, as <span class="math display">\[\hat{F} = \frac{H_E-H_O}{H_E} = 1 - \frac{H_O}{H_E}.
\label{eqn:FhatHO}\]</span> Hence, <span class="math inline">\(F\)</span> quantifies the deviation due to inbreeding of the observed heterozygosity from the one expected under random mating, relative to the latter. If we have multiple loci, we can replace <span class="math inline">\(H_O\)</span> and <span class="math inline">\(H_E\)</span> by their means over loci, <span class="math inline">\(\bar{H}_O\)</span> and <span class="math inline">\(\bar{H}_E\)</span>, respectively. Note that, in principle, we could also calculate <span class="math inline">\(F\)</span> for each individual locus first, and then take the average across loci. However, this procedure is more prone to introducing a bias if sample sizes vary across loci, which is not unlikely when we are dealing with real data.<br />
</p>
<p>For the genotype frequencies observed at the esterase locus given above estimate the inbreeding coefficient (F).</p>
<h2 id="summarizing-population-structure">Summarizing population structure</h2>
<p>We defined inbreeding as having parents that are more closely related to each other than two individuals drawn at random from some reference population. The question that naturally arises is: Which reference population should we use? While I might not look inbred in comparison to allele frequencies in the United Kingdom (UK), where I am from, my parents certainly are not two individuals drawn at random from the world-wide population. If we estimated my inbreeding coefficient <span class="math inline">\(F\)</span> using allele frequencies within the UK, it would be close to zero, but would likely be larger if we used world-wide frequencies. This is because there is a somewhat lower level of expected heterozygosity within the UK than in the human population across the world as a whole.<br />
Wright (1943, 1951) developed a set of ‘F-statistics’ (also called ‘fixation indices’) that formalize the idea of inbreeding with respect to different levels of population structure. He defined <span class="math inline">\(F_{\mathrm{XY}}\)</span> as the correlation between random gametes, drawn from the same level <span class="math inline">\(X\)</span>, relative to level <span class="math inline">\(Y\)</span>. We will return to why <span class="math inline">\(F\)</span>-statistics are statements about correlations between alleles in just a moment. One commonly uses <span class="math inline">\({F_{\mathrm{IS}}}\)</span> for the inbreeding coefficient between an individual (<span class="math inline">\(I\)</span>) and the subpopulation (<span class="math inline">\(S\)</span>). Consider a single locus, where in a subpopulation (<span class="math inline">\(S\)</span>) a fraction <span class="math inline">\(H_I=f_{12}\)</span> of individuals are heterozygous. In this subpopulation, let the frequency of allele <span class="math inline">\(A_1\)</span> be <span class="math inline">\(p_S\)</span>, such that the expected heterozygosity under random mating is <span class="math inline">\(H_S = 2 p_S (1 - p_S)\)</span>. We will write <span class="math inline">\({F_{\mathrm{IS}}}\)</span> as <span class="math display">\[{F_{\mathrm{IS}}}= 1-\frac{H_I}{H_S}= 1-\frac{f_{12}}{2p_Sq_S},
\label{eqn:FIS}\]</span> a direct analog of eqn. [eqn:Fhat]. Hence, <span class="math inline">\({F_{\mathrm{IS}}}\)</span> is the relative difference between observed and expected heterozygosity due to a deviation from random mating within the subpopulation. We could also compare the observed heterozygosity in individuals (<span class="math inline">\(H_I\)</span>) to that expected in the total population, <span class="math inline">\(H_T\)</span>. If the frequency of allele <span class="math inline">\(A_1\)</span> in the total population is <span class="math inline">\(p_T\)</span>, then we can write <span class="math inline">\({F_{\mathrm{IT}}}\)</span> as <span class="math display">\[{F_{\mathrm{IT}}}=1-\frac{H_I}{H_T}= 1-\frac{f_{12}}{2p_Tq_T},
\label{eqn:FIT}\]</span> which compares heterozygosity in individuals to that expected in the total population. As a simple extension of this, we could imagine comparing the expected heterozygosity in the subpopulation (<span class="math inline">\(H_S\)</span>) to that expected in the total population <span class="math inline">\(H_T\)</span>, via <span class="math inline">\({F_{\mathrm{ST}}}\)</span>: <span class="math display">\[{F_{\mathrm{ST}}}= 1-\frac{H_S}{H_T}=1-\frac{2p_Sq_S}{2p_Tq_T} \label{eqn:FST}.\]</span> If the total population contains the subpopulation then, as we will see below, due to the Wahlund effect (to be added) <span class="math inline">\(2p_Sq_S \leq
2p_Tq_T\)</span>, and so <span class="math inline">\({F_{\mathrm{IS}}}\leq {F_{\mathrm{IT}}}\)</span> and <span class="math inline">\({F_{\mathrm{ST}}}\geq 0\)</span>. We can relate the three <span class="math inline">\(F\)</span>-statistics to each other as <span class="math display">\[(1-\hat{{F_{\mathrm{IT}}}}) =\frac{H_I}{H_S} \frac{H_S}{H_T}=(1-{F_{\mathrm{IS}}})(1-{F_{\mathrm{ST}}}).
\label{eqn:F_relationships}\]</span> Hence, the reduction in heterozygosity within individuals compared to that expected in the total population can be decomposed to the reduction in heterozygosity of individuals compared to the subpopulation, and the reduction in heterozygosity from the total population to that in the subpopulation.<br />
If we want a summary of population structure across multiple subpopulations, we can average <span class="math inline">\(H_I\)</span> and/or <span class="math inline">\(H_S\)</span> across populations, and use a <span class="math inline">\(p_T\)</span> calculated by averaging <span class="math inline">\(p_S\)</span> across subpopulations (or our samples from sub-populations). For example, the average <span class="math inline">\({F_{\mathrm{ST}}}\)</span> across <span class="math inline">\(K\)</span> subpopulations (sampled with equal effort) is <span class="math display">\[{F_{\mathrm{ST}}}= 1 - \frac{\bar{H}_{S}}{H_T},\]</span> where <span class="math inline">\(\bar{H}_S = 1/K \sum_{i = 1}^{K} H_{S}^{(i)}\)</span>, and <span class="math inline">\(H_{S}^{(i)} = 2 p_{i} q_{i}\)</span> is the expected heterozygosity in subpopulation <span class="math inline">\(i\)</span>. Furthermore, if we have multiple sites, we can replace <span class="math inline">\(H_I\)</span>, <span class="math inline">\(H_S\)</span>, and <span class="math inline">\(H_T\)</span> with their averages across loci (as above).<br />
Let us now return to Wright’s definition of the <span class="math inline">\(F\)</span>-statistics as correlations between random gametes, drawn from the same level <span class="math inline">\(X\)</span>, relative to level <span class="math inline">\(Y\)</span>. Without loss of generality, we may think about <span class="math inline">\(X\)</span> as individuals and <span class="math inline">\(S\)</span> as the subpopulation. Rewriting <span class="math inline">\({F_{\mathrm{IS}}}\)</span> in terms of the observed homozygote frequencies (<span class="math inline">\(f_{11}\)</span>, <span class="math inline">\(f_{22}\)</span>) and expected homozygosities (<span class="math inline">\(p_{S}^2\)</span>, <span class="math inline">\(q_{S}^2\)</span>) we find <span class="math display">\[{F_{\mathrm{IS}}}= \frac{2p_Sq_S - f_{12}}{2p_Sq_S} = \frac{f_{11}+f_{22} -
p_S^2 - q_S^2}{2p_Sq_S},
\label{eqn:Fascorr}\]</span> using the fact that <span class="math inline">\(p^2+2pq+q^2=1\)</span>, and <span class="math inline">\(f_{12} = 1 - f_{11} - f_{12}\)</span>. The form of eqn. ([eqn:Fascorr]) reveals that <span class="math inline">\({F_{\mathrm{IS}}}\)</span> is the covariance between pairs of alleles found in an individual, divided by the expected variance under binomial sampling. Thus, <span class="math inline">\(F\)</span>-statistics can be understood as the correlation between alleles drawn from a population (or an individual) above that expected by chance (i.e. drawing alleles sampled at random from some broader population).<br />
We can also interpret <span class="math inline">\(F\)</span>-statistics as proportions of variance explained by different levels of population structure. To see this, let us think about <span class="math inline">\({F_{\mathrm{ST}}}\)</span> averaged over <span class="math inline">\(K\)</span> subpopulations, whose frequencies are <span class="math inline">\(p_1,\dots,p_K\)</span>. The frequency in the total population is <span class="math inline">\(p_T=\bar{p} = 1/K \sum_{i=1}^K p_i\)</span>. Then, we can write <span class="math display">\[{F_{\mathrm{ST}}}= \frac{2 \bar{p}\bar{q} - \frac{1}{K}\sum_{i=1}^K 2p_iq_i }{2
\bar{p}\bar{q}} = \frac{ \left(\frac{1}{K} \sum_{i=1}^K p_i^2 +
\frac{1}{K} \sum_{i=1}^K q_i^2 \right) -  \bar{p}^2-\bar{q}^2 }{2
\bar{p}\bar{q}} = \frac{\mathrm{Var}(p_i)}{\mathrm{Var}(\bar{p})},
\label{eqn:F_as_propvar}\]</span> which shows that <span class="math inline">\({F_{\mathrm{ST}}}\)</span> is the proportion of the variance explained by the subpopulation labels.</p>
<h2 id="other-approaches-to-population-structure">Other approaches to population structure</h2>
<p>There is a broad spectrum of methods to describe patterns of population structure in populaion genetic datasets. We’ll briefly discuss two broad-classes of methods, assigment methods and principal components analysis,that appear often in the literature.</p>
<h3 id="assignment-methods">Assignment Methods</h3>
<p>Here we’ll describe a simple probabilistic assignment to find the probability that an individual of unknown population comes from one of <span class="math inline">\(K\)</span> predefined populations. We’ll then briefly explain how to extend this to cluster individuals into <span class="math inline">\(K\)</span> initially unknown populations. This method is a simplified version of what Bayesian population genetics clustering algorithms such as STRUCTURE and ADMIXTURE do (Pritchard et al. Genetics 2000).</p>
<h4 id="a-simple-assignment-method">A simple assignment method</h4>
<p>We have genotype data from unlinked S bi-allelic loci for <span class="math inline">\(K\)</span> populations. The allele frequency of allele <span class="math inline">\(A_1\)</span> at locus <span class="math inline">\(l\)</span> in population <span class="math inline">\(k\)</span> is denoted by <span class="math inline">\(p_{k,l}\)</span>, so that the allele frequencies in population 1 are <span class="math inline">\(p_{1,1},\cdots p_{1,L}\)</span> and population 2 are <span class="math inline">\(p_{2,1},\cdots p_{2,L}\)</span> and so on.</p>
<p>You type a new individual from an unknown population at these <span class="math inline">\(L\)</span> loci. This individual’s genotype at locus <span class="math inline">\(l\)</span> is <span class="math inline">\(g_l\)</span>, where <span class="math inline">\(g_l\)</span> denotes the number of copies of allele <span class="math inline">\(A_1\)</span> this individual carries at this locus (<span class="math inline">\(g_l=0,1,2\)</span>).</p>
<p>The probability of this individual’s genotype at locus <span class="math inline">\(l\)</span> conditional on coming from population <span class="math inline">\(k\)</span> (i.e. their alleles being a random HW draw from population <span class="math inline">\(k\)</span>) is <span class="math display">\[P(g_l | \textrm{pop k}) = I(g_l=0) (1-p_{k,l})^2 +  I(g_l=1) 2 p_{k,l} (1-p_{k,l}) + I(g_l=2) p_{k,l}^2\]</span> where <span class="math inline">\(I(g_l=0)\)</span> is an indicator function which is <span class="math inline">\(1\)</span> if <span class="math inline">\(g_l=0\)</span> and zero otherwise, and likewise for the other indicator functions. This follows simply from HWE.</p>
<p>Assuming that the loci are independent, the probability of individual’s genotypes conditional on them coming from population <span class="math inline">\(k\)</span> is <span class="math display">\[P(\textrm{ind.} | \textrm{pop k})  = \prod_{l=1}^S P(g_l | \textrm{pop k}) \label{eqn_assignment}\]</span></p>
<p>We wish to know the probability that this new individual comes from population <span class="math inline">\(k\)</span>, i.e. <span class="math inline">\(P(\textrm{pop k} | \textrm{new ind.})\)</span>. We can obtain this through Bayes rule <span class="math display">\[P(\textrm{pop k} | \textrm{ind.})  = \frac{P(\textrm{ind.} | \textrm{pop k}) P(\textrm{pop k})}{P(\textrm{ind.})}\]</span> where <span class="math display">\[P(\textrm{ind.}) = \sum_{k=1}^K  P(\textrm{ind.} | \textrm{pop k}) P(\textrm{pop k})\]</span> is the normalizing constant. We interpret <span class="math inline">\(P(\textrm{pop k})\)</span> as the prior probability of the individual coming from population <span class="math inline">\(k\)</span>, unless we have some other prior knowledge we will assume that the new individual has a equal probability of coming from each population <span class="math inline">\(P(\textrm{pop k})=1/K\)</span>.</p>
<p>We intepret <span class="math display">\[P(\textrm{pop k} | \textrm{ind.})\]</span> as the posterior probability that our new individual comes from each of our <span class="math inline">\(1,\cdots, K\)</span> populations.</p>
<p>More sophisticated versions of this are now used to allow for hybrids, e.g, we can have a proportion <span class="math inline">\(q_k\)</span> of our individual’s genome come from population <span class="math inline">\(k\)</span> and estimate the set of <span class="math inline">\(q_k\)</span>’s.</p>
<p>We have two populations where the frequency of allele <span class="math inline">\(A_1\)</span> at two SNPs (<span class="math inline">\(A_1/A_2\)</span>) is given by</p>
<table>
<thead>
<tr class="header">
<th style="text-align: center;">Population</th>
<th style="text-align: center;">locus 1</th>
<th style="text-align: center;">locus 2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">A</td>
<td style="text-align: center;"><span class="math inline">\(0.1\)</span></td>
<td style="text-align: center;"><span class="math inline">\(0.85\)</span></td>
</tr>
<tr class="even">
<td style="text-align: center;">B</td>
<td style="text-align: center;"><span class="math inline">\(0.95\)</span></td>
<td style="text-align: center;"><span class="math inline">\(0.2\)</span></td>
</tr>
</tbody>
</table>
<p>We sample an individual whose genotype is <span class="math inline">\(A_1A_1\)</span> at the first locus and <span class="math inline">\(A_2A_2\)</span> at the second. What is the probability that our indvidual comes from population 1 vs population 2? Lets assume that with probability <span class="math inline">\(q_1\)</span> our individual draws an allele from population <span class="math inline">\(1\)</span> and that with probability <span class="math inline">\(q_2=1-q_1\)</span> they draw an allele from population <span class="math inline">\(2\)</span>. What is the probability of our individual’s genotype given <span class="math inline">\(q_1\)</span>? You could plot this probability as a function of <span class="math inline">\(q_1\)</span>. How does your plot change if our individual is heterozygote at both loci?</p>
<h4 id="clustering-based-on-assignment-methods">Clustering based on assignment methods</h4>
<p>While it is great to be able to assign our individuals to particular population, these ideas can be pushed to learn about how best to describe our genotype data in terms of discrete populations without assigning any of our individuals to populations <span><em>a priori</em></span>. We wish to cluster our individuals into <span class="math inline">\(K\)</span> unknown populations. We begin by assigning our individuals at random to these <span class="math inline">\(K\)</span> populations.</p>
<ul>
<li><p>Given these assignments we estimate the allele frequencies at all of our loci in each population.</p></li>
<li><p>Given these allele frequencies we chose to reassign each individual to a population <span class="math inline">\(k\)</span> with a probability given by eqn. (<span class="math inline">\(\ref{eqn_assignment}\)</span>).</p></li>
</ul>
<p>We iterate steps 1 and 2 for many iterations. If the data is sufficiently informative the assignments and allele frequencies will quickly converge.</p>
<p>To do this in a full bayesian scheme we need to place priors on the allele frequencies (e.g. a beta distribution).Technically we are using this is the joint posterior of our allele frequencies and assignments.</p>
<h3 id="principal-components-analysis">Principal components analysis</h3>
<p>The use of principal component analysis in population genetics was pioneered by Cavalli-Sforza. With large genotyping datasets PCA has made a come back. See Patterson et al 2006, PLoS Genetics and McVean, G. 2010 PLoS Genetics and for recent discussion.</p>
<p>Consider a dataset consisting of N individuals at S bi-allelic SNPs. The <span class="math inline">\(i^{th}\)</span> individual’s genotype data at locus <span class="math inline">\(\ell\)</span> takes value <span class="math inline">\(g_{i,\ell}\)</span>=0,1, or 2 (corresponding to the number of copies of allele <span class="math inline">\(A_1\)</span> an individual carrys at this SNP). We can think of this as a N x S matrix (where usually <span class="math inline">\(N \ll S\)</span>).</p>
<p>Denoting the sample mean allele freq at SNP <span class="math inline">\(\ell\)</span> by <span class="math inline">\(p_{\ell}\)</span> we usually standardize the genotype in the following way <span class="math display">\[\frac{g_{i,\ell} - 2 p_{\ell}}{\sqrt{p_{\ell}(1-p_{\ell})}}\]</span> i.e. at each SNP we center the genotypes by minusing of the mean genotype (<span class="math inline">\(2\epsilon_{\ell}\)</span>) and divide through by the expected variance assuming that alleles are sampled binomially from the mean frequency (<span class="math inline">\(\sqrt{p_{\ell} (1-p_{\ell})}\)</span>). Doing this to all of our genotypes we form a data matrix (of dimension N x S). We can then perform principal components analysis of this data matrix to cover the major axes of genotype variance in our sample.</p>
<p>It is worth taking a moment to delve further into what we are doing here. There’s a number of equivalent ways to thinking about what PCA is doing, one of these is to think that when we do PCA we are building the individual by individual covariance matrix and performing eigen-value decomposition of this matrix (with the eigen-vectors giving the PC). This individual by individual covariance matrix has entries the <span class="math inline">\((i,~j)^{th}\)</span> entry given by <span class="math display">\[\sum_{\ell=1}^S \frac{(g_{i,\ell} - 2p_{\ell})(g_{j,\ell} - 2p_{\ell})}{p_{\ell}(1-p_{\ell})}\]</span> note that this is the covariance, is very similar to those we encountered in discussing <span class="math inline">\(F\)</span>-statistics as correlations (equation ), expect now we are asking about the allelic covariance between two individuals above that expected if they were both drawn from the total sample at random (rather than the covariance of alleles within a single individual). So by performing PCA on the data we are learning about the major (orthogonal) axes of the kinship matrix.</p>
<h2 id="correlations-between-loci-linkage-disequilibrium-and-recombination">Correlations between loci, linkage disequilibrium, and recombination</h2>
<p>Up to now we have been interested in correlations between alleles at the same locus, e.g. correlations within individuals (inbreeding) or between individuals (relatedness). We have seen how relatedness between parents affects the extent to which their offspring is inbred. We now turn to correlations between alleles at different loci. To understand correlations between loci we need to understand recombination.<br />
</p>
<h4 id="recombination">Recombination</h4>
<p>Lets consider an individual heterozygous for a <span class="math inline">\(AB\)</span> and <span class="math inline">\(ab\)</span> haplotype. If no recombination occurs between our two loci in this individual, then these two haplotypes will be transmitted intact to the next generation. While if a recombination (or more generally an odd number of recombinations) occurs between our two loci on the haplotype transmitted to the child then <span class="math inline">\(\tfrac{1}{2}\)</span> the time the child receives a <span class="math inline">\(Ab\)</span> haplotype and <span class="math inline">\(\tfrac{1}{2}\)</span> the time the child receives a <span class="math inline">\(aB\)</span> haplotype. So recombination is breaking up the association between loci. We’ll define the recombination fraction (<span class="math inline">\(r\)</span>) to be the probability of an odd number of recombinations between our loci. In practice we’ll often be interested in relatively short regions where recombination is relatively rare, and so we might think that <span class="math inline">\(r=r_{BP}L \ll 1\)</span>, where <span class="math inline">\(r_{BP}\)</span> is the average recombination rate per base pair (typically <span class="math inline">\(\sim 10^{-8}\)</span>) and L is the number of base pairs separating our two loci.<br />
</p>
<h4 id="linkage-disequilibrium">Linkage disequilibrium</h4>
<p>The (horrible) phrase linkage disequilibrium (LD) refers to the statistical non-independence (i.e. a correlation) of alleles at different loci. Our two loci, which segregate alleles <span class="math inline">\(A/a\)</span> and <span class="math inline">\(B/b\)</span>, have allele frequencies of <span class="math inline">\(p_A\)</span> and <span class="math inline">\(p_B\)</span> respectively. The frequency of the two locus haplotype is <span class="math inline">\(p_{AB}\)</span>, and likewise for our other three combinations. If our loci were statistically independent then <span class="math inline">\(p_{AB} = p_Ap_B\)</span>, otherwise <span class="math inline">\(p_{AB} \neq p_Ap_B\)</span> We can define a covariance between the <span class="math inline">\(A\)</span> and <span class="math inline">\(B\)</span> alleles at our two loci as <span class="math display">\[D_{AB} = p_{AB} - p_Ap_B\]</span> and likewise for our other combinations at our two loci (<span class="math inline">\(D_{Ab},~D_{aB},~D_{ab}\)</span>). These <span class="math inline">\(D\)</span> statistics are all closely related to each other as <span class="math inline">\(D_{AB} = - D_{Ab}\)</span> and so on. Thus we only need to specify one <span class="math inline">\(D_{AB}\)</span> to know them all, so we’ll drop the subscript and just refer to <span class="math inline">\(D\)</span>. Also a handy result is that we can rewrite our haplotype frequency <span class="math inline">\(p_{AB}\)</span> as <span class="math display">\[p_{AB} = p_Ap_B+D. \label{eqn:ABviaD}\]</span> If <span class="math inline">\(D=0\)</span> we’ll say the two loci are in linkage equilibrium, while if <span class="math inline">\(D&gt;0\)</span> or <span class="math inline">\(D&lt;0\)</span> we’ll say that the loci are in linkage disequilibrium (we’ll perhaps want to test whether <span class="math inline">\(D\)</span> is statistically different from <span class="math inline">\(0\)</span> before making this choice). You should be careful to keep the concepts of linkage and linkage disequilibrium separate in your mind. Genetic linkage refers to the linkage of multiple loci due to the fact that they are transmitted through meiosis together (most often because the loci are on the same chromosome). Linkage disequilibrium merely refers to the correlation between the alleles at different loci, this may in part be due to the genetic linkage of these loci but does not necessarily imply this (e.g. genetically unlinked loci can be in LD due to population structure).<br />
Another common statistic for summarizing LD is <span class="math inline">\(r^2\)</span> which we write as <span class="math display">\[r^2 = \frac{D^2}{p_A(1-p_A) p_B(1-p_B) }\]</span> as <span class="math inline">\(D\)</span> is a covariance, and <span class="math inline">\(p_A(1-p_A) \)</span> is the variance of an allele drawn at random from locus <span class="math inline">\(A\)</span>, <span class="math inline">\(r^2\)</span> is the squared correlation coefficient.<br />
</p>
<p>You genotype 2 bi-allelic loci (A &amp; B) segregating in two mouse subspecies (1 &amp; 2) which mate randomly among themselves, but have not historically interbreed since they speciated. On the basis of previous work you estimate that the two loci are separated by a recombination fraction of 0.1. The frequencies of haplotypes in each population are:</p>
<table>
<thead>
<tr class="header">
<th style="text-align: center;">Pop</th>
<th style="text-align: center;"><span class="math inline">\(p_{AB}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(p_{Ab}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(p_{aB}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(p_{ab}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">1</td>
<td style="text-align: center;">.02</td>
<td style="text-align: center;">.18</td>
<td style="text-align: center;">.08</td>
<td style="text-align: center;">.72</td>
</tr>
<tr class="even">
<td style="text-align: center;">2</td>
<td style="text-align: center;">.72</td>
<td style="text-align: center;">.18</td>
<td style="text-align: center;">.08</td>
<td style="text-align: center;">.02</td>
</tr>
</tbody>
</table>
<p><span><strong>A)</strong></span> How much LD is there within populations, i.e. estimate D?<br />
<span><strong>B)</strong></span> If we mixed the two populations together in equal proportions what value would D take before any mating has had the chance to occur?<br />
</p>
<h4 id="the-decay-of-ld-due-to-recombination">The decay of LD due to recombination</h4>
<p>We will now examine what happens to LD over the generations if we only allow recombination to occur in a very large population (i.e. no genetic drift, i.e. the frequencies of our loci follow their expectations). To do so consider the frequency of our <span class="math inline">\(AB\)</span> haplotype in the next generation <span class="math inline">\(p_{AB}^{\prime}\)</span>. We lose a fraction <span class="math inline">\(r\)</span> of our <span class="math inline">\(AB\)</span> haplotypes to recombination ripping our alleles apart but gain a fraction <span class="math inline">\(rp_A p_B\)</span> per generation from other haplotypes recombining together to form <span class="math inline">\(AB\)</span> haplotypes. Thus in the next generation <span class="math display">\[p_{AB}^{\prime} = (1-r)p_{AB} + rp_Ap_B\]</span> this last term here is <span class="math inline">\(r(p_{AB}+p_{Ab})(p_{AB}+p_{aB})\)</span>, which multiplying this out is the probability of recombination in the different diploid genotypes that could generate a <span class="math inline">\(p_{AB}\)</span> haplotype.<br />
We can then write the change in the frequency of the <span class="math inline">\(p_{AB}\)</span> haplotype as <span class="math display">\[\Delta p_{AB} = p_{AB}^{\prime} -p_{AB} = -r p_{AB} + rp_Ap_B = - r D\]</span> so recombination will cause a decrease in the frequency of <span class="math inline">\(p_{AB}\)</span> if there is an excess of <span class="math inline">\(AB\)</span> haplotypes within the population (<span class="math inline">\(D&gt;0\)</span>), and an increase if there is a deficit of <span class="math inline">\(AB\)</span> haplotypes within the population (<span class="math inline">\(D&lt;0\)</span>). Our LD in the next generation is <span class="math inline">\(D^{\prime} =
p_{AB}^{\prime}\)</span>, so we can rewrite the above eqn. in terms of the <span class="math inline">\(D^{\prime} \)</span> <span class="math display">\[D^{\prime}= (1-r) D\]</span> so if the level of LD in generation <span class="math inline">\(0\)</span> is <span class="math inline">\(D_0\)</span> the level <span class="math inline">\(t\)</span> generations later (<span class="math inline">\(D_t\)</span>) is <span class="math display">\[D_t=  (1-r)^t D_0\]</span> so recombination is acting to decrease LD, and it does so geometrically at a rate given by <span class="math inline">\((1-r)\)</span>. If <span class="math inline">\(r \ll 1\)</span> then we can approximate this by an exponential and say that <span class="math display">\[D_t \approx  D_0 e^{-rt}\]</span><br />
</p>
<p>You find a hybrid population between the two mouse subspecies described in the question above, which appears to be comprised of equal proportions of ancestry from the two subspecies. You estimate LD between the two markers to be 0.0723. Assuming that this hybrid population is large and was formed by a single mixture event, can you estimate how long ago this population formed?<br />
</p>
</body>
</html>
