<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
  <meta name="author" content="Graham Coop^{1} ^1 Department of Evolution and Ecology &amp; Center for Population Biology, University of California, Davis. To whom correspondence should be addressed: gmcoop@ucdavis.edu This work is licensed under a Creative Commons Attribution 3.0 Unported License. http://creativecommons.org/licenses/by/3.0/ i.e. you are free to reuse and remix this work, but please include an attribution to the original.">
  <title>Notes on Population Genetics</title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="css/style.css">
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header>
<h1 class="title">Notes on Population Genetics</h1>
<h2 class="author">Graham Coop<span class="math inline">\(^{1}\)</span><br />
<span class="math inline">\(^1\)</span> Department of Evolution and Ecology &amp; Center for Population Biology,<br />
University of California, Davis.<br />
To whom correspondence should be addressed: <code>gmcoop@ucdavis.edu</code><br />
This work is licensed under a Creative Commons Attribution 3.0 Unported License.<br />
http://creativecommons.org/licenses/by/3.0/<br />
i.e. you are free to reuse and remix this work, but please include an attribution to the original.</h2>
</header>
<nav id="TOC">
<ul>
<li><a href="#allele-and-genotype-frequencies">Allele and genotype frequencies</a><ul>
<li><a href="#allele-frequencies">Allele frequencies</a></li>
<li><a href="#hardyweinberg-proportions">Hardy–Weinberg proportions</a></li>
<li><a href="#coefficient-of-kinship">Coefficient of kinship</a></li>
<li><a href="#inbreeding">Inbreeding</a></li>
<li><a href="#calculating-inbreeding-coefficients-from-data">Calculating inbreeding coefficients from data</a></li>
<li><a href="#summarizing-population-structure">Summarizing population structure</a></li>
<li><a href="#other-approaches-to-population-structure">Other approaches to population structure</a><ul>
<li><a href="#assignment-methods">Assignment Methods</a></li>
<li><a href="#principal-components-analysis">Principal components analysis</a></li>
</ul></li>
<li><a href="#correlations-between-loci-linkage-disequilibrium-and-recombination">Correlations between loci, linkage disequilibrium, and recombination</a></li>
</ul></li>
<li><a href="#genetic-drift-and-neutral-diversity">Genetic Drift and Neutral Diversity</a><ul>
<li><a href="#LossofHet">Loss of heterozygosity due to drift.</a></li>
<li><a href="#DriftMutationBalance">Levels of diversity maintained by a balance between mutation and drift</a></li>
<li><a href="#the-effective-population-size.">The effective population size.</a></li>
<li><a href="#the-coalescent-and-patterns-of-neutral-diversity">The Coalescent and patterns of neutral diversity</a></li>
<li><a href="#the-coalescent-process-of-a-sample-of-alleles.">The coalescent process of a sample of alleles.</a></li>
<li><a href="#the-fixation-of-neutral-alleles">The fixation of neutral alleles</a></li>
<li><a href="#neutral-diversity-and-population-structure">Neutral diversity and population structure</a></li>
</ul></li>
<li><a href="#the-phenotypic-resemblance-between-relatives">The phenotypic resemblance between relatives</a><ul>
<li><a href="#additive-genetic-variance-and-heritability">Additive genetic variance and heritability</a></li>
<li><a href="#the-covariance-between-relatives">The covariance between relatives</a></li>
<li><a href="#the-response-to-selection">The response to selection</a></li>
</ul></li>
<li><a href="#one-locus-models-of-selection">One-locus models of selection</a><ul>
<li><a href="#fitness">Fitness</a></li>
<li><a href="#haploid-selection-model">Haploid selection model</a></li>
<li><a href="#diploid-model">Diploid model</a><ul>
<li><a href="#diploid-directional-selection">Diploid directional selection</a></li>
<li><a href="#heterozygote-advantage">Heterozygote advantage</a></li>
</ul></li>
<li><a href="#mutationselection-balance">Mutation–selection balance</a><ul>
<li><a href="#inbreeding-depression">Inbreeding depression</a></li>
</ul></li>
<li><a href="#migrationselection-balance">Migration–selection balance</a><ul>
<li><a href="#some-theory-of-the-spatial-distribution-of-allele-frequencies-under-deterministic-models-of-selection">Some theory of the spatial distribution of allele frequencies under deterministic models of selection</a></li>
</ul></li>
</ul></li>
<li><a href="#stochasticity-and-genetic-drift-in-allele-frequencies">Stochasticity and Genetic Drift in allele frequencies</a><ul>
<li><a href="#stochastic-loss-of-strongly-selected-alleles">Stochastic loss of strongly selected alleles</a></li>
<li><a href="#the-interaction-between-genetic-drift-and-weak-selection.">The interaction between genetic drift and weak selection.</a><ul>
<li><a href="#the-fixation-of-slightly-deleterious-alleles.">The fixation of slightly deleterious alleles.</a></li>
<li><a href="#Section:fixation_weakly_sel">A Sketch Proof of the probability of fixation of weakly selected alleles</a></li>
</ul></li>
</ul></li>
<li><a href="#the-effect-of-linked-selection-on-patterns-of-neutral-diversity">The effect of linked selection on patterns of neutral diversity</a><ul>
<li><a href="#a-simple-recurrent-model-of-selective-sweeps">A simple recurrent model of selective sweeps</a></li>
</ul></li>
</ul>
</nav>
<h1 id="allele-and-genotype-frequencies">Allele and genotype frequencies</h1>
<h2 id="allele-frequencies">Allele frequencies</h2>
<p>Consider a diploid autosomal locus segregating at two alleles (<span class="math inline">\(A_1\)</span> and <span class="math inline">\(A_2\)</span>). Let <span class="math inline">\(N_{11}\)</span> and <span class="math inline">\(N_{12}\)</span> be the number of <span class="math inline">\(A_1A_1\)</span> homozygotes and <span class="math inline">\(A_1A_2\)</span> heterozygotes, respectively. Moreover, let <span class="math inline">\(N\)</span> be the total number of diploid individuals in the population. We can then define the relative frequencies of <span class="math inline">\(A_1A_1\)</span> and <span class="math inline">\(A_1A_2\)</span> genotypes as <span class="math inline">\(f_{11} = N_{11}/N\)</span> and <span class="math inline">\(f_{12} = N_{12}/N\)</span>, respectively. The frequency of allele <span class="math inline">\(A_1\)</span> in the population is then given by <span class="math display">\[p = \frac{2 N_{11} + N_{12}}{2N} = f_{11} + \frac{1}{2} f_{12}. \]</span> Note that this holds independently of Hardy–Weinberg proportions and equilibrium [see below]. The frequency of the alternate allele (<span class="math inline">\(A_2\)</span>) is then just <span class="math inline">\(q=1-p\)</span>.</p>
<h2 id="hardyweinberg-proportions">Hardy–Weinberg proportions</h2>
<p>Imagine a population mating at random with respect to genotypes, i.e.no inbreeding, no population structure, no sex differences in allele frequencies. The frequency of allele <span class="math inline">\(A_1\)</span> in the population at the time of reproduction is <span class="math inline">\(p\)</span>. An <span class="math inline">\(A_1A_1\)</span> genotype is made by reaching out into our population and independently drawing two <span class="math inline">\(A_1\)</span> allele gametes to form a zygote. Therefore, the probability that an individual is an <span class="math inline">\(A_1A_1\)</span> homozygote is <span class="math inline">\(p^2\)</span>. This probability is also the expected frequency of the <span class="math inline">\(A_1A_1\)</span> homozygote in the population. The expected frequency of the three possible genotypes is</p>
<table>
<thead>
<tr class="header">
<th style="text-align: center;"><span class="math inline">\(f_{11}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(f_{12}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(f_{22}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\(p^2\)</span></td>
<td style="text-align: center;"><span class="math inline">\(2pq\)</span></td>
<td style="text-align: center;"><span class="math inline">\(q^2\)</span></td>
</tr>
</tbody>
</table>
<p>.</p>
<p>Note that we only need to assume random mating with respect to our alleles in order for these expected frequencies to hold, as long at <span class="math inline">\(p\)</span> is the frequency of the <span class="math inline">\(A_1\)</span> allele in the population at the time when gametes fuse. See Figure [fig:HWE_CEU_YRI] for a nice empirical demonstration of Hardy-Weinberg proportions.</p>
<p>You are investigating a locus with three alleles, A, B, and C, with allele frequencies <span class="math inline">\(p_A\)</span>, <span class="math inline">\(p_B\)</span>, and <span class="math inline">\(p_C\)</span>. What fraction of the population is expected to be homozygotes under Hardy-Weinberg?</p>
<figure>
<img src="figures/CEU_YRI_separately_HWE.png" alt="Demonstrating Hardy-Weinberg proportions using 10,000 SNPs from the HapMap CEU European and YRI African populations. Within each of these populations I plot the allele frequency against the frequency of the 3 genotypes. Each SNP is represented by 3 different coloured points. The solid lines show the mean genotype frequency (calculated using a loess smoothing). The dashed line shows the predicted genotype frequency from Hardy Weinberg equilibrium. I find it really pretty that the analytical predictions work as well as they do. See http://gcbias.org/2011/10/13/population-genetics-course-resources-hardy-weinberg-eq/ here on this plot. " /><figcaption>Demonstrating Hardy-Weinberg proportions using 10,000 SNPs from the HapMap CEU European and YRI African populations. Within each of these populations I plot the allele frequency against the frequency of the 3 genotypes. Each SNP is represented by 3 different coloured points. The solid lines show the mean genotype frequency (calculated using a loess smoothing). The dashed line shows the predicted genotype frequency from Hardy Weinberg equilibrium. I find it really pretty that the analytical predictions work as well as they do. See <a href="blog post">http://gcbias.org/2011/10/13/population-genetics-course-resources-hardy-weinberg-eq/</a> here on this plot. <span data-label="fig:HWE_CEU_YRI"></span></figcaption>
</figure>
<h2 id="coefficient-of-kinship">Coefficient of kinship</h2>
<p>We will define two alleles to be identical by descent (IBD) if they are identical due to a common ancestor in the past few generations. For the moment, we ignore mutation, and we will be more precise about what we mean by ‘past few generations’ later on. For example, parent and child share exactly one allele identical by descent at a locus, assuming that the two parents of the child are randomly mated individuals from the population. In figure [fig:IBD_cousins_cartoon] I show a pedigree demonstrating some configurations of IBD.<br />
A key quantity is the probability that our pair of individuals share 0, 1, or 2 alleles identical by descent (see Figure [fig:IBD_0_1_2]). We denote these probabilities by <span class="math inline">\(r_0\)</span>, <span class="math inline">\(r_1\)</span>, and <span class="math inline">\(r_2\)</span> respectively. See Table [table:IBDprobs] for some examples.<br />
</p>
<figure>
<img src="figures/IBD_0_1_2.png" alt="Three pairs of diploid individuals sharing 0, 1, or 2 alleles IBD where lines show the sharing of alleles by descent (e.g. from a shared ancestor). " /><figcaption>Three pairs of diploid individuals sharing 0, 1, or 2 alleles IBD where lines show the sharing of alleles by descent (e.g. from a shared ancestor). <span data-label="fig:IBD_0_1_2"></span></figcaption>
</figure>
<figure>
<img src="figures/Child_first_cousins_Homozy_BD.png" alt="Alleles being transmitted through an inbred pedigree. The mum and the aunt share two alleles identical by descent (IBD). The cousins share one allele IBD. The offspring of first cousins is homozygous by descent at this locus." /><figcaption>Alleles being transmitted through an inbred pedigree. The mum and the aunt share two alleles identical by descent (IBD). The cousins share one allele IBD. The offspring of first cousins is homozygous by descent at this locus.<span data-label="fig:IBD_cousins_cartoon"></span></figcaption>
</figure>
<p>One summary of relatedness that will be important is the probability that two alleles picked at random, one from each of the two different individuals <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>, are identical by descent. We call this quantity the coefficient of kinship of individuals <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>, and denote it by <span class="math inline">\(F_{ij}\)</span>. It is calculated as <span class="math display">\[F_{ij}= 0 \times r_0 + \frac{1}{4} r_1  + \frac{1}{2} r_2. 
\label{eqn:coeffkinship}\]</span></p>
<p>The coefficient of kinship will appear multiple times, in both our discussion of inbreeding and in the context of phenotypic resemblance between relatives.<br />
</p>
<table>
<caption>Probability that two individuals of a given relationship share 0, 1, or 2 alleles identical by descent on the autosomes. <span class="math inline">\(^{*}\)</span> assuming this is the only relationship the pair of individuals share (above that expected from randomly sampling individuals from the population). <span data-label="table:IBDprobs"></span></caption>
<thead>
<tr class="header">
<th style="text-align: left;">Relationship (i,j)<span class="math inline">\(^{*}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(r_0\)</span></th>
<th style="text-align: center;"><span class="math inline">\(r_1\)</span></th>
<th style="text-align: center;"><span class="math inline">\(r_2\)</span></th>
<th style="text-align: center;"><span class="math inline">\(F_{ij}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">parent–child</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1/4</td>
</tr>
<tr class="even">
<td style="text-align: left;">full siblings</td>
<td style="text-align: center;">1/4</td>
<td style="text-align: center;">1/2</td>
<td style="text-align: center;">1/4</td>
<td style="text-align: center;">1/4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">identical (monzygotic) twins</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">1/2</td>
</tr>
<tr class="even">
<td style="text-align: left;"><span class="math inline">\(1^{st}\)</span> cousins</td>
<td style="text-align: center;">3/4</td>
<td style="text-align: center;">1/4</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1/16</td>
</tr>
</tbody>
</table>
<p>What are <span class="math inline">\(r_0\)</span>, <span class="math inline">\(r_1\)</span>, and <span class="math inline">\(r_2\)</span> for 1/2 sibs? (1/2 sibs share one parent but not the other).</p>
<p>Consider a biallelic locus where allele 1 is at frequency <span class="math inline">\(p\)</span>, and two individuals who have IBD allele sharing probabilities <span class="math inline">\(r_0\)</span>, <span class="math inline">\(r_1\)</span>, <span class="math inline">\(r_2\)</span>.<br />
What is the overall probability that these two individuals are both homozygous for allele 1?<br />
</p>
<figure>
<img src="figures/Cousins_IBD_chromo_cartoon.png" alt="First cousins sharing a stretch of chromosome identical by descent. The different grandparental diploid chromosomes are coloured so we can track them and recombinations between them across the generations. Notice that the identity by descent between the cousins persists for a long stretch of chromosome due to the limited number of generations for recombination." /><figcaption>First cousins sharing a stretch of chromosome identical by descent. The different grandparental diploid chromosomes are coloured so we can track them and recombinations between them across the generations. Notice that the identity by descent between the cousins persists for a long stretch of chromosome due to the limited number of generations for recombination.<span data-label="fig:IBD_cousins_chr_cartoon"></span></figcaption>
</figure>
<h2 id="inbreeding">Inbreeding</h2>
<p>We can define an inbred individual as an individual whose parents are more closely related to each other than two random individuals drawn from some reference population.<br />
When two related individuals produce an offspring, that individual can receive two alleles that are identical by descent, i.e. they can be homozygous by descent (sometimes termed autozygous), due to the fact that they have two copies of an allele through different paths through the pedigree. This increased likelihood of being homozygous relative to an outbred individual is the most obvious effect of inbreeding. It is also the one that will be of most interest to us, as it underlies a lot of our ideas about inbreeding depression and population structure. In figure [fig:IBD_cousins_cartoon] our offspring of first cousins is homozygous by descent having received the same IBD allele via two different routes around an inbreeding loop.<br />
As the offspring receives a random allele from each parent (<span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>), the probability that those two alleles are identical by descent is equal to the kinship coefficient <span class="math inline">\(F_{ij}\)</span> of the two parents (Eqn. [eqn:coeffkinship]). This follows from the fact that the genotype of the offspring is made by sampling an allele at random from each of our parents. We will use IBD for identical by descent.<br />
The only way the offspring can be heterozygous (<span class="math inline">\(A_1 A_2\)</span>) is if their two alleles at a locus are not IBD (otherwise they would necessarily be homozygous). Therefore, the probability that they are heterozygous is <span class="math display">\[(1-F) 2p q,
\label{eq:hetGenHW}\]</span> where we have dropped the indices <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> for simplicity. The offspring can be homozygous for the <span class="math inline">\(A_1\)</span> allele in two different ways. They can have two non-IBD alleles that are not IBD but happen to be of the allelic type <span class="math inline">\(A_1\)</span>, or their two alleles can be IBD, such that they inherited allele <span class="math inline">\(A_1\)</span> by two different routes from the same ancestor. Thus, the probability that an offspring is homozygous is <span class="math display">\[(1-F) p^2 + F p.\]</span> Therefore, the frequencies of the three possible genotypes can be written as given in Table [table:GeneralizedHWE], which provides a generalization of the Hardy–Weinberg proportions.<br />
</p>
<table>
<caption><strong>Generalized Hardy–Weinberg</strong><span data-label="table:GeneralizedHWE"></span></caption>
<thead>
<tr class="header">
<th style="text-align: center;"><span class="math inline">\(f_{11}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(f_{12}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(f_{22}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\((1-F) p^2 + F p\)</span></td>
<td style="text-align: center;"><span class="math inline">\((1-F) 2pq\)</span></td>
<td style="text-align: center;"><span class="math inline">\((1-F) q^2 + F q\)</span></td>
</tr>
</tbody>
</table>
<p>Note that the generalized Hardy–Weinberg proportions completely specify the genotype probabilities, as there are two parameters (<span class="math inline">\(p\)</span> and <span class="math inline">\(F\)</span>) and two degrees of freedom (as <span class="math inline">\(p\)</span> and <span class="math inline">\(q\)</span> have to sum to one). Therefore, any combination of genotype frequencies at a biallelic site can be specified by a combination of <span class="math inline">\(p\)</span> and <span class="math inline">\(F\)</span>.<br />
</p>
<p>The frequency of the <span class="math inline">\(A_1\)</span> allele is p the frequency of the <span class="math inline">\(A_2\)</span> allele is q. Assume that our population is randomly mating and that the genotype frequencies in the population follow from HW. We select two individuals at random to mate from this population. We then mate the children from this cross. What is the probability that the child from this sib-mating is homozygous?</p>
<h2 id="calculating-inbreeding-coefficients-from-data">Calculating inbreeding coefficients from data</h2>
<p>If the observed heterozygosity in a population is <span class="math inline">\(H_O\)</span>, and we assume that the generalized Hardy–Weinberg proportions hold, we can set <span class="math inline">\(H_O\)</span> equal to <span class="math inline">\(f_{12}\)</span>, and solve Eq.  for <span class="math inline">\(F\)</span> to obtain an estimate of the inbreeding coefficient as <span class="math display">\[\hat{F} = 1-\frac{f_{12}}{2pq} = \frac{2pq - f_{12}}{2pq}.
\label{eqn:Fhat}\]</span> As before, <span class="math inline">\(p\)</span> is the frequency of allele <span class="math inline">\(A_{1}\)</span> in the population. This can be rewritten in terms of the observed heterozygosity (<span class="math inline">\(H_O\)</span>) and the heterozygosity expected in the absence of inbreeding, <span class="math inline">\(H_E=2pq\)</span>, as <span class="math display">\[\hat{F} = \frac{H_E-H_O}{H_E} = 1 - \frac{H_O}{H_E}.
\label{eqn:FhatHO}\]</span> Hence, <span class="math inline">\(F\)</span> quantifies the deviation due to inbreeding of the observed heterozygosity from the one expected under random mating, relative to the latter. If we have multiple loci, we can replace <span class="math inline">\(H_O\)</span> and <span class="math inline">\(H_E\)</span> by their means over loci, <span class="math inline">\(\bar{H}_O\)</span> and <span class="math inline">\(\bar{H}_E\)</span>, respectively. Note that, in principle, we could also calculate <span class="math inline">\(F\)</span> for each individual locus first, and then take the average across loci. However, this procedure is more prone to introducing a bias if sample sizes vary across loci, which is not unlikely when we are dealing with real data.<br />
</p>
<p>Suppose the following genotype frequencies were observed for at an esterase locus in a population of Drosophila (A denotes the “fast” allele and B denotes the “slow” allele):</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: center;">AA</td>
<td style="text-align: center;">AB</td>
<td style="text-align: center;">BB</td>
</tr>
<tr class="even">
<td style="text-align: center;">0.6</td>
<td style="text-align: center;">0.2</td>
<td style="text-align: center;">0.2</td>
</tr>
</tbody>
</table>
<p>.</p>
<p>What is the estimate of the inbreeding coefficient at the esterase locus?</p>
<h2 id="summarizing-population-structure">Summarizing population structure</h2>
<p>We defined inbreeding as having parents that are more closely related to each other than two individuals drawn at random from some reference population. The question that naturally arises is: Which reference population should we use? While I might not look inbred in comparison to allele frequencies in the United Kingdom (UK), where I am from, my parents certainly are not two individuals drawn at random from the world-wide population. If we estimated my inbreeding coefficient <span class="math inline">\(F\)</span> using allele frequencies within the UK, it would be close to zero, but would likely be larger if we used world-wide frequencies. This is because there is a somewhat lower level of expected heterozygosity within the UK than in the human population across the world as a whole.<br />
Wright (1943, 1951) developed a set of ‘F-statistics’ (also called ‘fixation indices’) that formalize the idea of inbreeding with respect to different levels of population structure. See figure [fig:FST_pig] for a schematic diagram. He defined <span class="math inline">\(F_{\mathrm{XY}}\)</span> as the correlation between random gametes, drawn from the same level <span class="math inline">\(X\)</span>, relative to level <span class="math inline">\(Y\)</span>. We will return to why <span class="math inline">\(F\)</span>-statistics are statements about correlations between alleles in just a moment. One commonly uses <span class="math inline">\({F_{\mathrm{IS}}}\)</span> for the inbreeding coefficient between an individual (<span class="math inline">\(I\)</span>) and the subpopulation (<span class="math inline">\(S\)</span>). Consider a single locus, where in a subpopulation (<span class="math inline">\(S\)</span>) a fraction <span class="math inline">\(H_I=f_{12}\)</span> of individuals are heterozygous. In this subpopulation, let the frequency of allele <span class="math inline">\(A_1\)</span> be <span class="math inline">\(p_S\)</span>, such that the expected heterozygosity under random mating is <span class="math inline">\(H_S = 2 p_S (1 - p_S)\)</span>. We will write <span class="math inline">\({F_{\mathrm{IS}}}\)</span> as <span class="math display">\[{F_{\mathrm{IS}}}= 1-\frac{H_I}{H_S}= 1-\frac{f_{12}}{2p_Sq_S},
\label{eqn:FIS}\]</span> a direct analog of eqn. [eqn:Fhat]. Hence, <span class="math inline">\({F_{\mathrm{IS}}}\)</span> is the relative difference between observed and expected heterozygosity due to a deviation from random mating within the subpopulation. We could also compare the observed heterozygosity in individuals (<span class="math inline">\(H_I\)</span>) to that expected in the total population, <span class="math inline">\(H_T\)</span>. If the frequency of allele <span class="math inline">\(A_1\)</span> in the total population is <span class="math inline">\(p_T\)</span>, then we can write <span class="math inline">\({F_{\mathrm{IT}}}\)</span> as <span class="math display">\[{F_{\mathrm{IT}}}=1-\frac{H_I}{H_T}= 1-\frac{f_{12}}{2p_Tq_T},
\label{eqn:FIT}\]</span> which compares heterozygosity in individuals to that expected in the total population. As a simple extension of this, we could imagine comparing the expected heterozygosity in the subpopulation (<span class="math inline">\(H_S\)</span>) to that expected in the total population <span class="math inline">\(H_T\)</span>, via <span class="math inline">\({F_{\mathrm{ST}}}\)</span>: <span class="math display">\[{F_{\mathrm{ST}}}= 1-\frac{H_S}{H_T}=1-\frac{2p_Sq_S}{2p_Tq_T} \label{eqn:FST}.\]</span> If the total population contains the subpopulation then <span class="math inline">\(2p_Sq_S \leq
2p_Tq_T\)</span>, and so <span class="math inline">\({F_{\mathrm{IS}}}\leq {F_{\mathrm{IT}}}\)</span> and <span class="math inline">\({F_{\mathrm{ST}}}\geq 0\)</span>. We can relate the three <span class="math inline">\(F\)</span>-statistics to each other as <span class="math display">\[(1-{F_{\mathrm{IT}}}) =\frac{H_I}{H_S} \frac{H_S}{H_T}=(1-{F_{\mathrm{IS}}})(1-{F_{\mathrm{ST}}}).
\label{eqn:F_relationships}\]</span> Hence, the reduction in heterozygosity within individuals compared to that expected in the total population can be decomposed to the reduction in heterozygosity of individuals compared to the subpopulation, and the reduction in heterozygosity from the total population to that in the subpopulation.<br />
</p>
<figure>
<img src="figures/FST_pig.png" alt="A diagram showing the hierarchical nature of F statistics. We can compare the heterozygosity on individuals (H_i), to that found by randomly drawing alleles from the sub-population (H_S=2p_Sq_S), to that found in the total population (H_T=2p_Tq_T). " /><figcaption>A diagram showing the hierarchical nature of F statistics. We can compare the heterozygosity on individuals (<span class="math inline">\(H_i\)</span>), to that found by randomly drawing alleles from the sub-population (<span class="math inline">\(H_S=2p_Sq_S\)</span>), to that found in the total population (<span class="math inline">\(H_T=2p_Tq_T\)</span>). <span data-label="fig:FST_pig"></span></figcaption>
</figure>
<p>If we want a summary of population structure across multiple subpopulations, we can average <span class="math inline">\(H_I\)</span> and/or <span class="math inline">\(H_S\)</span> across populations, and use a <span class="math inline">\(p_T\)</span> calculated by averaging <span class="math inline">\(p_S\)</span> across subpopulations (or our samples from sub-populations). For example, the average <span class="math inline">\({F_{\mathrm{ST}}}\)</span> across <span class="math inline">\(K\)</span> subpopulations (sampled with equal effort) is <span class="math display">\[{F_{\mathrm{ST}}}= 1 - \frac{\bar{H}_{S}}{H_T},\]</span> where <span class="math inline">\(\bar{H}_S = 1/K \sum_{i = 1}^{K} H_{S}^{(i)}\)</span>, and <span class="math inline">\(H_{S}^{(i)} = 2 p_{i} q_{i}\)</span> is the expected heterozygosity in subpopulation <span class="math inline">\(i\)</span>. Furthermore, if we have multiple sites, we can replace <span class="math inline">\(H_I\)</span>, <span class="math inline">\(H_S\)</span>, and <span class="math inline">\(H_T\)</span> with their averages across loci (as above).<br />
Let us now return to Wright’s definition of the <span class="math inline">\(F\)</span>-statistics as correlations between random gametes, drawn from the same level <span class="math inline">\(X\)</span>, relative to level <span class="math inline">\(Y\)</span>. Without loss of generality, we may think about <span class="math inline">\(X\)</span> as individuals and <span class="math inline">\(S\)</span> as the subpopulation. Rewriting <span class="math inline">\({F_{\mathrm{IS}}}\)</span> in terms of the observed homozygote frequencies (<span class="math inline">\(f_{11}\)</span>, <span class="math inline">\(f_{22}\)</span>) and expected homozygosities (<span class="math inline">\(p_{S}^2\)</span>, <span class="math inline">\(q_{S}^2\)</span>) we find <span class="math display">\[{F_{\mathrm{IS}}}= \frac{2p_Sq_S - f_{12}}{2p_Sq_S} = \frac{f_{11}+f_{22} -
p_S^2 - q_S^2}{2p_Sq_S},
\label{eqn:Fascorr}\]</span> using the fact that <span class="math inline">\(p^2+2pq+q^2=1\)</span>, and <span class="math inline">\(f_{12} = 1 - f_{11} - f_{12}\)</span>. The form of eqn. ([eqn:Fascorr]) reveals that <span class="math inline">\({F_{\mathrm{IS}}}\)</span> is the covariance between pairs of alleles found in an individual, divided by the expected variance under binomial sampling. Thus, <span class="math inline">\(F\)</span>-statistics can be understood as the correlation between alleles drawn from a population (or an individual) above that expected by chance (i.e. drawing alleles sampled at random from some broader population).<br />
We can also interpret <span class="math inline">\(F\)</span>-statistics as proportions of variance explained by different levels of population structure. To see this, let us think about <span class="math inline">\({F_{\mathrm{ST}}}\)</span> averaged over <span class="math inline">\(K\)</span> subpopulations, whose frequencies are <span class="math inline">\(p_1,\dots,p_K\)</span>. The frequency in the total population is <span class="math inline">\(p_T=\bar{p} = 1/K \sum_{i=1}^K p_i\)</span>. Then, we can write <span class="math display">\[{F_{\mathrm{ST}}}= \frac{2 \bar{p}\bar{q} - \frac{1}{K}\sum_{i=1}^K 2p_iq_i }{2
\bar{p}\bar{q}} = \frac{ \left(\frac{1}{K} \sum_{i=1}^K p_i^2 +
\frac{1}{K} \sum_{i=1}^K q_i^2 \right) -  \bar{p}^2-\bar{q}^2 }{2
\bar{p}\bar{q}} = \frac{\mathrm{Var}(p_i)}{\mathrm{Var}(\bar{p})},
\label{eqn:F_as_propvar}\]</span> which shows that <span class="math inline">\({F_{\mathrm{ST}}}\)</span> is the proportion of the variance explained by the subpopulation labels.</p>
<h2 id="other-approaches-to-population-structure">Other approaches to population structure</h2>
<p>There is a broad spectrum of methods to describe patterns of population structure in populaion genetic datasets. We’ll briefly discuss two broad-classes of methods, assigment methods and principal components analysis,that appear often in the literature.</p>
<h3 id="assignment-methods">Assignment Methods</h3>
<p>Here we’ll describe a simple probabilistic assignment to find the probability that an individual of unknown population comes from one of <span class="math inline">\(K\)</span> predefined populations. We’ll then briefly explain how to extend this to cluster individuals into <span class="math inline">\(K\)</span> initially unknown populations. This method is a simplified version of what Bayesian population genetics clustering algorithms such as STRUCTURE and ADMIXTURE do (Pritchard et al. Genetics 2000).</p>
<h4 id="a-simple-assignment-method">A simple assignment method</h4>
<p>We have genotype data from unlinked S bi-allelic loci for <span class="math inline">\(K\)</span> populations. The allele frequency of allele <span class="math inline">\(A_1\)</span> at locus <span class="math inline">\(l\)</span> in population <span class="math inline">\(k\)</span> is denoted by <span class="math inline">\(p_{k,l}\)</span>, so that the allele frequencies in population 1 are <span class="math inline">\(p_{1,1},\cdots p_{1,L}\)</span> and population 2 are <span class="math inline">\(p_{2,1},\cdots p_{2,L}\)</span> and so on.</p>
<p>You type a new individual from an unknown population at these <span class="math inline">\(L\)</span> loci. This individual’s genotype at locus <span class="math inline">\(l\)</span> is <span class="math inline">\(g_l\)</span>, where <span class="math inline">\(g_l\)</span> denotes the number of copies of allele <span class="math inline">\(A_1\)</span> this individual carries at this locus (<span class="math inline">\(g_l=0,1,2\)</span>).</p>
<p>The probability of this individual’s genotype at locus <span class="math inline">\(l\)</span> conditional on coming from population <span class="math inline">\(k\)</span> (i.e. their alleles being a random HW draw from population <span class="math inline">\(k\)</span>) is <span class="math display">\[P(g_l | \textrm{pop k}) = I(g_l=0) (1-p_{k,l})^2 +  I(g_l=1) 2 p_{k,l} (1-p_{k,l}) + I(g_l=2) p_{k,l}^2\]</span> where <span class="math inline">\(I(g_l=0)\)</span> is an indicator function which is <span class="math inline">\(1\)</span> if <span class="math inline">\(g_l=0\)</span> and zero otherwise, and likewise for the other indicator functions. This follows simply from HWE.</p>
<p>Assuming that the loci are independent, the probability of individual’s genotypes conditional on them coming from population <span class="math inline">\(k\)</span> is <span class="math display">\[P(\textrm{ind.} | \textrm{pop k})  = \prod_{l=1}^S P(g_l | \textrm{pop k}) \label{eqn_assignment}\]</span></p>
<p>We wish to know the probability that this new individual comes from population <span class="math inline">\(k\)</span>, i.e. <span class="math inline">\(P(\textrm{pop k} | \textrm{new ind.})\)</span>. We can obtain this through Bayes rule <span class="math display">\[P(\textrm{pop k} | \textrm{ind.})  = \frac{P(\textrm{ind.} | \textrm{pop k}) P(\textrm{pop k})}{P(\textrm{ind.})}\]</span> where <span class="math display">\[P(\textrm{ind.}) = \sum_{k=1}^K  P(\textrm{ind.} | \textrm{pop k}) P(\textrm{pop k})\]</span> is the normalizing constant. We interpret <span class="math inline">\(P(\textrm{pop k})\)</span> as the prior probability of the individual coming from population <span class="math inline">\(k\)</span>, unless we have some other prior knowledge we will assume that the new individual has a equal probability of coming from each population <span class="math inline">\(P(\textrm{pop k})=1/K\)</span>.</p>
<p>We intepret <span class="math display">\[P(\textrm{pop k} | \textrm{ind.})\]</span> as the posterior probability that our new individual comes from each of our <span class="math inline">\(1,\cdots, K\)</span> populations.</p>
<p>More sophisticated versions of this are now used to allow for hybrids, e.g, we can have a proportion <span class="math inline">\(q_k\)</span> of our individual’s genome come from population <span class="math inline">\(k\)</span> and estimate the set of <span class="math inline">\(q_k\)</span>’s.</p>
<p>We have two populations where the frequency of capital allele at two SNPs (<span class="math inline">\(A/a\)</span> and <span class="math inline">\(B/b\)</span>) is given by</p>
<table>
<thead>
<tr class="header">
<th style="text-align: center;">Population</th>
<th style="text-align: center;">locus A</th>
<th style="text-align: center;">locus B</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">1</td>
<td style="text-align: center;"><span class="math inline">\(0.1\)</span></td>
<td style="text-align: center;"><span class="math inline">\(0.85\)</span></td>
</tr>
<tr class="even">
<td style="text-align: center;">2</td>
<td style="text-align: center;"><span class="math inline">\(0.95\)</span></td>
<td style="text-align: center;"><span class="math inline">\(0.2\)</span></td>
</tr>
</tbody>
</table>
<p>We sample an individual whose genotype is <span class="math inline">\(AA\)</span> at the first locus and <span class="math inline">\(bb\)</span> at the second. What is the posterior probability that our indvidual comes from population 1 vs population 2? Lets assume that with probability <span class="math inline">\(q_1\)</span> our individual draws an allele from population <span class="math inline">\(1\)</span> and that with probability <span class="math inline">\(q_2=1-q_1\)</span> they draw an allele from population <span class="math inline">\(2\)</span>. What is the probability of our individual’s genotype given <span class="math inline">\(q_1\)</span>? You could plot this probability as a function of <span class="math inline">\(q_1\)</span>. How does your plot change if our individual is heterozygote at both loci?</p>
<h4 id="clustering-based-on-assignment-methods">Clustering based on assignment methods</h4>
<p>While it is great to be able to assign our individuals to particular population, these ideas can be pushed to learn about how best to describe our genotype data in terms of discrete populations without assigning any of our individuals to populations <span><em>a priori</em></span>. We wish to cluster our individuals into <span class="math inline">\(K\)</span> unknown populations. We begin by assigning our individuals at random to these <span class="math inline">\(K\)</span> populations.</p>
<ul>
<li><p>Given these assignments we estimate the allele frequencies at all of our loci in each population.</p></li>
<li><p>Given these allele frequencies we chose to reassign each individual to a population <span class="math inline">\(k\)</span> with a probability given by eqn. (<span class="math inline">\(\ref{eqn_assignment}\)</span>).</p></li>
</ul>
<p>We iterate steps 1 and 2 for many iterations. If the data is sufficiently informative the assignments and allele frequencies will quickly converge.</p>
<p>To do this in a full bayesian scheme we need to place priors on the allele frequencies (e.g. a beta distribution).Technically we are using this is the joint posterior of our allele frequencies and assignments.</p>
<h3 id="principal-components-analysis">Principal components analysis</h3>
<p>The use of principal component analysis in population genetics was pioneered by Cavalli-Sforza. With large genotyping datasets PCA has made a come back. See Patterson et al 2006, PLoS Genetics and McVean, G. 2010 PLoS Genetics and for recent discussion.</p>
<p>Consider a dataset consisting of N individuals at S bi-allelic SNPs. The <span class="math inline">\(i^{th}\)</span> individual’s genotype data at locus <span class="math inline">\(\ell\)</span> takes value <span class="math inline">\(g_{i,\ell}\)</span>=0,1, or 2 (corresponding to the number of copies of allele <span class="math inline">\(A_1\)</span> an individual carrys at this SNP). We can think of this as a N x S matrix (where usually <span class="math inline">\(N \ll S\)</span>).</p>
<p>Denoting the sample mean allele freq at SNP <span class="math inline">\(\ell\)</span> by <span class="math inline">\(p_{\ell}\)</span> we usually standardize the genotype in the following way <span class="math display">\[\frac{g_{i,\ell} - 2 p_{\ell}}{\sqrt{p_{\ell}(1-p_{\ell})}}\]</span> i.e. at each SNP we center the genotypes by minusing of the mean genotype (<span class="math inline">\(2\epsilon_{\ell}\)</span>) and divide through by the expected variance assuming that alleles are sampled binomially from the mean frequency (<span class="math inline">\(\sqrt{p_{\ell} (1-p_{\ell})}\)</span>). Doing this to all of our genotypes we form a data matrix (of dimension N x S). We can then perform principal components analysis of this data matrix to cover the major axes of genotype variance in our sample.</p>
<p>It is worth taking a moment to delve further into what we are doing here. There’s a number of equivalent ways to thinking about what PCA is doing, one of these is to think that when we do PCA we are building the individual by individual covariance matrix and performing eigen-value decomposition of this matrix (with the eigen-vectors giving the PC). This individual by individual covariance matrix has entries the <span class="math inline">\((i,~j)^{th}\)</span> entry given by <span class="math display">\[\sum_{\ell=1}^S \frac{(g_{i,\ell} - 2p_{\ell})(g_{j,\ell} - 2p_{\ell})}{p_{\ell}(1-p_{\ell})}\]</span> note that this is the covariance, is very similar to those we encountered in discussing <span class="math inline">\(F\)</span>-statistics as correlations (equation ), expect now we are asking about the allelic covariance between two individuals above that expected if they were both drawn from the total sample at random (rather than the covariance of alleles within a single individual). So by performing PCA on the data we are learning about the major (orthogonal) axes of the kinship matrix.</p>
<h2 id="correlations-between-loci-linkage-disequilibrium-and-recombination">Correlations between loci, linkage disequilibrium, and recombination</h2>
<p>Up to now we have been interested in correlations between alleles at the same locus, e.g. correlations within individuals (inbreeding) or between individuals (relatedness). We have seen how relatedness between parents affects the extent to which their offspring is inbred. We now turn to correlations between alleles at different loci. To understand correlations between loci we need to understand recombination.<br />
</p>
<h4 id="recombination">Recombination</h4>
<p>Lets consider an individual heterozygous for a <span class="math inline">\(AB\)</span> and <span class="math inline">\(ab\)</span> haplotype. If no recombination occurs between our two loci in this individual, then these two haplotypes will be transmitted intact to the next generation. While if a recombination (or more generally an odd number of recombinations) occurs between our two loci on the haplotype transmitted to the child then <span class="math inline">\(\tfrac{1}{2}\)</span> the time the child receives a <span class="math inline">\(Ab\)</span> haplotype and <span class="math inline">\(\tfrac{1}{2}\)</span> the time the child receives a <span class="math inline">\(aB\)</span> haplotype. So recombination is breaking up the association between loci. We’ll define the recombination fraction (<span class="math inline">\(r\)</span>) to be the probability of an odd number of recombinations between our loci. In practice we’ll often be interested in relatively short regions where recombination is relatively rare, and so we might think that <span class="math inline">\(r=r_{BP}L \ll 1\)</span>, where <span class="math inline">\(r_{BP}\)</span> is the average recombination rate per base pair (typically <span class="math inline">\(\sim 10^{-8}\)</span>) and L is the number of base pairs separating our two loci.<br />
</p>
<h4 id="linkage-disequilibrium">Linkage disequilibrium</h4>
<p>The (horrible) phrase linkage disequilibrium (LD) refers to the statistical non-independence (i.e. a correlation) of alleles at different loci. Our two loci, which segregate alleles <span class="math inline">\(A/a\)</span> and <span class="math inline">\(B/b\)</span>, have allele frequencies of <span class="math inline">\(p_A\)</span> and <span class="math inline">\(p_B\)</span> respectively. The frequency of the two locus haplotype is <span class="math inline">\(p_{AB}\)</span>, and likewise for our other three combinations. If our loci were statistically independent then <span class="math inline">\(p_{AB} = p_Ap_B\)</span>, otherwise <span class="math inline">\(p_{AB} \neq p_Ap_B\)</span> We can define a covariance between the <span class="math inline">\(A\)</span> and <span class="math inline">\(B\)</span> alleles at our two loci as <span class="math display">\[D_{AB} = p_{AB} - p_Ap_B\]</span> and likewise for our other combinations at our two loci (<span class="math inline">\(D_{Ab},~D_{aB},~D_{ab}\)</span>). These <span class="math inline">\(D\)</span> statistics are all closely related to each other as <span class="math inline">\(D_{AB} = - D_{Ab}\)</span> and so on. Thus we only need to specify one <span class="math inline">\(D_{AB}\)</span> to know them all, so we’ll drop the subscript and just refer to <span class="math inline">\(D\)</span>. Also a handy result is that we can rewrite our haplotype frequency <span class="math inline">\(p_{AB}\)</span> as <span class="math display">\[p_{AB} = p_Ap_B+D. \label{eqn:ABviaD}\]</span> If <span class="math inline">\(D=0\)</span> we’ll say the two loci are in linkage equilibrium, while if <span class="math inline">\(D&gt;0\)</span> or <span class="math inline">\(D&lt;0\)</span> we’ll say that the loci are in linkage disequilibrium (we’ll perhaps want to test whether <span class="math inline">\(D\)</span> is statistically different from <span class="math inline">\(0\)</span> before making this choice). You should be careful to keep the concepts of linkage and linkage disequilibrium separate in your mind. Genetic linkage refers to the linkage of multiple loci due to the fact that they are transmitted through meiosis together (most often because the loci are on the same chromosome). Linkage disequilibrium merely refers to the correlation between the alleles at different loci, this may in part be due to the genetic linkage of these loci but does not necessarily imply this (e.g. genetically unlinked loci can be in LD due to population structure).<br />
Another common statistic for summarizing LD is <span class="math inline">\(r^2\)</span> which we write as <span class="math display">\[r^2 = \frac{D^2}{p_A(1-p_A) p_B(1-p_B) }\]</span> as <span class="math inline">\(D\)</span> is a covariance, and <span class="math inline">\(p_A(1-p_A) \)</span> is the variance of an allele drawn at random from locus <span class="math inline">\(A\)</span>, <span class="math inline">\(r^2\)</span> is the squared correlation coefficient.<br />
</p>
<p>You genotype 2 bi-allelic loci (A &amp; B) segregating in two mouse subspecies (1 &amp; 2) which mate randomly among themselves, but have not historically interbreed since they speciated. On the basis of previous work you estimate that the two loci are separated by a recombination fraction of 0.1. The frequencies of haplotypes in each population are:</p>
<table>
<thead>
<tr class="header">
<th style="text-align: center;">Pop</th>
<th style="text-align: center;"><span class="math inline">\(p_{AB}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(p_{Ab}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(p_{aB}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(p_{ab}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">1</td>
<td style="text-align: center;">.02</td>
<td style="text-align: center;">.18</td>
<td style="text-align: center;">.08</td>
<td style="text-align: center;">.72</td>
</tr>
<tr class="even">
<td style="text-align: center;">2</td>
<td style="text-align: center;">.72</td>
<td style="text-align: center;">.18</td>
<td style="text-align: center;">.08</td>
<td style="text-align: center;">.02</td>
</tr>
</tbody>
</table>
<p><span><strong>A)</strong></span> How much LD is there within populations, i.e. estimate D?<br />
<span><strong>B)</strong></span> If we mixed the two populations together in equal proportions what value would D take before any mating has had the chance to occur?<br />
</p>
<h4 id="the-decay-of-ld-due-to-recombination">The decay of LD due to recombination</h4>
<p>We will now examine what happens to LD over the generations if we only allow recombination to occur in a very large population (i.e. no genetic drift, i.e. the frequencies of our loci follow their expectations). To do so consider the frequency of our <span class="math inline">\(AB\)</span> haplotype in the next generation <span class="math inline">\(p_{AB}^{\prime}\)</span>. We lose a fraction <span class="math inline">\(r\)</span> of our <span class="math inline">\(AB\)</span> haplotypes to recombination ripping our alleles apart but gain a fraction <span class="math inline">\(rp_A p_B\)</span> per generation from other haplotypes recombining together to form <span class="math inline">\(AB\)</span> haplotypes. Thus in the next generation <span class="math display">\[p_{AB}^{\prime} = (1-r)p_{AB} + rp_Ap_B\]</span> this last term here is <span class="math inline">\(r(p_{AB}+p_{Ab})(p_{AB}+p_{aB})\)</span>, which multiplying this out is the probability of recombination in the different diploid genotypes that could generate a <span class="math inline">\(p_{AB}\)</span> haplotype.<br />
We can then write the change in the frequency of the <span class="math inline">\(p_{AB}\)</span> haplotype as <span class="math display">\[\Delta p_{AB} = p_{AB}^{\prime} -p_{AB} = -r p_{AB} + rp_Ap_B = - r D\]</span> so recombination will cause a decrease in the frequency of <span class="math inline">\(p_{AB}\)</span> if there is an excess of <span class="math inline">\(AB\)</span> haplotypes within the population (<span class="math inline">\(D&gt;0\)</span>), and an increase if there is a deficit of <span class="math inline">\(AB\)</span> haplotypes within the population (<span class="math inline">\(D&lt;0\)</span>). Our LD in the next generation is <span class="math inline">\(D^{\prime} =
p_{AB}^{\prime}\)</span>, so we can rewrite the above eqn. in terms of the <span class="math inline">\(D^{\prime} \)</span> <span class="math display">\[D^{\prime}= (1-r) D\]</span> so if the level of LD in generation <span class="math inline">\(0\)</span> is <span class="math inline">\(D_0\)</span> the level <span class="math inline">\(t\)</span> generations later (<span class="math inline">\(D_t\)</span>) is <span class="math display">\[D_t=  (1-r)^t D_0\]</span> so recombination is acting to decrease LD, and it does so geometrically at a rate given by <span class="math inline">\((1-r)\)</span>. If <span class="math inline">\(r \ll 1\)</span> then we can approximate this by an exponential and say that <span class="math display">\[D_t \approx  D_0 e^{-rt}\]</span><br />
</p>
<p>You find a hybrid population between the two mouse subspecies described in the question above, which appears to be comprised of equal proportions of ancestry from the two subspecies. You estimate LD between the two markers to be 0.0723. Assuming that this hybrid population is large and was formed by a single mixture event, can you estimate how long ago this population formed?<br />
</p>
<h1 id="genetic-drift-and-neutral-diversity">Genetic Drift and Neutral Diversity</h1>
<p>Various sources of randomness are inherent in evolution. One major source of stochasticity in population genetics is genetic drift. Genetic drift occurs because more or less copies of an allele by chance can be transmitted to the next generation. This can occur because by chance the individuals carrying a particular allele can leave more or less offspring in the next generation. In a sexual population genetic drift also occurs because mendelian transmission means that only one of the two alleles in an individual, chosen at random at a locus, is transmitted to the offspring.</p>
<p>Genetic drift can play a role in the dynamics of all alleles and populations, but it will play the biggest role for neutral alleles. A neutral polymorphism occurs when the segregating alleles at a polymorphic site have no discernable effect on the fitness (we’ll make clear what we mean by discernable later, for the moment think of this as “no effect” on fitness).</p>
<h2 id="LossofHet">Loss of heterozygosity due to drift.</h2>
<p>Genetic drift will, in the absence of new mutations, slowly purge our population of neutral genetic diversity as alleles slowly drift to high or low frequencies and are lost or fixed over time.<br />
Imagine a population of a constant size <span class="math inline">\(N\)</span> diploid individuals, and that we are examining a locus segregating for two alleles that are neutral with respect to each other. This population is randomly mating with respect to the alleles at this locus. See figures [fig:LossHet_two_alleles] and [fig:LossHet_many_alleles] to see how genetic drift proceeds, by tracking alleles within a small population.<br />
In generation <span class="math inline">\(t\)</span> our current level of heterozygosity is <span class="math inline">\(H_t\)</span>, i.e. the probability that two randomly sampled alleles in generation <span class="math inline">\(t\)</span> are non-identical is <span class="math inline">\(H_t\)</span>. Assuming that the mutation rate is zero (or vanishing small), what is our level of heterozygosity in generation <span class="math inline">\(t+1\)</span>?<br />
</p>
<figure>
<img src="figures/Loss_of_he_col_two_alleles.png" alt="Loss of heterozygosity over time, in the absence of new mutations. A diploid population of 5 individuals over the generations, with lines showing transmission. In the first generation every individual is a heterozygote.." /><figcaption>Loss of heterozygosity over time, in the absence of new mutations. A diploid population of 5 individuals over the generations, with lines showing transmission. In the first generation every individual is a heterozygote..<span data-label="fig:LossHet_two_alleles"></span></figcaption>
</figure>
<figure>
<img src="figures/Loss_of_het_2_many_alleles.png" alt="Loss of heterozygosity over time, in the absence of new mutations. A diploid population of 5 individuals. In the first generation I colour every allele a different colour so we can track their descendants." /><figcaption>Loss of heterozygosity over time, in the absence of new mutations. A diploid population of 5 individuals. In the first generation I colour every allele a different colour so we can track their descendants.<span data-label="fig:LossHet_many_alleles"></span></figcaption>
</figure>
<p>In the next generation (<span class="math inline">\(t+1\)</span>) we are looking at the alleles in the offspring of generation <span class="math inline">\(t\)</span>. If we randomly sample two alleles in generation <span class="math inline">\(t+1\)</span> which had different parental alleles in generation <span class="math inline">\(t\)</span> then it is just like drawing two random alleles from generation <span class="math inline">\(t\)</span>. So the probability that these two alleles in generation <span class="math inline">\(t+1\)</span>, that have different parental alleles in generation <span class="math inline">\(t\)</span>, are non-identical is <span class="math inline">\(H_t\)</span>.<br />
Conversely, if our pair of alleles have the same parental allele in the proceeding generation (i.e. the alleles are identical by descent one generation back) then these two alleles must be identical (as we are not allowing for any mutation).<br />
In a diploid population of size <span class="math inline">\(N\)</span> individuals there are <span class="math inline">\(2N\)</span> alleles. The probability that our two alleles have the same parental allele in the proceeding generation is <span class="math inline">\(1/(2N)\)</span>, the probability that they have different parental alleles is is <span class="math inline">\(1-1/(2N)\)</span>. So by the above argument the expected heterozygosity in generation <span class="math inline">\(t+1\)</span> is <span class="math display">\[H_{t+1} = \frac{1}{2N} \times 0 + \left(1-\frac{1}{2N} \right)H_t\]</span> By this argument if the heterozygosity in generation <span class="math inline">\(0\)</span> is <span class="math inline">\(H_0\)</span> our expected heterozygosity in generation <span class="math inline">\(t\)</span> is <span class="math display">\[H_t = \left(1-\frac{1}{2N} \right)^tH_0\]</span> i.e. the expected heterozygosity with our population is decaying geometrically with each passing generation. If we assume that <span class="math inline">\(1/(2N)
\ll 1\)</span> then we can approximate this geometric decay by an exponential decay, such that <span class="math display">\[H_t =H_0 \exp \left(-\frac{t}{2N} \right)\]</span> i.e. heterozygosity decays exponentially at a rate <span class="math inline">\(1/(2N)\)</span>.</p>
<p>You are in charge of maintaining a population of delta smelt in the Sacramento river delta. Using a large set of microsatellites you estimate that the mean level of heterozygosity in this population is 0.005. You set yourself a goal of maintaining a level of heterozygosity of at least 0.0049 for the next two hundred years. Assuming that the smelt have a generation time of 3 years, and that only genetic drift affects these loci. What is the smallest fully outbreeding population that you would need to maintain to meet this goal? (Note this is similar to the calculations being carried out by conservation biologists but my numbers are made up.).</p>
<h2 id="DriftMutationBalance">Levels of diversity maintained by a balance between mutation and drift</h2>
<p>Looking backwards in time from one generation to the next, we are going to say that two alleles which have the same parental allele (i.e. find their common ancestor) in the preceding generation have <span><em>c</em>oalesced</span>, and refer to this event as a <span><em>c</em>oalescent event</span>.</p>
<p>The probability that our pair of randomly sampled alleles have coalesced in the preceding generation is <span class="math inline">\(1/(2N)\)</span>, the probability that our pair of alleles fail to coalesce is <span class="math inline">\(1-1/(2N)\)</span>.</p>
<p>The probability that a mutation changes the identity of the transmitted allele is <span class="math inline">\(\mu\)</span> per generation. So the probability of no mutation occurring is <span class="math inline">\((1-\mu)\)</span>. We’ll assume that when a mutation occurs it creates some new allelic type which is not present in the population. This assumption (commonly called the infinitely-many-alleles model) makes the math slightly cleaner, and also is not too bad an assumption biologically. See figure [fig:Mut_Sel_balance] for a depiction of mutation-drift balance in this model over the generations.</p>
<p>This model lets us calculate when our two alleles last shared a common ancestor and whether these alleles are identical as a result of failing to mutate since this shared ancestor. For example we can work out the probability that our two randomly sampled alleles coalesced <span class="math inline">\(2\)</span> generations in the past (i.e. they fail to coalesce in generation <span class="math inline">\(1\)</span> and then coalescent in generation <span class="math inline">\(2\)</span>), and that they are identical as <span class="math display">\[\left(1- \frac{1}{2N} \right) \frac{1}{2N} (1-\mu)^4\]</span> note the power of <span class="math inline">\(4\)</span> is because our two alleles have to have failed to mutate through <span class="math inline">\(2\)</span> meioses each.</p>
<p>More generally the probability that our alleles coalesce in generation <span class="math inline">\(t+1\)</span> and are identical due to no mutation to either allele in the subsequent generations is <span class="math display">\[P(\textrm{coal. in t+1 \&amp; no mutations}) =  \frac{1}{2N} \left(1- \frac{1}{2N} \right)^t \left(1-\mu \right)^{2(t+1)}\]</span> assuming that <span class="math inline">\(\frac{1}{2N} \ll\)</span> and <span class="math inline">\(\mu \ll 1\)</span> then we can approximate this as <span class="math display">\[P(\textrm{coal. in t+1 \&amp; no mutations}) \approx  \frac{1}{2N}
e^{-t/(2N)} e^{-2\mu (t+1)}\]</span> to make this slightly easier on ourselves lets further assume that <span class="math inline">\(t
\approx t+1\)</span> and so rewrite this as <span class="math display">\[\approx \frac{1}{2N} e^{-t(2\mu+1/(2N))}\]</span></p>
<figure>
<img src="figures/Mut_drift_balance.png" alt="Mutation-drift balance. A diploid population of 5 individuals. In the first generation everyone has the same allele. Each generation the transmitted allele can mutate and we generate a new colour. In the bottom plot I trace the frequency of alleles in our population over time." /><figcaption>Mutation-drift balance. A diploid population of 5 individuals. In the first generation everyone has the same allele. Each generation the transmitted allele can mutate and we generate a new colour. In the bottom plot I trace the frequency of alleles in our population over time.<span data-label="fig:Mut_Sel_balance"></span></figcaption>
</figure>
<p>If we sample two alleles at random from the population we will not in general know when they share a common ancestor. In which case we will need to integrate out over when this coalescent event occurred. Doing this we find the probability that our two alleles are identical due to no mutation on either ancestral lineage since the pair shared a common ancestor to be <span class="math display">\[\frac{1}{2N} \int_0^{\infty} e^{-t(2\mu+1/(2N))} dt =
\frac{1/(2N)}{1/(2N)+2\mu} = \frac{1}{1+4N\mu}\]</span> The probability that our pair of alleles are non-identical is simply one minus this, i.e. <span class="math display">\[\frac{4N\mu}{1+4N\mu} \label{eqn:hetero}\]</span> This compound parameter <span class="math inline">\(4N\mu\)</span>, the population-scaled mutation rate, will come up a number of times so we’ll give it its own name <span class="math display">\[\theta = 4N\mu\]</span></p>
<p>So all else being equal, species with larger population sizes should have proportionally higher levels of neutral polymorphism.</p>
<h2 id="the-effective-population-size.">The effective population size.</h2>
<p>In practice populations rarely conform to our assumptions of being constant in size with low variance in reproduction success. Real populations experience dramatic fluctuations in size, and there is often high variance in reproductive success. Thus rates of drift in natural populations are often a lot higher than the census population size would imply.<br />
To cope with this population geneticists often invoke the concept of an effective population size (<span class="math inline">\(N_e\)</span>). In many situations (but not all), departures from model assumptions can be captured by substituting <span class="math inline">\(N_e\)</span> for <span class="math inline">\(N\)</span>.</p>
<p>Specifically the effective population size (<span class="math inline">\(N_e\)</span>) is the population size that would result in the same rate of drift in an idealized constant population size, obeying our modeling assumptions, as that observed in our true population.<br />
If population sizes vary rapidly in size, we can (if certain conditions are met) replace our population size by the harmonic mean population size. Consider a diploid population of variable size, whose size is <span class="math inline">\(N_t\)</span> <span class="math inline">\(t\)</span> generations into the past. The probability our pairs of alleles have not coalesced by the generation <span class="math inline">\(t^{th}\)</span> is given by <span class="math display">\[\prod_{i=1}^{t} \left(1-\frac{1}{2N_t} \right)\]</span> note that this is simply collapses to our original expression <span class="math inline">\(\left(1-\frac{1}{2N } \right)^t \)</span> if <span class="math inline">\(N_i\)</span> is constant. If <span class="math inline">\(1/(N_i)\)</span> is small, then we can approximate <span class="math inline">\(1-\frac{1}{2N_i}\)</span> by <span class="math inline">\(\exp(-\frac{1}{2N_i})\)</span>. Such that if <span class="math inline">\(N_i\)</span> is never too small <span class="math display">\[\prod_{i=1}^{t} \left(1-\frac{1}{2N_i} \right)
\approx \prod_{i=1}^{t} \exp \left( -\frac{1}{2N_i} \right)   =
\exp \left(- \sum_{i=1}^{t} \frac{1}{2N_i} \right) .\]</span> In our constant population size case the probability of failing to coalesce is <span class="math inline">\(\exp(-t/(2N))\)</span>. So the variable population coalescent probabilities are still of the same form but the exponent has changed. Comparing the exponent in the two cases we see <span class="math display">\[\frac{t}{2N} = \sum_{i=1}^{t} \frac{1}{2N_i}\]</span> so that if we want a constant effective population size (<span class="math inline">\(N_e\)</span>) that has the same coalescent probability as our variable population we need to set <span class="math inline">\(N=N_e\)</span> and rearrange this to see <span class="math display">\[N_e =\frac{1}{\frac{1}{t} \sum_{i=1}^{t} \frac{1}{N_i} }.\]</span> this is the harmonic mean of the varying population size. Thus our effective population size, the size of an idealized constant population which matches the rate of genetic drift, is the harmonic mean true population size over time. The harmonic mean is very strongly affected by small values, such that if our population size is one million <span class="math inline">\(99\%\)</span> of the time but drops to a <span class="math inline">\(1000\)</span> every hundred or so generations, <span class="math inline">\(N_e\)</span> will be much closer to <span class="math inline">\(1000\)</span> than a million. See figure [fig:LossHet_varying_pop] for a depiction of a repeatedly bottlenecked population losing diversity at a fast rate.<br />
</p>
<figure>
<img src="figures/Loss_of_he_col_alleles_varying_pop_dark.png" alt="Loss of heterozygosity over time in a bottlenecking population. A diploid population of 10 individuals, that bottlenecks down to three individuals repeatedly. In the first generation I colour every allele a different colour so we can track their descendants, there are no new mutations." /><figcaption>Loss of heterozygosity over time in a bottlenecking population. A diploid population of 10 individuals, that bottlenecks down to three individuals repeatedly. In the first generation I colour every allele a different colour so we can track their descendants, there are no new mutations.<span data-label="fig:LossHet_varying_pop"></span></figcaption>
</figure>
<p>Variance in reproductive success will also affect our effective population size. Even if our population has a large constant size of <span class="math inline">\(N\)</span> individuals, if only small proportion of them get to reproduce then the rate of drift will reflect this much small number of reproducing individuals. See figure [fig:LossHet_varying_RS] for a depiction of the higher rate of drift in a population where there is high variance in reproductive success.</p>
<p>If only <span class="math inline">\(N_M\)</span> males get to contribute to the next generation and <span class="math inline">\(N_F\)</span> females get to contribute to the next generation. When our two alleles pick an ancestor, <span class="math inline">\(25\%\)</span> of the time our alleles were both in a female ancestor in which case they coalesce with probability <span class="math inline">\(1/(2N_F)\)</span>, and <span class="math inline">\(25\%\)</span> of the time they are both in a male ancestor in which case they coalesce with probability <span class="math inline">\(1/(2N_M)\)</span>. The remaining <span class="math inline">\(50\%\)</span> of the time our ancestral lineages are in two individuals are different sexes in a generation so cannot coalescence. Therefore, our probability of coalescence in the preceding generation is <span class="math display">\[\frac{1}{4}\frac{1}{2N_M}+\frac{1}{4}\frac{1}{2N_F} =
\frac{1}{8}\frac{N_F+N_M}{N_FN_M}\]</span> i.e. the rate of coalescence is the harmonic mean of the two sexes population sizes, equating this to <span class="math inline">\(\frac{1}{2N_e}\)</span> we find <span class="math display">\[N_e = \frac{4N_FN_M}{N_F+N_M}\]</span> Thus if reproductive success is very skewed in one sex (e.g. <span class="math inline">\(N_M \ll
N/2\)</span>) our effective population size will be much reduced as a result.<br />
</p>
<figure>
<img src="figures/Loss_of_he_col_alleles_varying_RS.png" alt="High variance on reproductive success increases the rate of genetic drift. A diploid population of 10 individuals, where the circled individuals have much higher reproductive success. In the first generation I colour every allele a different colour so we can track their descendants, there are no new mutations." /><figcaption>High variance on reproductive success increases the rate of genetic drift. A diploid population of 10 individuals, where the circled individuals have much higher reproductive success. In the first generation I colour every allele a different colour so we can track their descendants, there are no new mutations.<span data-label="fig:LossHet_varying_RS"></span></figcaption>
</figure>
<h2 id="the-coalescent-and-patterns-of-neutral-diversity">The Coalescent and patterns of neutral diversity</h2>
<h4 id="pairwise-coalescent-time-distribution-and-the-number-of-pairwise-differences.">Pairwise Coalescent time distribution and the number of pairwise differences.</h4>
<p>Thinking back to our calculations we made about the loss of neutral heterozygosity and equilibrium levels of diversity (in Sections [LossofHet] and [DriftMutationBalance]), you’ll note that we could first specify what generation a pair of sequences coalesce in, and then calculate some properties of heterozygosity based on that. That’s because neutral mutations do not affect the probability that an individual transmits that allele, so don’t affect the way in which we can trace ancestral lineages back.<br />
As such it will often be helpful to consider the time to the common ancestor of a pair of sequences, and then think of the impact of that on patterns of diversity. See Figure [fig:Coalescent_simulation] for an example of this. The probability that a pair of alleles have failed to coalesce in <span class="math inline">\(t\)</span> generations and then coalesce in the <span class="math inline">\(t+1\)</span> generation back is <span class="math display">\[\frac{1}{2N} (1- \frac{1}{2N} )^{t} \approx \frac{1}{2N} e^{-t/(2N)} \label{eqn:coal_time_dist}\]</span> thus the coalescent time of a pair of sequences (<span class="math inline">\(T_2\)</span>) is approximately exponentially distributed with a rate <span class="math inline">\(1/(2N)\)</span>. We’ll denote that by saying that <span class="math inline">\(T_2 \sim Exp \left( 1/(2N) \right)\)</span>. The mean coalescent time of a pair of a pair of alleles is <span class="math inline">\(2N\)</span> generations<br />
</p>
<figure>
<img src="figures/Coalescent.png" alt="A simple simulation of the coalescent process. The simulation consists of a diploid population of 10 individuals (20 alleles). In each generation, each individual is equally likely to be the parent of an offspring (and the allele transmitted is indicated by a light grey line). We track a pair of alleles, chosen in the present day, back 14 generations untill they find a common ancestor." /><figcaption>A simple simulation of the coalescent process. The simulation consists of a diploid population of 10 individuals (20 alleles). In each generation, each individual is equally likely to be the parent of an offspring (and the allele transmitted is indicated by a light grey line). We track a pair of alleles, chosen in the present day, back 14 generations untill they find a common ancestor.<span data-label="fig:Coalescent_simulation"></span></figcaption>
</figure>
<p>Conditional on a pair of alleles coalescing <span class="math inline">\(t\)</span> generations ago there are <span class="math inline">\(2t\)</span> generations in which a mutation could occur. Thus the probability of our pair of alleles are separated by <span class="math inline">\(j\)</span> mutations since they last shared a common ancestor is <span class="math display">\[P(j | T_2 = t ) = {2t \choose j} \mu^{j} (1-\mu)^{2t-j}\]</span> i.e. mutations happen in <span class="math inline">\(j\)</span> generations, and do not happen in <span class="math inline">\(2t-j\)</span> generations (with <span class="math inline">\({2t \choose j}\)</span> ways this can possibly happen). Assuming that <span class="math inline">\(\mu \ll 1\)</span>, and that <span class="math inline">\(2t-j \approx 2t\)</span> then we can approximate the probability that we have <span class="math inline">\(j\)</span> mutations as a Poisson distribution <span class="math display">\[P(j | T_2 = t ) = \frac{ (2 \mu t )^{j} e^{-2\mu t}}{j!}\]</span> i.e. a Poisson with mean <span class="math inline">\(2\mu t \)</span>.<br />
As our expected coalescent time is <span class="math inline">\(2N\)</span> generations, the expected number of mutations separating two alleles drawn at random from the population is <span class="math display">\[{\mathbb{E}}(j) = 4N\mu = \theta\]</span> We’ll assume that mutations never happen at the same site twice, i.e. no multiple hits, such that we get to see all of the mutation events that separate our pair of sequences (we’ll call this the infinitely-many-sites assumption, which should be fine if <span class="math inline">\(N\mu_{BP} \ll 1\)</span>). Thus the number of mutations between a pair of sites is the observed number of differences between a pair of sequences.<br />
We’ll denote the observed number of pairwise differences at putatively neutral sites separating a pair of sequences as <span class="math inline">\(\pi\)</span> (we usually average this over a number of pairs of sequences for a region). So we can estimate of <span class="math inline">\(\theta\)</span> from <span class="math inline">\(\pi\)</span>, <span class="math inline">\(\widehat{\theta}_{\pi}\)</span> by setting <span class="math inline">\(\widehat{\theta}_{\pi}=\pi\)</span>. If we have an independent estimate of <span class="math inline">\(\mu\)</span>, then from setting <span class="math inline">\(\pi = \widehat{\theta}_{\pi} = 4N\mu\)</span> we can obtain an estimate of the population size <span class="math inline">\(N\)</span> that is consistent with our levels of neutral polymorphism.</p>
<h2 id="the-coalescent-process-of-a-sample-of-alleles.">The coalescent process of a sample of alleles.</h2>
<p>Usually we are not just interested pairs of alleles, or the average pairwise diversity, we are interested in the properties of diversity in samples of a number of alleles drawn from the population. To allow for this instead of just following a pair of lineages back until they coalesce, we can follow the history of a sample of alleles back through the population.</p>
<p>Consider first sampling three alleles at random from the population. The probability that all three alleles choose exactly the same ancestral allele one generation back is <span class="math inline">\(1/(2N)^2\)</span>. If <span class="math inline">\(N\)</span> is reasonably large then this is a very small probability. As such it is very unlikely that our three alleles coalesce at once, a in a moment we’ll see that it is safe to ingnore such unlikely events.<br />
The probability that a specific pair of alleles find a common ancestor in the preceding generation is still <span class="math inline">\(1/(2N)\)</span>. There are three possible pairs of alleles so the probability that no pair finds a common ancestor is <span class="math display">\[\left(1-\frac{1}{(2N)} \right)^3 \approx \left( 1- \frac{3}{2N} \right)\]</span> in making this approximation we are multiplying out the right hand-side and ignoring terms of <span class="math inline">\(1/N^2\)</span> and higher. See Figure [fig:Coalescent_simulation_3] for a random realization of this process.<br />
</p>
<figure>
<img src="figures/Coalescent_3.png" alt="A simple simulation of the coalescent process for three lineages. We track the ancestry of three modern-day alleles, the first pair (blue and purple) coalesce four generations back their are then two independent lineages we are tracking, this pair then coalesces twelve generations in the past. Note that different random realizations of this process will differ from each other a lot." /><figcaption>A simple simulation of the coalescent process for three lineages. We track the ancestry of three modern-day alleles, the first pair (blue and purple) coalesce four generations back their are then two independent lineages we are tracking, this pair then coalesces twelve generations in the past. Note that different random realizations of this process will differ from each other a lot.<span data-label="fig:Coalescent_simulation_3"></span></figcaption>
</figure>
<p>More generally when we sample <span class="math inline">\(i\)</span> alleles there are <span class="math inline">\({i \choose
 2}\)</span> pairs, i.e. <span class="math inline">\(i(i-1)/2\)</span> pairs, thus the probability that no pair of alleles coalesces in the preceding generation is <span class="math display">\[\left(1-\frac{1}{(2N)} \right)^{{i \choose
 2}} \approx \left( 1- \frac{{i \choose
 2}}{2N}\right)\]</span> while the probability of any pair coalescing is <span class="math inline">\(\approx \frac{{i \choose
 2}}{2N}\)</span>.<br />
We can ignore the possibility of more than pairs of alleles (e.g. tripletons) simultaneously coalescing at once as terms of <span class="math inline">\(1/N^2\)</span> and higher can be ignored as they are vanishingly rare. Obviously there are in reasonable sample sizes there are many more triples (<span class="math inline">\({i \choose 3}\)</span>), and higher order combinations, than pairs (<span class="math inline">\({i \choose 2}\)</span>) but if <span class="math inline">\(i \ll N\)</span> then we are safe to ignore these terms.</p>
<p>When there are <span class="math inline">\(i\)</span> alleles the probability that we wait until the <span class="math inline">\(t+1\)</span> generation before any pair of alleles coalesce is <span class="math display">\[\frac{{i \choose
 2}}{2N}\left( 1- \frac{{i \choose
 2}}{2N}\right)^{t} \approx  \frac{{i \choose
 2}}{2N} \exp \left( - \frac{{i \choose
 2}}{2N} t \right)\]</span> thus the waiting time <span class="math inline">\(T_i\)</span> to the first coalescent event in a sample of <span class="math inline">\(i\)</span> alleles is exponentially distributed with rate <span class="math inline">\(\frac{{i \choose
 2}}{2N}\)</span>, i.e. <span class="math inline">\(T_i \sim Exp\left(\frac{{i \choose
 2}}{2N} \right)\)</span>. The mean waiting time till any of pair within our sample to coalesce is <span class="math inline">\(2N/{i \choose
 2}\)</span>.<br />
When a pair of alleles first find a common ancestral allele some number of generations back further into the past we only have to keep track of that common ancestral allele for the pair. Thus when a pair of alleles in our sample of <span class="math inline">\(i\)</span> alleles coalesce, we then switch to having to follow <span class="math inline">\(i-1\)</span> alleles back. Then when a pair of these <span class="math inline">\(i-1\)</span> alleles coalesce, we then have to follow <span class="math inline">\(i-2\)</span> alleles back. This process continues until we coalesce back to a sample of two, and from there to a single most recent common ancestor (MRCA).<br />
To simulate a coalescent genealogy at a locus for a sample of <span class="math inline">\(n\)</span> alleles we therefore simply follow this algorithm</p>
<ol>
<li><p>set <span class="math inline">\(i=n\)</span>.</p></li>
<li><p>We simulate a random variable to be the time <span class="math inline">\(t_i\)</span> to the next coalescent event from <span class="math inline">\(t_i \sim
 Exp\left(\frac{{i \choose
 2}}{2N} \right)\)</span></p></li>
<li><p>choose a pair of alleles to coalesce at random from all possible pairs.</p></li>
<li><p>set <span class="math inline">\(i=i-1\)</span></p></li>
<li><p>continue looping of steps 1-3 until <span class="math inline">\(i=1\)</span> i.e. the most recent common ancestor of the sample is found.</p></li>
</ol>
<p>by following this algorithm we are generating realizations of the genealogy of our sample.<br />
We will first consider the time to the most recent common ancestor of the entire sample (<span class="math inline">\(T_{MRCA}\)</span>). This is <span class="math display">\[T_{MRCA} = \sum_{i=n}^2 T_i\]</span> generations back. As our coalescent times for different <span class="math inline">\(i\)</span> are independent, the expected time to the most recent common ancestor is <span class="math display">\[{\mathbb{E}}(T_{MRCA}) = \sum_{i=n}^2 {\mathbb{E}}(T_i) = \sum_{i=n}^2  2N/{i \choose
 2}\]</span> using the fact that <span class="math inline">\(\frac{1}{i(i-1)}=\frac{1}{i-1} - \frac{1}{i}\)</span> with a bit of rearrangement we can rewrite this is <span class="math display">\[{\mathbb{E}}(T_{MRCA}) = 4N\left(1- \frac{1}{n} \right)\]</span> so the average <span class="math inline">\(T_{MRCA}\)</span> scales linearly with population size. Interestingly, as we move to larger and larger samples (i.e. <span class="math inline">\(n \gg 1\)</span>) the average time to the most recent common ancestor is converging on <span class="math inline">\(4N\)</span>. What’s happening here is that in large samples our lineages typically coalesce rapidly at the start and very soon coalesce down to a much smaller number of lineages.<br />
Above we argued that a mutation is only becomes a fixed difference if it is lucky enough to be the ancestor of the entire population. As we saw above this occurs with probability <span class="math inline">\(1/(2N)\)</span>. How long does is take on average for such an allele to fix within our population. We’ve just seen that it takes <span class="math inline">\(4N\)</span> generations for a large sample of alleles to all trace their ancestry back to a single most recent common ancestor. Thus it must take roughly <span class="math inline">\(4N\)</span> generations for a neutral allele present in a single copy within the the population to the ancestor of all alleles within our population. This argument can be made more precise, but in general we would still find that it takes <span class="math inline">\(\approx 4N\)</span> generations for a neutral allele to go from its introduction to fixation with the population.<br />
The total amount of time in the genealogy (<span class="math inline">\(T_{tot}\)</span>) <span class="math display">\[T_{tot} = \sum_{i=n}^2 iT_i\]</span> as when there are <span class="math inline">\(i\)</span> lineages each contributes a time <span class="math inline">\(T_i\)</span> to the total time. Taking the expectation of the total time in the genealogy <span class="math display">\[{\mathbb{E}}(T_{tot}) = \sum_{i=n}^2 i \frac{2N}{{i \choose
 2} } = \sum_{i=n}^2 \frac{4N}{i -1} =\sum_{i=n-1}^1 \frac{4N}{i}\]</span> so our expected total amount of time in the genealogy scales linearly with our population size. Our expected total amount of time is also increasing with sample size but is doing so very slowly. To see this more carefully we can see that for large <span class="math inline">\(n\)</span> <span class="math display">\[{\mathbb{E}}(T_{tot}) = \sum_{i=n-1}^1 \frac{4N}{i} \approx 4N \int_1^n \frac{1}{i} di
= 4N \log(n-1)\]</span> here we are approximating our sum by an integral, which will work for large <span class="math inline">\(n\)</span>. So our expected total amount of time in the genealogy is growing with <span class="math inline">\(n\)</span> but it is doing so very slowly. This again follows from the fact that in large samples the initial coalescence usually happens very rapidly, so that extra samples adds little to the total amount of time in the tree.<br />
We saw above that the number of mutational differences between a pair of alleles that coalescence <span class="math inline">\(T_2\)</span> generations ago was Poisson with a mean of <span class="math inline">\(2 \mu T_2\)</span>. A mutation that occurs on any branch of our genealogy will cause a segregating polymorphism in the sample (making our infinitely-many-sites assumption). Thus if the total time in the genealogy is <span class="math inline">\(T_{tot}\)</span> there is <span class="math inline">\(T_{tot}\)</span> generations for mutations. So the total number of mutations segregating in our sample (<span class="math inline">\(S\)</span>) is Poisson with mean <span class="math inline">\(\mu T_{tot}\)</span>. Thus the expected number of segregting in history a sample of size <span class="math inline">\(n\)</span> is <span class="math display">\[{\mathbb{E}}(S) = \mu {\mathbb{E}}(T_{tot}) = \sum_{i=n-1}^1 \frac{4N\mu }{i} = \theta
\sum_{i=n-1}^1 \frac{1}{i}\]</span> Thus we can use this formula to derive another estimate of the population scaled mutation rate, by setting our observed number of segregating sites in a sample (<span class="math inline">\(S\)</span>) equal to this expectation. We’ll call this estimator <span class="math inline">\(\widehat{\theta}_W\)</span> <span class="math display">\[\widehat{\theta}_W =\frac{ S}{\sum_{i=n-1}^1 \frac{1}{i}}\]</span> this estimator was devised by Watterson, hence the <span class="math inline">\(W\)</span>.</p>
<h2 id="the-fixation-of-neutral-alleles">The fixation of neutral alleles</h2>
<p>It is very unlikely that a rare neutral allele accidentally drifts up to fixation, it is much more likely that such an allele is eventually lost from the population. However, there is a large and constant influx of rare alleles into the population due to mutation, so even if it is very unlikely that an individual allele fixes within the population, some neutral alleles will fix.<br />
</p>
<h4 id="probability-of-the-eventual-fixation-of-a-neutral-allele.">Probability of the eventual fixation of a neutral allele.</h4>
<p>An allele which reaches fixation within a population, is an ancestor to the entire population. In a particular generation there can be only single allele that all other alleles at the locus in later generation can claim as an ancestor. As at a neutral locus all of our alleles are exchangeable, as they have no effect on the number of descendents an individual leaves, so any allele is equally likely to be the ancestor of the entire population. In a diploid population size of size <span class="math inline">\(N\)</span>, there are <span class="math inline">\(2N\)</span> alleles all of which are equally likely to be the ancestor of the entire population at some later time point. So if our allele is present in a single copy, the chance that is the ancestor to the entire population in some future generation is <span class="math inline">\(1/(2N)\)</span>, i.e. the chance our neutral allele is eventually fixed is <span class="math inline">\(1/(2N)\)</span>. See Figure [fig:subs_simulation], our orange allele in the first generation is one of 10 differently coloured alleles, and so has a <span class="math inline">\(1/10\)</span> chance of being the ancestor of the entire population at some later time point (as it is by the 9th generation).<br />
More generally if our neutral allele is present in <span class="math inline">\(i\)</span> copies in the population, of <span class="math inline">\(2N\)</span> alleles, the probability that this allele is fixed is <span class="math inline">\(i/(2N)\)</span>. I.e. the probability that a neutral allele is eventually fixed is simply given by its frequency (<span class="math inline">\(p\)</span>) in the population. We can also derive this result by letting <span class="math inline">\(Ns \rightarrow
0\)</span> in eqn. .</p>
<figure>
<img src="figures/Substitution_sim.png" alt="Each allele initially present in a small diploid population is given a different colour so we can track their descendants over time. By the 9th generation all of the alleles present in the population can trace their ancestry back to the orange allele." /><figcaption>Each allele initially present in a small diploid population is given a different colour so we can track their descendants over time. By the 9th generation all of the alleles present in the population can trace their ancestry back to the orange allele.<span data-label="fig:subs_simulation"></span></figcaption>
</figure>
<h4 id="rate-of-substitution-of-neutral-alleles.">Rate of substitution of neutral alleles.</h4>
<p>A substitution between populations that do not exchange gene flow is simply a fixation event within one population. The rate of substitution is therefore the rate at which new alleles fix in the population, so that the long-term substitution rate is the rate at which mutations arise that will eventually become fixed within our population.<br />
Assume that there are two classes of mutational changes that can occur with a region, highly deleterious mutations and neutral mutations. A fraction <span class="math inline">\(C\)</span> of all mutational changes are highly deleterious, and can not possibly contribute to substitution nor polymorphism (i.e. <span class="math inline">\(Ns \gg 1\)</span>). While a fraction <span class="math inline">\(1-C\)</span> are neutral. If our mutation rate is <span class="math inline">\(\mu\)</span> per transmitted allele per generation, then a total of <span class="math inline">\(2N \mu (1-C)\)</span> neutral mutations enter our population each generation.<br />
Each of these neutral mutations has a <span class="math inline">\(1/(2N)\)</span> probability chance of eventually becoming fixed in the population. Therefore, the rate at which neutral mutations arise that eventually become fixed within our population is <span class="math display">\[2N\mu(1-C)\frac{1}{2N} = \mu(1-C)\]</span> thus the rate of substitution under a model where newly arising alleles are either highly deleterious or neutral, is simply given by the mutation rate towards neutral alleles, i.e. <span class="math inline">\(\mu(1-C)\)</span>.<br />
Consider a pair of species have diverged for <span class="math inline">\(T\)</span> generations, i.e. orthologous sequences shared between the species last shared a common ancestor <span class="math inline">\(T\)</span> generations ago. If they have maintained a constant <span class="math inline">\(\mu\)</span> over that time, will have accumulated an average of <span class="math display">\[2\mu(1-C)T\]</span> neutral substitutions. This assumes that <span class="math inline">\(T\)</span> is a lot longer than the time it takes to fix a neutral allele, such that the total number of alleles introduced into the population that will eventually fix is the total number of substitutions. We’ll see below that a neutral allele takes on average <span class="math inline">\(4N\)</span> generations to fix from its introduction into the population.<br />
This is a really pretty result as the population size has completely canceled out of the neutral substitution rate. However, there is another way to see this in a more straightward way. If I look at a sequence in me compared to say a particular chimp, I’m looking at the mutations that have occurred in both of our germlines since they parted ways <span class="math inline">\(T\)</span> generations ago. Since neutral alleles do not alter the probability of their transmission to the next generation, we are simply looking at the mutations that have occurred in <span class="math inline">\(2T\)</span> generations worth of transmissions. Thus the average number of neutral mutational differences separating our pair of species is simply <span class="math inline">\(2\mu (1-C) T\)</span>.<br />
</p>
<h2 id="neutral-diversity-and-population-structure">Neutral diversity and population structure</h2>
<p>Upto now we have assumed that our alleles that we have modelled in the coalescent setting are drawn from a randomly mating population such that any pair of lineages is equally likely to coalesce with each other. However, when there is population structure this assumption is violated.<br />
We have previously written the measure of population structure <span class="math inline">\({F_{\mathrm{ST}}}\)</span> as <span class="math display">\[{F_{\mathrm{ST}}}= \frac{H_T-H_S}{H_T}\]</span> where <span class="math inline">\(H_S\)</span> is the probability that two alleles sampled at random from a subpopulation differ, and <span class="math inline">\(H_T\)</span> is the probability that two alleles sampled at random from the total population differ.</p>
<h4 id="a-simple-population-split-model">A simple population split model</h4>
<p>Imagine a population of constant size of <span class="math inline">\(N_e\)</span> diploid individuals that <span class="math inline">\(\tau\)</span> generations in the past split into two daughter populations (sub-populations) each of size <span class="math inline">\(N_e\)</span> individuals, who do not subsequently exchange migrants. In the current day we sample an equal number of alleles from both subpopulations.</p>
<p>Consider a pair of alleles sampled within one of our sub-populations, they have experienced a population of size <span class="math inline">\(N_e\)</span> and so the probability that they differ is <span class="math inline">\(H_S = \theta/(1+\theta)\)</span> (where <span class="math inline">\(\theta=4N_e\mu\)</span>). The heterozygosity in our total population is a little more tricky to calculate. Assuming that we equally sample both sub-populations, when we draw two alleles from our total sample, <span class="math inline">\(50\%\)</span> of the time they are drawn from the same subpopulation and <span class="math inline">\(50\%\)</span> of the time they are drawn from different subpopulations. Therefore, our total heterozygosity is given by <span class="math display">\[H_T = {\tfrac{1}{2}}H_S + {\tfrac{1}{2}}H_B\]</span> where <span class="math inline">\(H_B\)</span> is the probability that a pair of alleles drawn from our two different sub-populations differ from each other. Our pair of alleles can not find a common ancestor with each other for at least <span class="math inline">\(\tau\)</span> generations into he past as they are in distinct populations (not connected by migration). The probability that one or other of them mutates in this time is <span class="math inline">\(1-(1-\mu)^{2T}\)</span>. With probability <span class="math inline">\((1-\mu)^{2T} \)</span> neither of our alleles mutate in the <span class="math inline">\(T\)</span> generations back in time before they find themselves back in the combined ancestral population. Conditional on failing to mutating before the combined ancestral population, the probability that they do manage to mutate before coalescing in that population of size <span class="math inline">\(N_e\)</span> is <span class="math inline">\(\theta/(\theta+1)\)</span>. Putting these components together <span class="math display">\[H_B = \left( 1-(1-\mu)^{2T} \right) + (1-\mu)^{2T}
  \frac{\theta}{\theta+1}\]</span> We can plug this into our expression for <span class="math inline">\(H_T\)</span>, and then that in turn into <span class="math inline">\({F_{\mathrm{ST}}}\)</span>.</p>
<figure>
<img src="figures/drift_split.png" alt="Change in allele frequencies following a population split." /><figcaption>Change in allele frequencies following a population split.<span data-label="fig:drift_split"></span></figcaption>
</figure>
<p>To understand this better we can make a simple approximation based on our mutation rate being very low, such that <span class="math inline">\(N_e \mu \ll 1\)</span> ao <span class="math inline">\(H_S \approx
4N_e\mu\)</span>, and that <span class="math inline">\(\mu \ll 1\)</span> and <span class="math inline">\(\mu T \ll 1\)</span>. Assuming this, then <span class="math display">\[H_B \approx 2 \mu T + 4N_e\mu.\]</span> So that <span class="math display">\[{F_{\mathrm{ST}}}\approx \frac{ \mu T}{\mu T +  4N_e\mu }  \]</span> note that <span class="math inline">\(\mu\)</span> cancels out of this. In this simple toy model <span class="math inline">\({F_{\mathrm{ST}}}\)</span> is increasing because the amount of between population diversity increases with the divergence time of the two populations (initially linearly with <span class="math inline">\(T\)</span>). It does so at a rate give by <span class="math inline">\(T/(4N_e)\)</span> so that differentiation will be higher between populations separated by long divergence times or with small effective population sizes.</p>
<h4 id="a-simple-model-of-migration-between-an-island-and-the-mainland.">A simple model of migration between an island and the mainland.</h4>
<p>We can also use the coalescent to think about patterns of differentiation under a simple model of migration drift equilibrium. Lets consider a small island population that is relatively isolated from a large mainland population, and that both of these populations are constant in size. We’ll assume that the expected heterozygosity for a pair of alleles sampled on the mainland is <span class="math inline">\(H_M\)</span>.</p>
<p>Our island has a population size <span class="math inline">\(N_{I}\)</span> that is very small compared to our mainland population. Each generation some low fraction <span class="math inline">\(m\)</span> of our individuals on the island have migrant parents from the mainland the generation before. Our island may also send migrants back to the mainland, but these are a drop in the ocean compared to the large population size on the mainland and their effect can be ignored.</p>
<p>If we sample an allele on the island back and trace its ancestral lineage backward in time, each generation our ancestral allele have a low probability <span class="math inline">\(m\)</span> of being descended from the mainland in the proceeding generation (if we go far enough the allele eventually has to be descended from an allele on the mainland). The probability that a pair of alleles sampled on the island are descended from a shared recent common ancestral allele on the island, is the probability that our pair of alleles coalesce before either lineage migrates. For example, the probability that our pair of alleles coalesce <span class="math inline">\(t+1\)</span> generations back is <span class="math display">\[\frac{1}{2N_I}(1-m)^{2(t+1)} \left(1-\frac{1}{2N_I} \right)^{t} \approx
\frac{1}{2N_I} \exp\left( -t\left (\frac{1}{2N_I} + 2m\right) \right),\]</span> with the approximation following from assuming that <span class="math inline">\(m \ll 1\)</span> &amp; <span class="math inline">\(1/(2N_I)
\ll 1\)</span> (note that this is very similar to our derivation of heterozygosity above). The probability that our alleles coalescence before either one of them migrates off the island, irrespective of the time, is <span class="math display">\[\int_0^{\infty} \frac{1}{2N_I} \exp\left( -t\left (\frac{1}{2N_I} +
    2m\right) \right) dt = \frac{1/(2N_I) }{1/(2N_I) +
    2m}.\]</span></p>
<p>Lets assume that the mutation rate is very low such as it is very unlikely that the pair of alleles mutate before they coalesce on the island. Therefore, the only way that the alleles can be different from each other is if one or other of them migrates to the mainland, which happens with probability <span class="math display">\[1 - \frac{1/(2N_I) }{1/(2N_I) + 2m}\]</span> Conditional on one or other of our alleles migrating to the mainland, both of our alleles represent independent draws from the mainland and so differ from each other with probability <span class="math inline">\(H_M\)</span>. Therefore, the level of heterozygosity on the island is given by <span class="math display">\[H_I = (1 - \frac{1/(2N_I) }{1/(2N_I) + 2m})H_M\]</span> So the reduction of heterozygosity on the island compared to the mainland is <span class="math display">\[F_{IM} = 1- \frac{H_I}{H_M} = \frac{ 1/(2N_I) }{1/(2N_I) + 2m} = \frac{ 1 }{1 + 4N_Im}.\]</span> The level of inbreeding on the island compared to the mainland will be high in the migration rate is low and the effective population size of the island is low, as allele frequencies on the island are drifting and diversity is not being replenished on the island by migration. The key parameter here is the number individuals on the island replaced by immigrants from the mainland each generation (<span class="math inline">\(N_I m\)</span>).</p>
<p>We have framed this as being about the reduction in genetic diversity on the island compared to the mainland. However, if we consider collecting a individuals on the island and mainland in proportion to population sizes the total level of heterozygosity would be <span class="math inline">\(H_T=H_M\)</span>, as samples from our mainland would greatly outnumber those from our island. Therefore, considering our island our sub-population we have derived another simple model of <span class="math inline">\(F_{ST}\)</span> .</p>
<h1 id="the-phenotypic-resemblance-between-relatives">The phenotypic resemblance between relatives</h1>
<p>We can use our understanding of the sharing of alleles between relatives to understand the phenotypic resemblance between relatives in quantitative phenotypes. We can then use this to understand the evolutionary change in quantitative phenotypes in response to selection.<br />
Let’s imagine that the genetic component of the variation in our trait is controlled by <span class="math inline">\(L\)</span> autosomal loci that act in an additive manner. The frequency of allele <span class="math inline">\(1\)</span> at locus <span class="math inline">\(l\)</span> is <span class="math inline">\(p_l\)</span>, with each copy of allele <span class="math inline">\(1\)</span> at this locus increasing your trait value by <span class="math inline">\(a_l\)</span> above the population mean. The phenotype of an individual, let’s call her <span class="math inline">\(i\)</span>, is <span class="math inline">\(X_i\)</span>. Her genotype at SNP <span class="math inline">\(l\)</span>, is <span class="math inline">\(G_{i,l}\)</span>. Here <span class="math inline">\(G_{i,l}=0,~1,\)</span> or <span class="math inline">\(2\)</span> represents the number of copies of allele <span class="math inline">\(1\)</span> she has at this SNP. Her expected phenotype, given her genotype, is then <span class="math display">\[X_{A,i} = \mu + {\mathbb{E}}(X_i | G_{i,1},\cdots,G_{i,L}) = \mu+ \sum_{l=1}^L \label{pheno_geno}
G_{i,l} a_{l}\]</span> where <span class="math inline">\(\mu\)</span> is the mean phenotype in our population. Now in reality the genetic phenotype is a function of the expression of those alleles in a particular environment. Therefore, we can think of this expected phenotype as being an average across a set of environments that occur in the population.<br />
When we measure our individual’s phenotype we see <span class="math display">\[X_i =   \mu+X_{A,i} + X_{E,i} \label{pheno_geno_environ}\]</span> where <span class="math inline">\(X_E\)</span> is the deviation from the mean phenotype due to the environment. This <span class="math inline">\(X_E\)</span> included the systematic effects of the environment our individual finds herself in and all of the noise during development, growth, and the various random insults that life throws at our individual. If a reasonable number of loci contribute to variation in our trait then we can approximate the distribution of <span class="math inline">\(X_{A,i}\)</span> by a normal distribution due to the central limit theorem (see Figure [fig:QT1]). Thus if we can approximate the distribution of the effect of environmental variation on our trait (<span class="math inline">\(X_{E,i}\)</span>) also by a normal distribution, which is reasonable as there are many small environmental effects, then the distribution of phenotypes within the population (<span class="math inline">\(X_i\)</span>) will be normally distributed (see Figure [fig:QT1]).<br />
</p>
<figure>
<img src="figures/QT1.png" alt="The convergence of the phenotypic distribution to a normal distribution. Each of the three histograms shows the distribution of the phenotype in a large sample, for increasing large numbers of loci (L). I have simulated each individual’s phenotype following equation [pheno_geno] and [pheno_geno_environ]. Specifically I simulate each individuals biallelic genotype at L loci, assuming Hardy-Weinberg proportions and that the allele is at 50% frequency. I assume that all of the alleles have equal effects and combine them additively together. I then add an environmental contribution, which is normally distributed with variance 0.05. Note that in the left two pictures you can see peaks corresponding to different genotypes." /><figcaption>The convergence of the phenotypic distribution to a normal distribution. Each of the three histograms shows the distribution of the phenotype in a large sample, for increasing large numbers of loci (<span class="math inline">\(L\)</span>). I have simulated each individual’s phenotype following equation [pheno_geno] and [pheno_geno_environ]. Specifically I simulate each individuals biallelic genotype at <span class="math inline">\(L\)</span> loci, assuming Hardy-Weinberg proportions and that the allele is at 50% frequency. I assume that all of the alleles have equal effects and combine them additively together. I then add an environmental contribution, which is normally distributed with variance <span class="math inline">\(0.05\)</span>. Note that in the left two pictures you can see peaks corresponding to different genotypes.<span data-label="fig:QT1"></span></figcaption>
</figure>
<p>Note that as this is an additive model we can decompose eqn. [pheno_geno_environ] into the effects of the two alleles at each locus, in particular we can rewrite it as <span class="math display">\[X_i = \mu + X_{iM}+X_{iP} +X_{iE}\]</span> where <span class="math inline">\(X_{iM}\)</span> and <span class="math inline">\(X_{iP}\)</span> are the contribution to the phenotype of the allele that our individual received from her mother (maternal alleles) and father (paternal alleles) respectively. This will come in handy in just a moment when we start thinking about the phenotype covariance of relatives.<br />
Now obviously this model seems silly at first sight as alleles don’t only act in an additive manner, as they interact with alleles at the same loci (dominance) and at different loci (epistasis). Later we’ll relax this assumption, however, we’ll find that if we are interested in evolutionary change over short time-scales it is actually only the “additive component” of genetic variation that will (usually) concern us. We will define this more formally later on, but for the moment we can offer the intuition that parents only get to pass on a single allele at each locus on to the next generation. As such, it is the effect of these transmitted alleles, averaged over possible matings, that is an individual’s average contribution to the next generation (i.e. the additive effect of the alleles that their genotype consists of).</p>
<h3 id="additive-genetic-variance-and-heritability">Additive genetic variance and heritability</h3>
<p>As we are talking about an additive genetic model we’ll talk about the additive genetic variance (<span class="math inline">\(V_A\)</span>), the variance due to the additive effects of segregating genetic variation. This is a subset of the total genetic variance if we allow for non-additive effects.<br />
The variance of our phenotype across individuals (<span class="math inline">\(V\)</span>) can write this as <span class="math display">\[V = Var(X_A) + Var(X_E) = V_A+V_E\]</span> in writing this we are assuming that there is no covariance between <span class="math inline">\(X_{G,i}\)</span> and <span class="math inline">\(X_{E,i}\)</span> i.e. there is no covariance between genotype and environment.<br />
Our additive genetic variance can be written as <span class="math display">\[V_A = \sum_{l=1}^L Var(G_{i,l} a_{l})\]</span> where <span class="math inline">\(Var(G_{i,l} a_{l})\)</span> is the contribution to the additive variance among individuals of the <span class="math inline">\(l\)</span> locus. Assuming random mating we can write our additive genetic variance as <span class="math display">\[V_A = \sum_{l=1}^L a_{l}^2 2 p_l(1-p_l)\]</span> where the <span class="math inline">\( 2 p_l(1-p_l)\)</span> term follows the binomial sampling of two alleles per individual at each locus.<br />
</p>
<h4 id="the-narrow-sense-heritability">The narrow sense heritability</h4>
<p>We would like a way to think about what proportion of the variation in our phenotype across individuals is due to genetic differences as opposed to environmental differences. Such a quantity will be key in helping us think about the evolution of phenotypes. For example, if variation in our phenotype had no genetic basis then no matter how much selection changes the mean phenotype within a generation the trait will not change over generations.<br />
We’ll call the proportion of the variance that is genetic the heritability, and denote it by <span class="math inline">\(h^2\)</span>. We can then write this as <span class="math display">\[h^2 = \frac{Var(X_A)}{V} = \frac{V_A}{V}\]</span> remember that we thinking about a trait where all of the alleles act in a perfectly additive manner. In this case our heritability <span class="math inline">\(h^2\)</span> is referred to as the narrow sense heritability, the proportion of the variance explained by the additive effect of our loci. When we allow dominance and epistasis into our model we’ll also have to define the broad sense heritability (the total proportion of the phenotypic variance attributable to genetic variation).<br />
The narrow sense heritability of a trait is a useful quantity, indeed we’ll see shortly that it is exactly what we need to understand the evolutionary response to selection on a quantitative phenotype. We can calculate the narrow sense heritability by using the resemblance between relatives. For example, if our phenotype was totally environmental we should not expect relatives to resemble each other any more than random individuals drawn from the population. Now the obvious caveat here is that relatives also share an environment, so may resemble each other due to shared environmental effects.<br />
</p>
<h3 id="the-covariance-between-relatives">The covariance between relatives</h3>
<p>So we’ll go ahead and calculate the covariance in phenotype between two individuals (<span class="math inline">\(1\)</span> and <span class="math inline">\(2\)</span>) who have a phenotype <span class="math inline">\(X_1\)</span> and <span class="math inline">\(X_2\)</span> respectively. <span class="math display">\[Cov(X_1,X_2) =
Cov\left((X_{1M}+X_{1P}+X_{1E}),((X_{2M}+X_{2P}+X_{2E}) \right)\]</span> We can expand this out in terms of the covariance between the various components in these sums.<br />
To make our task easier we (and most analyses) will assume two things</p>
<ol>
<li><p>that we can ignore the covariance of the environments between individuals (i.e. <span class="math inline">\(Cov(X_{1E},X_{2E})=0\)</span>)</p></li>
<li><p>that we can ignore the covariance between the environment variation experience by an individual and the genetic variation in another individual (i.e. <span class="math inline">\(Cov(X_{1E},(X_{2M}+X_{2P}))=0\)</span>).</p></li>
</ol>
<p>The failure of these assumptions to hold can severely undermine our estimates of heritability, but we’ll return to that later. Moving forward with these assumptions, we can write our phenotypic covariance between our pair of individuals as <span class="math display">\[Cov(X_1,X_2) =
Cov((X_{1M},X_{2M})+Cov(X_{1M},X_{2P})+Cov(X_{1P},X_{2M})
+Cov(X_{1P},X_{2P}) \label{cov_rels_1}\]</span> This is saying that under our simple additive model we can see the covariance in phenotypes between individuals as the covariance between the allelic effects in our individuals. We can use our results about the sharing of alleles between relatives to obtain these terms. But before we write down the general case lets quickly work through some examples.<br />
</p>
<h4 id="the-covariance-between-identical-twins">The covariance between Identical Twins</h4>
<p>Lets first consider the case of a pair of identical twins from two unrelated parents. Our pair of twins share their maternal and paternal allele identical by descent (<span class="math inline">\(X_{1M}=X_{2M}\)</span> and <span class="math inline">\(X_{1P}=X_{2P}\)</span>). As their maternal and paternal alleles are not correlated draws from the population, i.e. have no probability of being <span class="math inline">\(IBD\)</span> as we’ve said the parents are unrelated, the covariance between their effects on the phenotype is zero (i.e. <span class="math inline">\(Cov(X_{1P},X_{2M})=Cov(X_{1M},X_{2P})=0\)</span>). In that case eqn. [cov_rels_1] is <span class="math display">\[Cov(X_1,X_2) = Cov((X_{1M},X_{2M})+Cov(X_{1P},X_{2P}) = 2Var(X_{1M})
= V_A\]</span> Now in general identical twins are not going to be super helpful for us in estimating <span class="math inline">\(h^2\)</span> as under models with non-additive effects identical twins have higher covariance than we’d expect as they resemble each other also because of the dominance effects as they don’t just share alleles they share their entire genotype.<br />
</p>
<h4 id="the-covariance-in-phenotype-between-mother-and-child">The covariance in phenotype between mother and child</h4>
<p>. If the mother and father are unrelated individuals (i.e. are two random draws from the population) then the mother and a child share one allele IBD at each locus (i.e. <span class="math inline">\(r_1=1\)</span> and <span class="math inline">\(r_0=r_2=0\)</span>). Half the time our mother transmits her paternal allele to the child, in which case <span class="math inline">\(X_{P1}=X_{M2}\)</span> and so <span class="math inline">\(Cov(X_{P1},X_{M2})=Var(X_{P1})\)</span> and all the other covariances in eqn. [cov_rels_1] zero, and half the time she transmits her maternal allele to the child <span class="math inline">\(Cov(X_{M1},X_{M2})=Var(X_{M1})\)</span> and all the other terms zero. By this argument <span class="math inline">\(Cov(X_1,X_2) = {\tfrac{1}{2}}Var(X_{M1}) + {\tfrac{1}{2}}Var(X_{P1}) = {\tfrac{1}{2}}V_A\)</span>.<br />
</p>
<h4 id="the-covariance-between-general-pairs-of-relatives-under-an-additive-model">The covariance between general pairs of relatives under an additive model</h4>
<p>The two examples make clear that to understand the covariance between phenotypes of relatives we simply need to think about the alleles they share IBD. Consider a pair of relatives (<span class="math inline">\(1\)</span> and <span class="math inline">\(1\)</span>) with a probability <span class="math inline">\(r_0\)</span>, <span class="math inline">\(r_1\)</span>, and <span class="math inline">\(r_2\)</span> of sharing zero, one, or two alleles IBD respectively. When they share zero alleles <span class="math inline">\(Cov((X_{1M}+X_{1P}),(X_{2M}+X_{2P}))=0\)</span>, when they share one allele <span class="math inline">\(Cov((X_{1M}+X_{1P}),(X_{2M}+X_{2P}))=
Var(X_{1M})=\frac{1}{2}V_A\)</span>, and when they share two alleles <span class="math inline">\(Cov((X_{1M}+X_{1P}),(X_{2M}+X_{2P}))=
V_A\)</span>. Therefore, the general covariance between two relatives is <span class="math display">\[Cov(X_1,X_2) = r_0 \times 0 + r_1 \frac{1}{2}V_A + r_2  V_A =
2 F_{1,2} V_A\]</span><br />
So under a simple additive model of the genetic basis of a phenotype to measure the narrow sense heritability we need to measure the covariance between a set of pairs of relatives (assuming that we can remove the effect of shared environmental noise). From the covariance between relatives we can calculate <span class="math inline">\(V_A\)</span>, we can then divide this by the total phenotypic variance to get <span class="math inline">\(h^2\)</span>.<br />
</p>
<p>In polygynous blackbird populations (i.e. males mate with several females), paternal half-sibs can be identified. Suppose that the covariance of tarsus lengths among half-sibs is 0.25 <span class="math inline">\(cm^2\)</span> and that the total phenotypic variance is 4 <span class="math inline">\(cm^2\)</span>. Use these data to estimate <span class="math inline">\(h^2\)</span> for tarsus length in this population.</p>
<p>Another way that we can estimate the narrow sense heritability is through the regression of child’s phenotype on the parental mid-point phenotype. The parental mid-point phenotype is simple the average of the mum and dad’s phenotype. Denoting the child’s phenotype by <span class="math inline">\(X_{kid}\)</span> and mid-point phenotype by <span class="math inline">\(X_{mid}\)</span> so that if we take the regression <span class="math inline">\(X_{kid} \sim X_{mid}\)</span> this regression has slope <span class="math inline">\(\beta = Cov(X_{kid},X_{mid})/Var(X_{mid})\)</span>. The covariance of <span class="math inline">\(Cov(X_{kid},X_{mid})={\tfrac{1}{2}}V_A\)</span>, and <span class="math inline">\(Var(X_{mid}) = {\tfrac{1}{2}}V\)</span> as by taking the average of the parents we have halved the variance, such that the slope of the regression is <span class="math display">\[\beta= \frac{Cov(X_{kid},X_{mid})}{Var(X_{mid})} = \frac{V_A}{V} = h^2\]</span> i.e. the regression of the child’s phenotype on the parental midpoint phenotype is an estimate of the narrow sense heritability. This is a common way to estimate heritability, although it doesn’t bypass the need to control for environmental correlations between relatives.<br />
Our regression allows us to attempt to predict the phenotype of the child given the child; how well we can do this depends on the slope. If the slope is close to zero then the parental phenotypes hold no information about the phenotype of the child, while if the slope is close to one then the parental mid-point is a good guess at the child’s phenotype.<br />
</p>
<figure>
<img src="figures/QT2.png" alt="Regression of parental mid-point phenotype on child’s phenotype. The three panels show decreasing levels of environmental variance (V_E) holding the additive genetic variance constant (V_A=1). In these figures I simulate 100 loci, as described in the caption of Figure [fig:QT1]. I simulate the genotypes and phenotypes of the two parents, and then simulate the child’s genotype following mendelian transmission. The blue line shows x=y the red line shows the best fitting linear regression line. " /><figcaption>Regression of parental mid-point phenotype on child’s phenotype. The three panels show decreasing levels of environmental variance (<span class="math inline">\(V_E\)</span>) holding the additive genetic variance constant (<span class="math inline">\(V_A=1\)</span>). In these figures I simulate <span class="math inline">\(100\)</span> loci, as described in the caption of Figure [fig:QT1]. I simulate the genotypes and phenotypes of the two parents, and then simulate the child’s genotype following mendelian transmission. The blue line shows <span class="math inline">\(x=y\)</span> the red line shows the best fitting linear regression line. </figcaption>
</figure>
<p>More formally the expected phenotype of the child given the parental phenotypes is <span class="math display">\[{\mathbb{E}}(X_{kid} | X_{mum},X_{dad}) = \mu +
\beta(X_{mid} - \mu) =\mu + h^2(X_{mid} - \mu)  \label{predict_kid}\]</span> this follows from the definition of linear regression. So to find the child’s predicted phenotype we simply take the mean phenotype and add on the difference between our parental mid-point multiplied by our narrow sense heritability.<br />
</p>
<h4 id="estimating-additive-genetic-variance-across-a-variety-of-different-relationships.">Estimating additive genetic variance across a variety of different relationships.</h4>
<p>In many natural populations we may have access to individuals of a range of different relationships to each other (through monitoring of the paternity of individuals), but relatively few individuals of a given relationship (e.g. sibs). We can try and use this information as fully as possible in a mixed model framework. Considering equation [pheno_geno_environ] we can write an individual’s phenotype <span class="math inline">\(X_i\)</span> as <span class="math display">\[X_i =  \mu  + X_{A,i} + e_i\]</span> where <span class="math inline">\(e_i \sim N(0,V_E)\)</span> and <span class="math inline">\(X_{A,i}\)</span> is normally distributed across individuals with covariance matrix <span class="math inline">\(V_A A\)</span> where the the entries for an pair of individuals i and i are <span class="math inline">\(A_{ij}= 2 F_{i,j}\)</span> and <span class="math inline">\(A_{ii} 1\)</span>. Given the matrix <span class="math inline">\(A\)</span> we can estimate <span class="math inline">\(V_A\)</span>. We can also add fixed effects into this model to account for generation effects, additional mixed effects could also be included to account for shared environments between particular individuals (e.g. a shared nest). This is sometimes called the “animal model”, and also goes by the name of variance components analysis.</p>
<h4 id="multiple-traits.">Multiple traits.</h4>
<p>Traits often covary with each other, due to both environmentally induced effects (e.g. due to the effects of diet on multiple traits) and due to the expression of underlying genetic covariance between traits. In turn this genetic covariance can reflect pleiotropy, a mechanistic effect of an allele on multiple traits (e.g. variants that effect skin pigmentation often effect hair color) or the genetic linkage of loci independently affecting multiple traits. If we are interested in evolution over short time-scales we can (often) ignore the genetic basis of this correlation.</p>
<p>Consider two traits <span class="math inline">\(X_{1,i}\)</span> and <span class="math inline">\(X_{2,i}\)</span> in an indivdual <span class="math inline">\(i\)</span>, these could be say the indivdual’s leg length and nose length. As before we can write these as <span class="math display">\[\begin{aligned}
X_{1,i} &amp;= \mu_1+ X_{1,A,i} + X_{1,E,i}  \nonumber \\
X_{2,i} &amp;= \mu_2 +X_{2,A,i} + X_{2,E,i} \nonumber \\\end{aligned}\]</span> As before we can talk about the total phenotypic variance (<span class="math inline">\(V_1,V_2\)</span>), environmental variance (<span class="math inline">\(V_{1,E}\)</span> and <span class="math inline">\(V_{2,E}\)</span>), and the additive variance and in trait one and two and (<span class="math inline">\(V_{1,A}\)</span>, <span class="math inline">\(V_{2,A}\)</span>). But now we also have to consider the total covariance <span class="math inline">\(V_{1,2}=Cov(X_{1},X_{2})\)</span>, the environmentally induced covariance between the traits (<span class="math inline">\(V_{E,1,2}=Cov(X_{1,E}
,X_{2,E} )\)</span>) and the additive genetic covariance (<span class="math inline">\(V_{A,1,2}
=Cov(X_{1,A} ,X_{2,A} )\)</span>) between trait one and two.</p>
<p>We can store these values in a matrices <span class="math display">\[\bf{V}= \left( \begin{array}{cc} 
V_{1} &amp; V_{1,2} \\
V_{1,2} &amp; V_{2} \\
\end{array} \right) \label{P_matrix}\]</span> and <span class="math display">\[\bf{G}= \left( \begin{array}{cc} 
V_{1,A} &amp; V_{A,1,2} \\
V_{A,1,2} &amp; V_{2,A} \\
\end{array} \right)  \label{G_matrix}\]</span> we can generalize this to an abitrary number of traits.</p>
<p>We can estimate these quantities, in a similar way to before, by studying the covariance in different traits between relatives:</p>
<p><span class="math display">\[Cov(X_{1,i},X_{2,j}) = 2 F_{i,j} V_{A,1,2}\]</span></p>
<h3 id="the-response-to-selection">The response to selection</h3>
<p>Evolution by natural selection requires:</p>
<ol>
<li><p>Variation in a phenotype</p></li>
<li><p>That survival is non-random with respect to this phenotypic variation.</p></li>
<li><p>That this variation is heritable.</p></li>
</ol>
<p>Points 1 and 2 encapsulate our idea of Natural Selection, but evolution by natural selection will only occur if the 3rd condition is met. It is the heritable nature of variation that couples change within a generation due to natural selection, to change across generations (evolutionary change).<br />
Lets start by thinking about the change within a generation due to directional selection, where selection acts to change the mean phenotype within a generation. For example, a decrease in mean height within a generation, due to taller organisms having a lower chance of surviving to reproduction than shorter organisms. Specifically, we’ll denote our mean phenotype at reproduction by <span class="math inline">\(\mu_S\)</span>, i.e. after selection has acted, and our mean phenotype before selection acts by <span class="math inline">\(\mu_{BS}\)</span>. This second quantity may be hard to measure, as obviously selection acts throughout the life-cycle, so it might be easier to think of this as the mean phenotype if selection hadn’t acted. So the mean phenotype changes within a generation is <span class="math inline">\(\mu_{S} - \mu_{BS}= S\)</span>.<br />
We are interested in predicting the distribution of phenotypes in next generation, in particular we are interested in the mean phenotype in the next generation to understand how directional selection has contributed to evolutionary change. We’ll denote the mean phenotype in offspring, i.e. the mean phenotype in the next generation before selection acts, as <span class="math inline">\(\mu_{NG}\)</span>. The change across generations we’ll call the response to selection <span class="math inline">\(R\)</span> and put this equal to <span class="math inline">\(\mu_{NG}- \mu_{BS}\)</span>.<br />
The mean phenotype in the next generation is <span class="math display">\[\mu_{NG} = {\mathbb{E}}\left( {\mathbb{E}}(X_{kid} | X_{mum},X_{dad}) \right)\]</span> where the outer expectation is over the randomly mating of individuals who survive to reproduce. We can use eqn. [predict_kid] to obtain an expression for this <span class="math display">\[\mu_{NG} = \mu_{BS} +
\beta ( {\mathbb{E}}(X_{mid}) - \mu_{BS})\]</span> so to obtain <span class="math inline">\(\mu_{NG}\)</span> we need to compute <span class="math inline">\({\mathbb{E}}(X_{mid})\)</span> the expected mid-point phenotype of pairs of individuals who survive to reproduce. Well this is just the expected phenotype in the individuals who survived to reproduce (<span class="math inline">\(\mu_{S}\)</span>), so <span class="math display">\[\mu_{NG} = \mu_{BS} +
h^2 (\mu_S - \mu_{BS})\]</span> So we can write our response to selection as <span class="math display">\[R = \mu_{NG} -\mu_{BS}  =
h^2 (\mu_S - \mu_{BS}) = h^2 S \label{breeders_eqn}\]</span> So our response to selection is proportional to our selection differential, and the constant of proportionality is the narrow sense heritability. This equation is sometimes termed the Breeders equation. It is a statement that the evolutionary change across generations (<span class="math inline">\(R\)</span>) is proportional to the change caused by directional selection within a generation, and the strength of this relationship is determined by the narrow sense heritability.<br />
Using the fact that <span class="math inline">\(h^2=V_A/V\)</span> we can rewrite this in a different form as <span class="math display">\[R= V_A \frac{S}{V}\]</span> i.e. our response to selection is the additive genetic variance of our trait (<span class="math inline">\(V_A\)</span>) multiplied by the change within a generation as a fraction of the total phenotypic variance (<span class="math inline">\(S/V\)</span>, sometimes called the the selection gradient <span class="math inline">\(\beta\)</span>).<br />
If our selection pressure is sustained over many generations we can use our breeders equation to predict the response. If we are willing to assume that our heritability does not change and we maintain a constant selection gradient, then after <span class="math inline">\(n\)</span> generations our phenotype mean will have shifted <span class="math display">\[n h^2 S\]</span> i.e. our population will keep up a linear response to selection.</p>
<p><img src="figures/QT3.png" alt="image" /></p>
<p>A change in mean phenotype within a generation occurs because of the differential fitness of our organisms. To think more carefully about this change within a generation lets think about a simple fitness model where our phenotype affects the viability of our organisms (i.e. the probability they survive to reproduce). The probability that an individual has a phenotype <span class="math inline">\(X\)</span> before selection is <span class="math inline">\(p(X)\)</span>, so that the mean phenotype before selection is <span class="math display">\[\mu_{BS} = {\mathbb{E}}[X] =  \int_{-\infty}^{\infty} x p(x) dx\]</span> The probability that an organism with a phenotype <span class="math inline">\(X\)</span> survives to reproduce is <span class="math inline">\(w(X)\)</span>, and we’ll think about this as the fitness of our organism. The probability distribution of phenotypes in those who do reproduce is <span class="math display">\[{\mathbb{P}}(X | \textrm{survive}) =  \frac{p(x) w(x)}{
\int_{-\infty}^{\infty} p(x) w(x) dx}.\]</span> where the denominator is a normalization constant which ensures that our phenotypic distribution integrates to one. The denominator also has the interpretation of being the mean fitness of the population, which we’ll call <span class="math inline">\({\overline{w}}\)</span>, i.e. <span class="math display">\[{\overline{w}}=  
\int_{-\infty}^{\infty} p(x) w(x) dx.\]</span></p>
<p>Therefore, we can write the mean phenotype in those who survive to reproduce as <span class="math display">\[\mu_S = \frac{1}{{\overline{w}}}\int_{-\infty}^{\infty} x p(x) w(x) dx\]</span></p>
<p>If we mean center our population, i.e. set the phenotype before selection to zero, then <span class="math display">\[S= \frac{1}{{\overline{w}}}\int_{-\infty}^{\infty} x p(x) w(x) dx\]</span> if <span class="math inline">\(\mu_S=0\)</span>. Inspecting this more closely we can see that <span class="math inline">\(S\)</span> has the form of a covariance between our phenotype <span class="math inline">\(X\)</span> and our fitness <span class="math inline">\(w(X)\)</span> (<span class="math inline">\(Cov(X,w(X))\)</span>). Thus our change in mean phenotype is directly a measure of the covariance of our phenotype and our fitness. Rewriting our breeder’s equation using this observation we see <span class="math display">\[R = \frac{V_A}{V}  Cov(X,w(X))\]</span> we see that the response to selection is due to the fact that our fitness (viability) of our organisms/parents covaries with our phenotype, and that our child’s phenotype is correlated with the parent phenotype.</p>
<p>A population of red deer were trapped on Jersey (an island off of England) during the last inter-glacial period. From the fossil record we can see that the population rapidly adapted to their new surroundings, presumably due to reduced predation and limited food. Within 6,000 years they evolved from an estimated mean weight of the population of 200kg to an estimated mean weight of 36kg (a 6 fold reduction! True story see: <a href="http://www.nature.com/nature/journal/v342/n6249/abs/342539a0.html">Lister, A.M. Nature 1989</a>). Using a current day population on the mainland you estimate that the generation time of red deer is 5 years and that the narrow sense heritability of the phenotype is 0.5. (Assuming discrete generations).<br />
<span><strong>A)</strong></span> Estimate the mean change per generation in the mean body weight.<br />
<span><strong>B)</strong></span> Estimate the change in mean body weight caused by selection within a generation.<br />
<span><strong>C)</strong></span> What do you have to assume to perform the calculations in B. Assuming we only have fossils from the founding population and the population after 6000 years, should we assume that the calculations accurately reflect what actually occurred within our population?</p>
<h4 id="the-response-of-multiple-traits-to-selection-the-multivariate-breeders-equation.">The response of multiple traits to selection, the multivariate breeder’s equation.</h4>
<p>We can generalize these results for multiple traits, to ask how selection on multiple phenotypes plays out over short time intervals. We’ll write our change in the mean our multiple phenotypes within a generation as the vector <span class="math inline">\(\bf{S}\)</span> and our response across multiple generations as the vector <span class="math inline">\(\bf{R}\)</span>. These two quantities are related by <span class="math display">\[\bf{R} = \bf{G} \bf{V}^{-1} \bf{S} = \bf{G} \bf{\beta}\]</span> where <span class="math inline">\(\bf{V}\)</span> and <span class="math inline">\(\bf{G}\)</span> are our matrices of the variance-covariance of phenotypes and additive genetic values (eqn. ) and <span class="math inline">\(\bf{\beta}\)</span> is a vector of selection gradients (i.e. the change within a generation as a fraction of the total phenotypic variance). To make this a bit more intuitive, consider two traits we are writing <span class="math display">\[\begin{aligned}
R_1 &amp; = V_{A,1} \beta_1 + V_{A,1,2} \beta_2 \nonumber \\
R_2 &amp; = V_{A,2} \beta_2 + V_{A,1,2} \beta_1  \nonumber \\\end{aligned}\]</span> where the <span class="math inline">\(1\)</span> and <span class="math inline">\(2\)</span> index our two different traits. This is a statement that our response in any one phenotype is modified by selection on other traits that covary with that trait. This offers a good way to think about how genetic trade offs play out over evolution over short time-scales.</p>
<h1 id="one-locus-models-of-selection">One-locus models of selection</h1>
<h2 id="fitness">Fitness</h2>
<p>As we have seen, natural selection occurs when there are differences between individuals in fitness. We may define fitness in various ways. Most commonly, it is defined with respect to the contribution of a phenotype or genotype to the next generation. Differences in fitness can arise at any point during the life cycle. For instance, different genotypes or phenotypes may have different survival probabilities from one stage in their life to the stage of reproduction (viability), or they may differ in the number of offspring produced (fertility), or both. Here, we define the absolute fitness of a genotype as the expected number of offspring of an individual of that genotype.<br />
</p>
<h2 id="haploid-selection-model">Haploid selection model</h2>
<p>We start out by modelling selection in a haploid model, as this is mathematically relatively simple. Let the number of individuals carrying alleles <span class="math inline">\(A_1\)</span> and <span class="math inline">\(A_2\)</span> in generation <span class="math inline">\(t\)</span> be <span class="math inline">\(P_t\)</span> and <span class="math inline">\(Q_t\)</span>. Then, the relative frequencies at time <span class="math inline">\(t\)</span> of alleles <span class="math inline">\(A_1\)</span> and <span class="math inline">\(A_2\)</span> are <span class="math inline">\(p_t = P_t / (P_t + Q_t)\)</span> and <span class="math inline">\(q_t = Q_t / (P_t + Q_t) = 1 - p_t\)</span>. Further, assume that individuals of type <span class="math inline">\(A_1\)</span> and <span class="math inline">\(A_2\)</span> on average produce <span class="math inline">\(W_1\)</span> and <span class="math inline">\(W_2\)</span> offspring individuals, respectively. We call <span class="math inline">\(W_i\)</span> the absolute fitness.<br />
Therefore, in the next generation, the absolute number of carriers of <span class="math inline">\(A_1\)</span> and <span class="math inline">\(A_2\)</span> are <span class="math inline">\(P_{t+1} = W_1 P_t\)</span> and <span class="math inline">\(Q_{t+1} = W_2 Q_t\)</span>, respectively. The mean absolute fitness of the population at time <span class="math inline">\(t\)</span> is <span class="math display">\[\label{eq:meanAbsFit}
	{\overline{W}}_t = W_1 \frac{P_t}{P_t + Q_t} + W_2 \frac{Q_t}{P_t + Q_t} = W_1 p_t + W_2 q_t,\]</span> i.e. the sum of the fitness of the two types weighted by their relative frequencies. Note that the mean fitness depends on time, as it is a function of the allele frequencies, which are themselves time dependent.<br />
The frequency of allele <span class="math inline">\(A_1\)</span> in the next generation is then given by <span class="math display">\[\label{eq:eq:recHaplMod1}
	p_{t+1} = \frac{P_{t+1}}{P_{t+1} + Q_{t+1}} = \frac{W_1 P_t}{W_1 P_t + W_2 Q_t}
		= \frac{W_1 p_t}{W_1 p_t + W_2 q_t} = \frac{W_1}{{\overline{W}}_t} p_t.\]</span></p>
<p>Importantly, eqn. ([eq:eq:recHaplMod1]) tells us that the change in <span class="math inline">\(p\)</span> only depends on a ratio of fitnesses. Therefore, we need to specify fitness only up to an arbitrary constant. As long as we multiply all fitnesses by the same value, that constant will cancel out and eqn. ([eq:eq:recHaplMod1]) will hold. Based on this argument, it is very common to scale absolute fitnesses by the absolute fitness of one of the genotypes, e.g. the most or the least fit genotype, to obtain relative fitnesses. Here, we will use <span class="math inline">\(w_i\)</span> for the relative fitness of genotype <span class="math inline">\(i\)</span>. If we choose to scale by the absolute fitness of genotype <span class="math inline">\(A_1\)</span>, we obtain the relative fitnesses <span class="math inline">\(w_1 = W_1/W_1 = 1\)</span> and <span class="math inline">\(w_2 = W_2/W_1\)</span>.<br />
Without loss of generality, we can therefore rewrite eqn. ([eq:eq:recHaplMod1]) as <span class="math display">\[\label{eq:recHaplMod2}
	p_{t+1} = \frac{w_1}{{\overline{w}}} p_t,\]</span> dropping the dependence of the mean fitness on time in our notation, but remembering it. The change in frequency from one generation to the next is then given by <span class="math display">\[\Delta p_t = p_{t+1} - p_t= \frac{ w_1 p_t}{ {\overline{w}}} - p_t = \frac{w_1 p_t - {\overline{w}}p_t}{{\overline{w}}} = \frac{w_1 p_t - (w_1 p_t + w_2 q_t) p_t}{{\overline{w}}} = \frac{w_1 - w_2}{{\overline{w}}} p_t q_t,
\label{eq:deltap_haploid}\]</span> recalling that <span class="math inline">\(q_t = 1 - p_t\)</span>.<br />
Assuming that the fitnesses of the two alleles are constant over time, the number of the two allelic types <span class="math inline">\(\tau\)</span> generations after time <span class="math inline">\(t\)</span> are <span class="math inline">\(P_{t+\tau} = (W_1)^{\tau} P_t\)</span> and <span class="math inline">\(Q_{t+\tau}=  (W_2)^{\tau} Q_t\)</span>, respectively. Therefore, the relative frequency of allele <span class="math inline">\(A_1\)</span> after <span class="math inline">\(\tau\)</span> generations past <span class="math inline">\(t\)</span> is <span class="math display">\[p_{t+\tau} = \frac{ (W_1)^{\tau} P_t}{ (W_1)^{\tau} P_t+(W_2)^{\tau} Q_t} = \frac{ (w_1)^{\tau} P_t}{ (w_1)^{\tau} P_t+(w_2)^{\tau} Q_t} = \frac{p_t}{p_t + (w_2/w_1)^{\tau} q_t},
	\label{eq:haploid_tau_gen}\]</span> where the last step includes dividing the whole term by <span class="math inline">\(w_1\)</span> and switching from absolute to relative allele frequencies.<br />
Rearranging eqn.  and setting <span class="math inline">\(t = 0\)</span>, we can work out the time <span class="math inline">\(\tau\)</span> for the frequency of <span class="math inline">\(A_1\)</span> to change from <span class="math inline">\(p_0\)</span> to <span class="math inline">\(p_{\tau}\)</span>. First, we write <span class="math display">\[p_{\tau} = \frac{p_0}{p_0 + (w_2/w_1)^{\tau} q_0}\]</span> and rearrange this to obtain <span class="math display">\[\label{eq:estTau}
	\frac{p_{\tau}}{q_{\tau}} = \frac{p_0}{q_0} \left(\frac{w_1}{w_2}\right)^{\tau}.\]</span> Solving this for <span class="math inline">\(\tau\)</span> yields <span class="math display">\[\label{eq:solTau}
	\tau = \log \left(\frac{p_{\tau} q_0}{q_{\tau} p_0}\right) /  \log\left(  \frac{w_1}{w_2} \right).\]</span><br />
In practice, it is often helpful to parametrize the relative fitnesses <span class="math inline">\(w_i\)</span> in a specific way. For example, we may set <span class="math inline">\(w_1 = 1\)</span> and <span class="math inline">\(w_2 = 1 - s\)</span>, where <span class="math inline">\(s\)</span> is called the selection coefficient. Using this parametrization, <span class="math inline">\(s\)</span> is simply the difference in relative fitnesses between the two alleles. Equation becomes <span class="math display">\[\label{eq:haploid_tau_gen_expl}
	p_{t+\tau} = \frac{p_{t}}{p_{t} + q_{t} (1 - s)^{\tau}},\]</span> as <span class="math inline">\(w_2 / w_1 = 1 - s\)</span>. Then, if <span class="math inline">\(s \ll 1\)</span>, we can approximate <span class="math inline">\((1-s)^{\tau}\)</span> in the denominator by <span class="math inline">\(\exp(-s\tau)\)</span> to obtain <span class="math display">\[\label{eq:haploid_logistic growth}
	p_{t+\tau} \approx \frac{p_t}{p_t + q_t e^{-s\tau}}.\]</span> This equation takes the form of a logistic function. That is because we are looking at the relative frequencies of two ‘populations’ (of alleles <span class="math inline">\(A_1\)</span> and <span class="math inline">\(A_2\)</span>) that are growing (or declining) exponentially, under the constraint that <span class="math inline">\(p\)</span> and <span class="math inline">\(q\)</span> always sum to 1.<br />
Moreover, eqn.  for the time <span class="math inline">\(\tau\)</span> it takes for a certain change in frequency to occur becomes <span class="math display">\[\label{eq:estTauExpl}
	\tau = - \log \left(\frac{p_{\tau} q_0}{q_{\tau} p_0}\right) /  \log\left(1-s\right).\]</span> Assuming again that <span class="math inline">\(s \ll 1\)</span>, this simplifies to <span class="math display">\[\label{eq:estTauExplSimpl}
	\tau \approx \frac{1}{s} \log \left(\frac{p_{\tau} q_0}{q_{\tau} p_0}\right).\]</span></p>
<p>One particular case of interest is the time it takes to go from an absolute frequency of 1 to near fixation in a population of size <span class="math inline">\(N\)</span>. In this case, we have <span class="math inline">\(p_0 = 1/N\)</span>, and we may set <span class="math inline">\(p_{\tau} = 1 - 1/N\)</span>, which is very close to fixation. Of course, we then have <span class="math inline">\(q_0 = 1 - 1/N\)</span> and <span class="math inline">\(q_{\tau} = 1 - 1/N\)</span>. If <span class="math inline">\(N\)</span> is sufficiently large, we may for mathematical convenience approximate <span class="math inline">\(q_0\)</span> by <span class="math inline">\(q_0 = 1\)</span>. Plugging these values into eqn. , we obtain <span class="math display">\[\label{eq:fixTimeSimpl}
	\tau \approx \frac{1}{s} \log\left( \frac{1 - 1/N}{(1 - 1/N)\ 1/N}  \right) = \frac{1}{s} \log(N)\]</span> as an approximation for the time to fixation.</p>
<h4 id="haploid-model-with-fluctuating-selection">Haploid model with fluctuating selection</h4>
<p>We can now consider the case where the fitnesses depend on time, and say that <span class="math inline">\(w_{1,t}\)</span> and <span class="math inline">\(w_{2,t}\)</span> are the fitnesses of the two types in generation <span class="math inline">\(t\)</span>. The frequency of allele <span class="math inline">\(A_1\)</span> in generation <span class="math inline">\(t+1\)</span> is <span class="math display">\[p_{t+1} = \frac{w_{1,t}}{{\overline{w}}_t} p_t,\]</span> which simply follows from eqn. . The ratio of the frequency of allele <span class="math inline">\(A_1\)</span> to that of allele <span class="math inline">\(A_2\)</span> in generation <span class="math inline">\(t+1\)</span> is <span class="math display">\[\frac{p_{t+1}}{q_{t+1}} = \frac{w_{1,t}}{w_{2,t}}  \frac{p_{t}}{q_{t}}.\]</span> Therefore, if we think of the two alleles starting in generation <span class="math inline">\(t\)</span> at frequencies <span class="math inline">\(p_t\)</span> and <span class="math inline">\(q_t\)</span>, then <span class="math inline">\(\tau\)</span> generations later, <span class="math display">\[\frac{p_{t+\tau}}{q_{t+\tau}} = \left(\prod_{i=t}^{\tau-1} \frac{w_{1,i}}{w_{2,i}}  \right) \frac{p_{t}}{q_{t}}.\]</span><br />
The question of which allele is increasing or decreasing in frequency comes down to whether <span class="math inline">\(\left(\prod_{i=t}^{\tau-1} \frac{w_{1,i}}{w_{2,i}}  \right)\)</span> is <span class="math inline">\(&gt;1\)</span> or <span class="math inline">\(&lt;1\)</span>. As it is a little hard to think about this ratio, we can instead take the <span class="math inline">\(\tau^{\mathrm{th}}\)</span> root of it and consider <span class="math display">\[\sqrt[\tau]{\left(\prod_{i=t}^{\tau-1} \frac{w_{1,i}}{w_{2,i}}  \right)} = \frac{\sqrt[\tau]{\prod_{i=t}^{\tau-1}w_{1,i}}}{\sqrt[\tau]{\prod_{i=t}^{\tau-1}w_{2,i}}}.\]</span> The term <span class="math inline">\(\sqrt[\tau]{\prod_{i=t}^{\tau-1}w_{1,i}}\)</span> is the geometric mean fitness of allele <span class="math inline">\(A_1\)</span> over the <span class="math inline">\(\tau\)</span> generations past generation <span class="math inline">\(t\)</span>. Therefore, allele <span class="math inline">\(A_1\)</span> will only increase in frequency if it has a higher geometric mean fitness than allele <span class="math inline">\(A_2\)</span> (at least in our simple deterministic model).<br />
</p>
<h2 id="diploid-model">Diploid model</h2>
<p>We will now move on to a diploid model of a single locus with two segregating alleles. We will assume that the difference in fitness between the three genotypes comes from differences in viability, i.e. differential survival of individuals from the formation of zygotes to reproduction. We denote the absolute fitnesses of genotypes <span class="math inline">\(A_1A_1\)</span>, <span class="math inline">\(A_1A_2\)</span>, and <span class="math inline">\(A_2A_2\)</span> by <span class="math inline">\(W_{11}\)</span>, <span class="math inline">\(W_{12}\)</span>, and <span class="math inline">\(W_{22}\)</span>. Specifically, <span class="math inline">\(W_{ij}\)</span> is the probability that a zygote of genotype <span class="math inline">\(A_iA_j\)</span> survives to reproduction. Assuming that individuals mate at random, the number of zygotes that are of the three genotypes and form generation <span class="math inline">\(t\)</span> are <span class="math display">\[Np_t^2, ~~~  N2p_tq_t, ~~~ Nq_t^2.\]</span></p>
<p>The mean fitness of the population of zygotes is then <span class="math display">\[{\overline{W}}_t = W_{11} p_t^2+W_{12} 2p_tq_t  +  W_{22} q_t^2.\]</span> Again, this is simply the weighted mean of the genotypic fitnesses.<br />
How many zygotes of each of the three genotypes survive to reproduce? An individual of genotype <span class="math inline">\(A_1A_1\)</span> has a probability of <span class="math inline">\(W_{11}\)</span> of surviving to reproduce, and similarly for other genotypes. Therefore, the expected number of <span class="math inline">\(A_1A_1\)</span>, <span class="math inline">\(A_1A_2\)</span>, and <span class="math inline">\(A_2A_2\)</span> individuals who survive to reproduce is <span class="math display">\[NW_{11} p_t^2, ~~~ NW_{12} 2p_tq_t , ~~~ N W_{22} q_t^2.\]</span> It then follows that the total number of individuals who survive to reproduce is <span class="math display">\[N \left(W_{11} p_t^2+W_{12} 2p_tq_t  +  W_{22} q_t^2 \right).\]</span> This is simply the mean fitness of the population multiplied by the population size (i.e. <span class="math inline">\(N {\overline{w}}\)</span>).<br />
The relative frequency of <span class="math inline">\(A_1A_1\)</span> individuals at reproduction is simply the number of <span class="math inline">\(A_1A_1\)</span> genotype individuals at reproduction (<span class="math inline">\(NW_{11} p_t^2\)</span>) divided by the total number of individuals who survive to reproduce (<span class="math inline">\(N {\overline{W}}\)</span>), and likewise for the other two genotypes. Therefore, the relative frequency of individuals with the three different genotypes at reproduction is <span class="math display">\[\frac{NW_{11} p_t^2}{N{\overline{W}}}, ~~~ \frac{NW_{12} 2p_tq_t}{N{\overline{W}}} , ~~~ \frac{N W_{22} q_t^2}{N{\overline{W}}}\]</span> (see Table [dip_fitness_table]).<br />
</p>
<table>
<caption>Relative genotype frequencies after one episode of viability selection.<span data-label="dip_fitness_table"></span></caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: center;"><span class="math inline">\(A_1A_1\)</span></th>
<th style="text-align: center;"><span class="math inline">\(A_1A_2\)</span></th>
<th style="text-align: center;"><span class="math inline">\(A_2A_2\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Absolute no. at birth</td>
<td style="text-align: center;"><span class="math inline">\(Np_t^2\)</span></td>
<td style="text-align: center;"><span class="math inline">\(N2p_tq_t\)</span></td>
<td style="text-align: center;"><span class="math inline">\(Nq_t^2\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">Fitnesses</td>
<td style="text-align: center;"><span class="math inline">\(W_{11}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(W_{12}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(W_{22}\)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Absolute no. at reproduction</td>
<td style="text-align: center;"><span class="math inline">\(NW_{11} p_t^2\)</span></td>
<td style="text-align: center;"><span class="math inline">\(NW_{12} 2p_tq_t\)</span></td>
<td style="text-align: center;"><span class="math inline">\(N W_{22} q_t^2\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">Relative freq. at reproduction</td>
<td style="text-align: center;"><span class="math inline">\(\frac{NW_{11} p_t^2}{N {\overline{W}}} = \frac{W_{11}}{{\overline{W}}} p_{t}^2\)</span></td>
<td style="text-align: center;"><span class="math inline">\(\frac{NW_{12} 2p_tq_t}{N
{\overline{W}}} = \frac{W_{12}}{{\overline{W}}} 2 p_{t} q_{t}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(\frac{N W_{22} q_t^2}{N{\overline{W}}} = \frac{W_{22}}{{\overline{W}}} q_{t}^2\)</span></td>
</tr>
</tbody>
</table>
<p>As there is no difference in the fecundity of the three genotypes, the allele frequencies in the zygotes forming the next generation are simply the allele frequency among the reproducing individuals of the previous generation. Hence, the frequency of <span class="math inline">\(A_1\)</span> in generation <span class="math inline">\(t+1\)</span> is <span class="math display">\[p_{t+1} = \frac{W_{11} p_t^2 + W_{12} p_tq_t}{{\overline{W}}}
	\label{pgen_dip}.\]</span></p>
<p>Note that, again, the absolute value of the fitnesses is irrelevant to the frequency of the allele. Therefore, we can just as easily replace the absolute fitnesses with the relative fitnesses. That is, we may replace <span class="math inline">\(W_{ij}\)</span> by <span class="math inline">\(w_{ij} = W_{ij}/W_{11}\)</span>, for instance.<br />
</p>
<p>You have been studying an annual wildflower for many generations with two color morphs orange and white. You have discovered that a single bi-allelic locus controls flower color, with the white allele being recessive. The pollinator of these plants is an almost blind bat, so individuals are pollinated at random with respect to flower color. Your population census of 200 individuals showed that the population consisted of 168 orange-flowered individuals, and 32 white-flowered individuals.<br />
Heavy February rainfall creates optimal growing conditions for an exotic herbivorous beetle with a preference for orange-flowered individuals. This year it arrives at your study site with a ravenous appetite. Only 50% of orange-flowered individuals survive its wrath, while 90% of white-flowered individuals survive until the end of the growing season. Additionally, surviving orange flowered individuals produce 80 seeds on average, while surviving white-flowered individuals produce 100 seeds on average. <span><strong>A</strong></span> What is the initial frequency of the white allele, and what do you have to assume to obtain this?<br />
<span><strong>B</strong></span> What is the frequency of the white allele in the seeds forming the next generation?<br />
</p>
<p>The change in frequency from generation <span class="math inline">\(t\)</span> to <span class="math inline">\(t+1\)</span> is <span class="math display">\[\Delta p_t = p_{t+1} -p_{t}= \frac{w_{11} p_t^2 + w_{12} p_tq_t}{{\overline{w}}} - p_t. \label{deltap_dip1}\]</span> To simplify this equation, we will first define two variables <span class="math inline">\({\overline{w}}_1\)</span> and <span class="math inline">\({\overline{w}}_2\)</span> as <span class="math display">\[\begin{aligned}
	{\overline{w}}_1 &amp; = w_{11} p_t + w_{12} q_t, \\
	{\overline{w}}_2 &amp; =  w_{12} p_t+ w_{22} q_t.\end{aligned}\]</span> These are called the marginal fitnesses of allele <span class="math inline">\(A_1\)</span> and <span class="math inline">\(A_2\)</span>, respectively. They are so called as <span class="math inline">\({\overline{w}}_1\)</span> is the average fitness of an allele <span class="math inline">\(A_1\)</span>, i.e. the fitness of <span class="math inline">\(A_1\)</span> in a homozygote weighted by the probability it is in a homozygote (<span class="math inline">\(p_t\)</span>) plus the fitness of <span class="math inline">\(A_1\)</span> in a heterozygote weighted by the probability it is in a heterozygote (<span class="math inline">\(q_t\)</span>). We further note that the mean relative fitness can be expressed in terms of the marginal fitnesses as <span class="math display">\[\label{eq:meanFitInTermsOfMargFit}
	{\overline{w}}= {\overline{w}}_1 p_t + {\overline{w}}_2 q_t,\]</span> where, for notational simplicity, we have omitted the dependence of mean and marginal fitnesses on time.<br />
We can then rewrite eqn.  using <span class="math inline">\({\overline{w}}_1\)</span> and <span class="math inline">\({\overline{w}}_2\)</span> as <span class="math display">\[\Delta p_t = \frac{ ({\overline{w}}_1-{\overline{w}}_2)}{{\overline{w}}} p_t q_t.
	\label{deltap_dip2}\]</span> The sign of <span class="math inline">\(\Delta p_t\)</span>, i.e. whether allele <span class="math inline">\(A_1\)</span> increases of decreases in frequency, depends only on the sign of <span class="math inline">\(({\overline{w}}_1-{\overline{w}}_2)\)</span>. The frequency of <span class="math inline">\(A_1\)</span> will keep increasing over the generations so long as its marginal fitness is higher than that of <span class="math inline">\(A_2\)</span>, i.e. <span class="math inline">\({\overline{w}}_1 &gt; {\overline{w}}_2\)</span>, while if <span class="math inline">\({\overline{w}}_1 &lt; {\overline{w}}_2\)</span>, the frequency of <span class="math inline">\(A_1\)</span> will decrease. Note the similarity between eqn.  and the respective expression for the haploid model in eqn. . (We will return to the special case where <span class="math inline">\({\overline{w}}_1 = {\overline{w}}_2\)</span> shortly).<br />
We can also rewrite as <span class="math display">\[\Delta p_t =\frac{1}{2} \frac{p_tq_t}{{\overline{w}}} \frac{d {\overline{w}}}{dp},
\label{deltap_dip3}\]</span> the demonstration of this we leave to the reader. This form shows that the frequency of <span class="math inline">\(A_1\)</span> will increase (<span class="math inline">\(\Delta p_t &gt; 0\)</span>) if the mean fitness is an increasing function of the frequency of <span class="math inline">\(A_1\)</span> (i.e. if <span class="math inline">\(\frac{d {\overline{w}}}{dp}&gt;0\)</span>). On the other hand, the frequency of <span class="math inline">\(A_1\)</span> will decrease (<span class="math inline">\(\Delta p_t &lt; 0\)</span>) if the mean fitness is a decreasing function of the frequency of <span class="math inline">\(A_1\)</span> (i.e. if <span class="math inline">\(\frac{d {\overline{w}}}{dp}&lt;0\)</span>). Thus, although selection acts on individuals, under this simple model, selection is acting to increase the mean fitness of the population. <span><span> The rate of this increase is proportional to the variance in allele frequencies within the population (<span class="math inline">\(p_tq_t\)</span>).</span></span><br />
</p>
<p>Show that eqns.  and are equivalent.<br />
</p>
<p>So far, our treatment of the diploid model of selection has been in terms of generic fitnesses <span class="math inline">\(w_{ij}\)</span>. In the following, we will use particular parametrizations to gain insight about two specific modes of selection: directional selection and heterozygote advantage.</p>
<h3 id="diploid-directional-selection">Diploid directional selection</h3>
<p>Directional selection means that one of the two alleles always has higher marginal fitness than the other one. Let us assume that <span class="math inline">\(A_1\)</span> is the fitter allele, so that <span class="math inline">\(w_{11} \geq w_{12} \geq w_{22}\)</span>, and hence <span class="math inline">\({\overline{w}}_1 &gt; {\overline{w}}_2\)</span>. As we are interested in changes in allele frequencies, we <span><span> may use</span></span> relative fitnesses. We parameterize the reduction in relative fitness in terms of a selection coefficient, similar to the one we met in the haploid selection section, as follows:<br />
</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;">genotype</td>
<td style="text-align: center;"><span class="math inline">\(A_1A_1\)</span></td>
<td style="text-align: center;"><span class="math inline">\(A_1A_2\)</span></td>
<td style="text-align: center;"><span class="math inline">\(A_2A_2\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">absolute fitness</td>
<td style="text-align: center;"><span class="math inline">\(W_{11}\)</span></td>
<td style="text-align: center;"><span class="math inline">\( \geq W_{12} \geq\)</span></td>
<td style="text-align: center;"><span class="math inline">\(W_{22}\)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">relative fitness (generic)</td>
<td style="text-align: center;"><span class="math inline">\(w_{11} = W_{11}/W_{11}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(w_{12} = W_{12}/W_{11}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(w_{22} = W_{22}/W_{11}\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">relative fitness (specific)</td>
<td style="text-align: center;"><span class="math inline">\(1\)</span></td>
<td style="text-align: center;"><span class="math inline">\(1-sh\)</span></td>
<td style="text-align: center;"><span class="math inline">\(1-s\)</span>.</td>
</tr>
</tbody>
</table>
<p><br />
</p>
<p>Here, the selection coefficient <span class="math inline">\(s\)</span> is the difference in relative fitness between the two homozygotes, and <span class="math inline">\(h\)</span> is the dominance coefficient. <span><span> For selection to be directional, we require that <span class="math inline">\(0 \leq h \leq 1\)</span> holds. The dominance coefficient allows us to move between two extremes. One is when <span class="math inline">\(h = 0\)</span>, such that allele <span class="math inline">\(A_1\)</span> is fully dominant and <span class="math inline">\(A_2\)</span> fully recessive. In this case, the heterozygote <span class="math inline">\(A_1A_2\)</span> is as fit as the <span class="math inline">\(A_1A_1\)</span> homozgyote genotype. The inverse holds when <span class="math inline">\(h = 1\)</span>, such that allele <span class="math inline">\(A_1\)</span> is fully recessive and <span class="math inline">\(A_2\)</span> fully dominant.</span></span><br />
We can then rewrite eqn.  as <span class="math display">\[\Delta p_t = \frac{p_ths + q_t s(1-h)}{{\overline{w}}}p_tq_t ,
\label{deltap_direct}\]</span> where <span class="math display">\[{\overline{w}}_t = 1-2p_tq_t sh-q_t^2s.\]</span><br />
</p>
<p>Comparing the red (<span class="math inline">\(h=0\)</span>) and black (<span class="math inline">\(h=0.5\)</span>) trajectories in Figure [fig:diploid_traj], provide an explanation for why <span class="math inline">\(A_1\)</span> increases faster initially if <span class="math inline">\(h=0\)</span>, but then approaches fixation more slowly compared to the case of <span class="math inline">\(h=0.5\)</span>.</p>
<p>A special case is when <span class="math inline">\(h = 0.5\)</span>. This case is the case of no dominance, as the interaction among alleles with respect to fitness is strictly additive. Then, eqn.  simplifies to <span class="math display">\[\Delta p_t = \frac{1}{2}\frac{s}{{\overline{w}}}p_tq_t .
	\label{deltap_add}\]</span></p>
<p>If selection is very weak, i.e. <span class="math inline">\(s \ll 1\)</span>, the denominator (<span class="math inline">\({\overline{w}}\)</span>) is close to <span class="math inline">\(1\)</span> and we have <span class="math display">\[\Delta p_t = \frac{1}{2} s p_t q_t .
	\label{deltap_add_simpl}\]</span> It is instructive to compare eqn.  to the respective expression under the haploid model. To this purpose, start from the generic term for <span class="math inline">\(\Delta p_t\)</span> under the haploid model in eqn.  and set <span class="math inline">\(w_1 = 1\)</span> and <span class="math inline">\(w_2 = 1-s\)</span>. Again, assume that <span class="math inline">\(s\)</span> is small, so that eqn.  becomes <span class="math inline">\(\Delta p_t = s p_t q_t\)</span>. Hence, if <span class="math inline">\(s\)</span> is small, the diploid model of directional selection without dominance is identical to the haploid model, up to a factor of <span class="math inline">\(1/2\)</span>. That factor is due to the choice of the parametrisation; we could have set <span class="math inline">\(w_{11} = 1\)</span>, <span class="math inline">\(w_{12} = 1-s\)</span>, and <span class="math inline">\(w_{22} = 1-2s\)</span> in diploid model instead, in which case the agreement with the haploid model would be perfect.<br />
From this analogy, we can borrow some insight we gained for the haploid model. Specifically, the trajectory of the frequency of allele <span class="math inline">\(A_1\)</span> in the diploid model without dominance follows a logistic growth curve similar to . Similarly, eqn.  for the haploid model suggests that in the diploid model without dominance it takes</p>
<p><span class="math display">\[\tau \approx \frac{2}{s} \log(2N)\]</span></p>
<p>generations for the favourable allele (<span class="math inline">\(A_1\)</span>) to transit from its entry into the population (<span class="math inline">\(p_0 =1/(2N)\)</span>) to close to fixation (<span class="math inline">\(p_{\tau} =1-1/(2N)\)</span>). Note again the difference by a factor of 2 due to the choice of parametrization. Also, the total number of alleles is <span class="math inline">\(2N\)</span> in the diploid model, rather than <span class="math inline">\(N\)</span>, which explains another factor of 2 in the argument of the logarithm. More generally we can use this correspondance of the trajectory additive diploid model to the haploid model to understand how quickly allele frequencies should change given a selection coefficient between arbitrary frequencies.<br />
</p>
<figure>
<img src="figures/simple_diploid_trajs.png" alt="The trajectory of the frequency of allele A_1, starting from p_{0}=0.01, for a selection coefficient s=0.01 and three different dominance coefficients." /><figcaption>The trajectory of the frequency of allele <span class="math inline">\(A_1\)</span>, starting from <span class="math inline">\(p_{0}=0.01\)</span>, for a selection coefficient <span class="math inline">\(s=0.01\)</span> and three different dominance coefficients.<span data-label="fig:diploid_traj"></span></figcaption>
</figure>
<p>An autosomal pesticide resistance allele is at 50% frequency in a species of flies. We stop using the pesticide, and within 20 years the frequency of the allele is 5% in the new-born flies. There are two fly generations per year. Assuming that the allele affects fitness in an additive fashion, estimate the selection coefficient acting against the resistance allele.</p>
<h3 id="heterozygote-advantage">Heterozygote advantage</h3>
<p>What if the heterozygotes are fitter than either of the homozygotes? In this case, it is useful to parameterize the relative fitnesses as follows:<br />
</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;">genotype</td>
<td style="text-align: center;"><span class="math inline">\(A_1A_1\)</span></td>
<td style="text-align: center;"><span class="math inline">\(A_1A_2\)</span></td>
<td style="text-align: center;"><span class="math inline">\(A_2A_2\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">absolute fitness</td>
<td style="text-align: center;"><span class="math inline">\(w_{11}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(&lt;w_{12}&gt;\)</span></td>
<td style="text-align: center;"><span class="math inline">\(w_{22}\)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">relative fitness (generic)</td>
<td style="text-align: center;"><span class="math inline">\(w_{11}=W_{11}/W_{12}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(w_{12} = W_{12}/W_{12}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(w_{22} = W_{22}/W_{12}\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">relative fitness (specific)</td>
<td style="text-align: center;"><span class="math inline">\(1-s_1\)</span></td>
<td style="text-align: center;"><span class="math inline">\(1\)</span></td>
<td style="text-align: center;"><span class="math inline">\(1-s_2\)</span></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<p>Here, <span class="math inline">\(s_1\)</span> and <span class="math inline">\(s_2\)</span> are the differences between the relative fitnesses of the two homozygotes and the heterozygote. Note that to obtain relative fitnesses we have divided absolute fitness by the heterozygote fitness. We could use the same parameterization as in the model of directional selection, but the reparameterization we have chosen here makes the math prettier.<br />
In this case, when allele <span class="math inline">\(A_1\)</span> is rare, it is often found in a heterozygous state and so it increases in frequency. However, when allele <span class="math inline">\(A_1\)</span> is common, it is often found in the homozygote state, while the allele <span class="math inline">\(A_2\)</span> is often found in the heterozygote state; it is now <span class="math inline">\(A_2\)</span> that increases in frequency at the expense of allele <span class="math inline">\(1\)</span>. Thus, at least in the deterministic model, neither allele can reach fixation and both alleles will be maintained as a balanced polymorphism in the population at an equilibrium frequency.<br />
We can solve for this equilibrium frequency by setting <span class="math inline">\(\Delta p_t = 0\)</span> in eqn. , i.e. <span class="math inline">\(p_tq_t ({\overline{w}}_1-{\overline{w}}_2)=0\)</span>. Doing so, we find that there are three equilibria, all of which are stable. Two of them are not very interesting (<span class="math inline">\(p=0\)</span> or <span class="math inline">\(q=0\)</span>), but the third one is the polymorphic equilibrium, where <span class="math inline">\({\overline{w}}_1-{\overline{w}}_2\)</span> holds. Using the parametrization above, we see that the marginal fitnesses of the two alleles are formally equivalent. Insertion of the selection coefficients <span class="math inline">\(s_1\)</span> and <span class="math inline">\(s_2\)</span> yields <span class="math display">\[p_e = \frac{s_2}{s_1+s_2}\]</span> for the equilibrium frequency of interest. This is also the frequency of <span class="math inline">\(A_1\)</span> at which the mean fitness of the population is maximised.<br />
</p>
<h4 id="underdominance.">Underdominance.</h4>
<p>Another case that is of potential interest is the case of fitness underdominance, where the heterozygote is less fit than either of the homozygotes. This can be parametrized as follows:<br />
</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;">genotype</td>
<td style="text-align: center;"><span class="math inline">\(A_1A_1\)</span></td>
<td style="text-align: center;"><span class="math inline">\(A_1A_2\)</span></td>
<td style="text-align: center;"><span class="math inline">\(A_2A_2\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">absolute fitness</td>
<td style="text-align: center;"><span class="math inline">\(w_{11}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(&lt;w_{12}&gt;\)</span></td>
<td style="text-align: center;"><span class="math inline">\(w_{22}\)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">relative fitness (generic)</td>
<td style="text-align: center;"><span class="math inline">\(w_{11}=W_{11}/W_{12}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(w_{12} = W_{12}/W_{12}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(w_{22} = W_{22}/W_{12}\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">relative fitness (specific)</td>
<td style="text-align: center;"><span class="math inline">\(1+s_1\)</span></td>
<td style="text-align: center;"><span class="math inline">\(1\)</span></td>
<td style="text-align: center;"><span class="math inline">\(1+s_2\)</span></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<p>This case also permits three equilibria, <span class="math inline">\(p=0\)</span>, <span class="math inline">\(p=1\)</span>, and a polymorphic equilibrium <span class="math inline">\(p=p_U\)</span>. However, now only the first two equilibria are stable, while the polymorphic equilibrium is unstable. If <span class="math inline">\(p&lt;p_U\)</span> then <span class="math inline">\(\Delta p_t \)</span> is negative and allele <span class="math inline">\(A_1\)</span> will be lost, while if <span class="math inline">\(p&gt;p_U\)</span>, allele <span class="math inline">\(A_1\)</span> will become fixed.<br />
While such alleles might not spread within populations (if <span class="math inline">\(p_U \gg
0\)</span> and selection is reasonably strong), they are of interest in the study of speciation and hybrid zones. That is because alleles <span class="math inline">\(A_1\)</span> and <span class="math inline">\(A_2\)</span> may have arisen in a stepwise fashion, i.e. not by a single mutation, in separate subpopulations. Now, heterozygote disadvantage will play a potential role in species maintenance, if isolation of the subpopulations is not complete.<br />
</p>
<p>You are studying the polymorphism that affects flight speed in butterflies. The polymorphism does not appear to affect fecundity. Homozygotes for the B allele are slow in flight and so only 40% of them survive to have offspring. Heterozygotes for the polymorphism (Bb) fly quickly and have a 70% probability of surviving to reproduce. The homozygotes for the alternative allele (bb) fly very quickly indeed, but often die of exhaustion, with only 10% of them making it to reproduction.<br />
<span><strong>A)</strong></span> What is the equilibrium frequency of the B allele?<br />
<span><strong>B)</strong></span> Calculate the marginal fitnesses of the B and the b allele at the equilbrium frequency.</p>
<h4 id="diploid-fluctuating-fitness">Diploid fluctuating fitness</h4>
<p>We would like to think about the case where the diploid absolute fitnesses are time-dependent. The three genotypes then have fitnesses <span class="math inline">\(w_{11,t}\)</span>, <span class="math inline">\(w_{12,t}\)</span>, and <span class="math inline">\(w_{22,t}\)</span> in generation <span class="math inline">\(t\)</span>. However, this case is much less tractable than the haploid case, as segregation makes it tricky to keep track of the genotype frequencies. We can make some progress and gain some intuition by thinking about how the frequency of allele <span class="math inline">\(A_1\)</span> changes when it is rare.<br />
When <span class="math inline">\(A_1\)</span> is rare, i.e. <span class="math inline">\(p_t \ll 1\)</span>, its frequency in the next generation can be approximated as <span class="math display">\[p_{t+1} \approx \frac{w_{12}}{{\overline{w}}} p_t.\]</span> To obtain this, we have ignored the <span class="math inline">\(p_{t}^2\)</span> term and assumed that <span class="math inline">\(q_t \approx 1\)</span> in the numerator. Following a similar argument to approximate <span class="math inline">\(q_{t+1}\)</span>, we can write <span class="math display">\[\frac{p_{t+1}}{q_{t+1}} = \frac{w_{12,t}}{w_{22,t}}  \frac{p_{t}}{q_{t}}.\]</span> Then, starting from out from <span class="math inline">\(p_0\)</span> and <span class="math inline">\(q_0\)</span> in generation <span class="math inline">\(0\)</span>, <span class="math inline">\(t+1\)</span> generations later, we have <span class="math display">\[\frac{p_{t+1}}{q_{t+1}} = \left( \prod_{i=0}^{t-1} \frac{w_{12,i}}{w_{22,i}}  \right) \frac{p_{0}}{q_{0}}.\]</span> From this, we can see, following our haploid argument from above, that the frequency of allele <span class="math inline">\(A_1\)</span> will increase when rare only if <span class="math display">\[\frac{\sqrt[t]{\prod_{i=0}^{t-1}w_{12,i}}}{\sqrt[t]{\prod_{i=0}^{t-1}w_{22,i}}}&gt;1 \label{geometric_1wins},\]</span> i.e. if the heterozygote has higher geometric mean fitness than the <span class="math inline">\(A_2A_2\)</span> homozygote.<br />
The question now is, whether allele <span class="math inline">\(A_1\)</span> will approach fixation in the population, or whether there are cases in which we can obtain a balanced polymorphism. To investigate that, we can simply repeat our analysis for <span class="math inline">\(q \ll 1\)</span>, and see that in that case <span class="math display">\[\frac{p_{t+1}}{q_{t+1}} = \left( \prod_{i=0}^{t-1} \frac{w_{11,i}}{w_{12,i}}  \right) \frac{p_{0}}{q_{0}}.\]</span> Now, for allele <span class="math inline">\(A_1\)</span> to carry on increasing in frequency and to approach fixation, the <span class="math inline">\(A_1A_1\)</span> genotype has to be out-competing the heterozygotes. For allele <span class="math inline">\(A_1\)</span> to approach fixation, we need the geometric mean of <span class="math inline">\(w_{11,i}\)</span> to be greater than the geometric mean fitness of heterozygotes (<span class="math inline">\(w_{12,i}\)</span>). At the same time, if heterozygotes have higher geometric mean fitness than the <span class="math inline">\(A_1A_1\)</span> homozygotes, then the <span class="math inline">\(A_2\)</span> allele will increase in frequency when it is rare. Therefore, a balanced polymorphism can result when the heterozygote has higher geometric fitness than either of the homozygotes.<br />
Intriguingly, we can have a balanced polymorphism even if the heterozygote is never the fittest genotype in any generation. To see this, consider the simple example, where there are two environments alternate from generation to generation:<br />
</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;">genotype</td>
<td style="text-align: center;"><span class="math inline">\(A_1A_1\)</span></td>
<td style="text-align: center;"><span class="math inline">\(A_1A_2\)</span></td>
<td style="text-align: center;"><span class="math inline">\(A_2A_2\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">relative fitness in environment A</td>
<td style="text-align: center;"><span class="math inline">\(w_{11,A}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(&gt;w_{12,A}&gt;\)</span></td>
<td style="text-align: center;"><span class="math inline">\(w_{22,A}\)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">relative fitness in environment B</td>
<td style="text-align: center;"><span class="math inline">\(w_{11,B}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(&lt;w_{12,B}&lt;\)</span></td>
<td style="text-align: center;"><span class="math inline">\(w_{22,B}\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">Geometric mean fitness</td>
<td style="text-align: center;"><span class="math inline">\(w_{11,B}\)</span></td>
<td style="text-align: center;"><span class="math inline">\(&lt;w_{12,B}&gt;\)</span></td>
<td style="text-align: center;"><span class="math inline">\(w_{22}\)</span></td>
</tr>
</tbody>
</table>
<p><br />
In this case, the polymorphism will remain balanced in the population, despite the fact that the heterozygote is never the fitest genotype.</p>
<p>Imagine a randomly-mating population of hermaphrodites. In this population a derived allele (D) segregates that distorts transmission in its favour over the ancestral allele (d) in the production of all the gametes of heterozygotes. The drive leads to <span class="math inline">\(r\%\)</span> of the gametes of heterozygotes (D/d) to carry the D allele (<span class="math inline">\(r&gt;50\%\)</span>). The D allele causes viability problems in the homozygote state such that the relative fitnesses are <span class="math inline">\(w_{dd}=1\)</span>, <span class="math inline">\(w_{Dd}=1\)</span>, <span class="math inline">\(w_{DD}=1-e\)</span>. The allele is currently at frequency p in the population at birth. Assuming that the population is very large and no mutation occurs:<br />
<span><strong>A)</strong></span> What is the frequency of the D allele in the next generation, before selection has had a chance to act?<br />
<span><strong>B)</strong></span> What conditions do you need for a polymorphic equilibrium to be maintained? At what is the equilibrium frequency of this balanced polymorphism?<br />
<span><strong>C)</strong></span> Imagine the cost of the driver were additive <span class="math inline">\(w_{dd}=1\)</span>, <span class="math inline">\(w_{Dd}=1-e\)</span>, <span class="math inline">\(w_{DD}=1-2e\)</span>. Under what conditions can the driver invade the population? Can a polymorphic equilibrium be maintained?</p>
<h2 id="mutationselection-balance">Mutation–selection balance</h2>
<p>Mutation is constantly introducing new alleles into the population. Therefore, variation can be maintained within a population not only if selection is balancing (e.g. through heterozygote advantage or fluctuating selection over time, as we have seen in the previous section), but also due to a balance between mutation and selection. A case of particular interest is when mutation introduces deleterious alleles and selection acts against these alleles. To study this balance, we return to the model of directional selection, where allele <span class="math inline">\(A_1\)</span> is advantageous, i.e.</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;">genotype</td>
<td style="text-align: center;"><span class="math inline">\(A_1A_1\)</span></td>
<td style="text-align: center;"><span class="math inline">\(A_1A_2\)</span></td>
<td style="text-align: center;"><span class="math inline">\(A_2A_2\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">absolute fitness</td>
<td style="text-align: center;"><span class="math inline">\(W_{11}\)</span></td>
<td style="text-align: center;"><span class="math inline">\( \geq W_{12} \geq\)</span></td>
<td style="text-align: center;"><span class="math inline">\(W_{22}\)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">relative fitness</td>
<td style="text-align: center;"><span class="math inline">\(w_{11}=1\)</span></td>
<td style="text-align: center;"><span class="math inline">\(w_{12}=1-sh\)</span></td>
<td style="text-align: center;"><span class="math inline">\(w_{22}=1-s\)</span>.</td>
</tr>
</tbody>
</table>
<p><br />
</p>
<p>For a start, we consider the case where allele <span class="math inline">\(A_2\)</span> is not completely recessive (<span class="math inline">\(h&gt;0\)</span>), so that the heterozygotes suffer at least some disadvantage. We denote by <span class="math inline">\(\mu = \mu_{1\rightarrow2}\)</span> the mutation rate per generation from <span class="math inline">\(A_1\)</span> to the deleterious allele <span class="math inline">\(A_2\)</span>, and assume that there is no reverse mutation (<span class="math inline">\(\mu_{2\rightarrow1} = 0\)</span>). Let us assume that selection against <span class="math inline">\(A_2\)</span> is relatively strong compared to the mutation rate, so that it is justified to assume that <span class="math inline">\(A_2\)</span> is always rare, i.e. <span class="math inline">\(q_t = 1-p_t \ll 1\)</span>. Compared to previous sections, for mathematical clarity, we also switch from following the frequency <span class="math inline">\(p_t\)</span> of <span class="math inline">\(A_1\)</span> to following the frequency <span class="math inline">\(q_t\)</span> of <span class="math inline">\(A_2\)</span>. Of course, this is without loss of generality. The change in frequency of <span class="math inline">\(A_2\)</span> due to selection can be written as <span class="math display">\[\Delta_S q_t = \frac{{\overline{w}}_2 - {\overline{w}}_1}{{\overline{w}}} p_t q_t  \approx  -hs q_t.
	\label{eq:dirSelApprox}\]</span> This approximation can be found by assuming that <span class="math inline">\(q^2 \approx 0\)</span>, <span class="math inline">\(p \approx 1\)</span>, and that <span class="math inline">\({\overline{w}}\approx w_1\)</span>.</p>
<p>All of these assumptions make sense if <span class="math inline">\(q \ll 1\)</span>. From eqn.  we see that selection acts to reduce the frequency of <span class="math inline">\(A_2\)</span> (as both <span class="math inline">\(h\)</span> and <span class="math inline">\(s\)</span> are positive), and it does so geometrically across the generations. That is, if the initial frequency of <span class="math inline">\(A_2\)</span> is <span class="math inline">\(q_0\)</span>, then its frequency at time <span class="math inline">\(t\)</span> is approximately <span class="math display">\[q_t = q_0 (1 - hs)^t.
	\label{eq:dirSelExplApprox}\]</span></p>
<p>We will now consider the change in frequency induced by mutation. Recalling that <span class="math inline">\(\mu\)</span> is the mutation rate from <span class="math inline">\(A_1\)</span> to <span class="math inline">\(A_2\)</span> per generation, the frequency of <span class="math inline">\(A_2\)</span> after mutation <span class="math display">\[q^{\prime} =  \mu p_t = \mu(1 - q_t).\]</span> Assuming that <span class="math inline">\(\mu \ll 1\)</span> and that <span class="math inline">\(q \ll 1\)</span>, the change in the frequency of allele <span class="math inline">\(A_2\)</span> due to mutation (<span class="math inline">\(\Delta_M q_t\)</span>) can be approximated by <span class="math display">\[\Delta_M q_t = q^{\prime} - q_t =  \mu.
	\label{eq:mutApprox}\]</span> Hence, when <span class="math inline">\(A_2\)</span> is rare and the mutation rate is low, mutation acts to linearly increase the frequency of the deleterious allele <span class="math inline">\(A_2\)</span>.<br />
If selection is to balance deleterious mutation, there combined effect over one generation has to be zero. Therefore, to find the mutation–selection equilibrium, we set <span class="math display">\[\Delta_M q_t + \Delta_S q_t = 0,\]</span> insert eqns. and , and solve for <span class="math inline">\(q\)</span> to obtain <span class="math display">\[q_e = q_t = \frac{\mu}{hs}.
	\label{eqn:mut_sel_bal}\]</span> We see that the frequency of the deleterious allele <span class="math inline">\(A_2\)</span> is balanced at the mutation rate (<span class="math inline">\(\mu\)</span>) divided by the reduction in relative fitness in the heterozygote (<span class="math inline">\(hs\)</span>).</p>
<p><br />
</p>
<p>It is worth pointing out that the fitness of the <span class="math inline">\(A_2A_2\)</span> homozygote has not entered this calculation, as <span class="math inline">\(A_2\)</span> is so rare that it is hardly ever found in the homozygous state. Therefore, if <span class="math inline">\(A_2\)</span> has any deleterious effect in a heterozygous state (i.e. if <span class="math inline">\(h&gt;0\)</span>) it is this effect that determines the frequency at which <span class="math inline">\(A_2\)</span> is maintained in the population. Also, note that by writing the total change in allele frequency as <span class="math inline">\(\Delta_M q_t + \Delta_S q_t
\)</span> we have implicitly assumed that we can ignore terms of order <span class="math inline">\(\mu
\times s\)</span>. That is, we have assumed that there is no interaction between mutation and selection. We can do so as we assumed that both <span class="math inline">\(\mu\)</span> and <span class="math inline">\(s\)</span> are small.<br />
What effect do such deleterious mutations at mutation–selection balance have on the population? It is common to express this effect in terms of a reduction of the mean relative fitness of the population. For a single site at which a deleterious mutation is segregating at <span class="math inline">\(q_e = \mu/(hs)\)</span>, the mean relative fitness is reduced to <span class="math display">\[{\overline{w}}= 1- 2p_e q_e hs - q_e^2s \approx 1-2\mu.\]</span> Somewhat remarkably, the drop in mean fitness due to a site segregating at mutation–selection balance is independent of the selection coefficient against the heterozygote; it depends only the mutation rate. Note that this applies only if the mutation is not totally recessive, i.e. if <span class="math inline">\(h &gt; 0\)</span>.</p>
<p>A reduction of <span class="math inline">\(1 - 2\mu\)</span> is very small, given that the mutation rate of a gene is likely <span class="math inline">\(&lt;10^{-5}\)</span>. However, if there are many loci segregating at mutation–selection balance, this can accumulate to a substantial so-called genetic load, and a major cause of variation in fitness-related traits among individuals.</p>
<p>As an aside, if an allele was truly recessive (although few likely are), we have <span class="math inline">\(h=0\)</span>, and so eqn.  is not valid. However, we can make an argument similar to the one above to show that, for truly recessive alleles, <span class="math display">\[q_e = \sqrt{\frac{\mu}{s}}.\]</span></p>
<p>You are studying an outbred population of mice living in a farmer’s field. Mutations occur at a gene called nurseryrhyme that cause a totally recessive form of blindness. These blind mice do not survive to reproduce as the farmer’s wife cuts off their tail (and other bits) with a carving knife. Surveying the field for baby mice you find that 3 in ten thousand mice are blind.<br />
<span><strong>A</strong></span> Assuming that the population mates at random, what is the mutation rate of blindness causing alleles?<br />
<span><strong>B</strong></span> Following more careful study you now find that there is actually a <span class="math inline">\(20
\%\)</span> reduction in the viability of heterozygotes for these mutations. What would you now estimate as the mutation rate for this gene? <span><strong>C)</strong></span> Explain how and why your answers differ?</p>
<h3 id="inbreeding-depression">Inbreeding depression</h3>
<p>All else being equal, eqn.  suggests that mutations that have a smaller effect in the heterozygote can segregate at higher frequency under mutation–selection balance. As a consequence, alleles that have strongly deleterious effects in the homozygous state can segregate at low frequencies in the population, as long as they do not have a strong effect in heterozygotes. Thus, outbred populations may have many alleles with recessive deleterious effects segregating within them.</p>
<p>On consequence of this is that inbred individuals from usually outbred populations may have dramatically lower fitnesses than outbred individuals. This is a consequence of being homozygous at many loci for alleles with recessive deleterious effects. Indeed, this seems to be a common observation, dating back to systematic surveys by Darwin. In typically outbred populations, the mean fitness of individuals decreases with the inbreeding coefficient, i.e. this so-called inbreeding depression is a common observation.</p>
<h4 id="purging-the-inbreeding-load.">Purging the inbreeding load.</h4>
<p>Populations that regularly inbreed over sustained periods of time are expected to partially purge this load of deleterious alleles. This is because such populations have exposed many of these alleles in a homozygous state, and so selection can more readily remove these alleles from the population.</p>
<h2 id="migrationselection-balance">Migration–selection balance</h2>
<p>Another reason for the persistence of deleterious alleles in a population is that there is a constant influx of maladaptive alleles from other populations where these alleles are locally adapted. This seems unlikely to be as broad an explanation for the persistence of deleterious alleles genome-wide as mutation-selection balance. However, a brief discussion of such alleles is worthwhile as it helps to inform our ideas about local adaptation.<br />
As a first pass at this lets consider a haploid two allele model with two different populations, where the relative fitnesses of our alleles are as follows</p>
<table>
<thead>
<tr class="header">
<th style="text-align: center;">allele</th>
<th style="text-align: center;"><span class="math inline">\(1\)</span></th>
<th style="text-align: center;"><span class="math inline">\(2\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">population 1</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">1-s</td>
</tr>
<tr class="even">
<td style="text-align: center;">population 2</td>
<td style="text-align: center;">1-s</td>
<td style="text-align: center;">1</td>
</tr>
</tbody>
</table>
<p>As a simple model of migration lets suppose within a population a fraction of <span class="math inline">\(m\)</span> individuals are migrants from the other population, and <span class="math inline">\(1-m\)</span> individuals are from the same deme.<br />
To quickly sketch a solution to this well set up a situation analogous to our mutation-selection balance model. to do this lets assume that selection is strong compared to migration (<span class="math inline">\(s \gg m\)</span>) then allele <span class="math inline">\(1\)</span> will be almost fixed in population <span class="math inline">\(1\)</span> and allele <span class="math inline">\(2\)</span> will be almost fixed in population <span class="math inline">\(2\)</span>. If that is the case, migration changes the frequency of allele <span class="math inline">\(2\)</span> in population <span class="math inline">\(1\)</span> (<span class="math inline">\(q_1\)</span>) by <span class="math display">\[\Delta_{Mig.} q_1 \approx m\]</span> while as noted above <span class="math inline">\(\Delta_{S} q_1= -sq_1\)</span>, so that migration and selection are at an equilibrium when <span class="math inline">\(\Delta_{S} q_1+
\Delta_{Mig.}q_1\)</span>, i.e. an equilibrium frequency of allele <span class="math inline">\(2\)</span> in population <span class="math inline">\(1\)</span> of <span class="math display">\[q_{e,1} = \frac{m}{s}\]</span> so that migration is playing to role of mutation and so migration-selection balance (at least under strong selection) is analogous to mutation selection balance.<br />
We can use this same model by analogy for the case of migration-selection balance in a diploid model, in that case we replace our haploid <span class="math inline">\(s\)</span> by the cost to heterozygotes <span class="math inline">\(hs\)</span>.</p>
<p>You are investigating a small river population of sticklebacks, which receives infrequent migrants from a very large marine population. At a set of (putatively) neutral biallelic markers the freshwater population has frequencies: 0.2, 0.7, 0.8 at the same markers the marine population has frequencies: 0.4, 0.5 and 0.7. From studying patterns of heterozygosity at a large collection of markers, you have estimated the long term effective size of your freshwater population is 2000 individuals.<br />
<span><strong>A)</strong></span> What is <span class="math inline">\(F_{ST}\)</span> across these neutral markers in the freshwater population, with respect to the large marine population (i.e. treat the marine population as the total)?<br />
<span><strong>B)</strong></span> You are also studying an unlinked locus involved in the regulation of salt uptake. In the marine population the ancestral allele is at close to fixation, but in your river population the derived allele is at 0.99 frequency. Estimate the selective disadvantage of the ancestral allele in your river population. [Hint how can you use neutral differentiation to estimate the migration rate?]</p>
<h3 id="some-theory-of-the-spatial-distribution-of-allele-frequencies-under-deterministic-models-of-selection">Some theory of the spatial distribution of allele frequencies under deterministic models of selection</h3>
<p>Imagine a continuous haploid population spread out along a line. Individuals disperse a random distance <span class="math inline">\(\Delta x\)</span> from its birthplace to the location where it reproduces, where <span class="math inline">\(\Delta x\)</span> is drawn from the probability density <span class="math inline">\(g(~)\)</span>. To make life simple we will assume that <span class="math inline">\(g(\Delta x)\)</span> is normally distributed with mean zero and standard deviation <span class="math inline">\(\sigma\)</span>, i.e. migration is unbiased an individuals migrate an average distance of <span class="math inline">\(\sigma\)</span>.<br />
Our frequency of allele <span class="math inline">\(2\)</span> at time <span class="math inline">\(t\)</span> in the population at spatial location <span class="math inline">\(x\)</span> is <span class="math inline">\(q(x,t)\)</span>. Assuming that only dispersal occurs, how does our allele frequency change in the next generation? Our allele frequency in the next generation at location <span class="math inline">\(x\)</span> reflects the migration from different locations in the proceeding generation. Our population at location <span class="math inline">\(x\)</span> receives a contribution <span class="math inline">\(g(\Delta x)q(x+\Delta x,t)\)</span> of allele <span class="math inline">\(2\)</span> from the population at location <span class="math inline">\(x+\Delta x\)</span>, such that the frequency of our allele at <span class="math inline">\(x\)</span> in the next generation is <span class="math display">\[q(x,t+1) = \int_{-\infty}^{\infty} g(\Delta
x)q(x+\Delta x,t) d \Delta x.\]</span></p>
<p>To obtain <span class="math inline">\(q(x+\Delta x,t)\)</span>, lets take a Taylor series expansion of <span class="math inline">\(q(x, t)\)</span></p>
<p><span class="math display">\[q(x+\Delta x,t) = q(x,t) + \Delta x \frac{dq(x,t)}{dx}+ \tfrac{1}{2}(\Delta x)^2 \frac{d^2q(x,t)}{dx^2}+\cdots\]</span></p>
<p>then</p>
<p><span class="math display">\[q(x,t+1) = q(x,t) +\left( \int_{-\infty}^{\infty} \Delta x g(\Delta x) 
  d \Delta x \right) \frac{dq(x,t)}{dx} + \tfrac{1}{2}\left( \int_{-\infty}^{\infty}(\Delta x)^2 g(\Delta x) 
  d \Delta x \right)  \frac{d^2q(x,t)}{dx^2}+\cdots\]</span></p>
<p><span class="math inline">\(g(~)\)</span> has a mean of zero so <span class="math inline">\( \int_{-\infty}^{\infty} \Delta x g(\Delta x) d
\Delta x =0\)</span> and has variance <span class="math inline">\(\sigma^2\)</span> so <span class="math inline">\(\int_{-\infty}^{\infty}(\Delta
x)^2 g(\Delta x) d \Delta x = \sigma^2\)</span> and all higher terms are zero (as all high moments of the normal are zero). Looking at the change in frequency <span class="math inline">\(\Delta q(x,t) = q(x,t+1)-q(x,t)\)</span> then</p>
<p><span class="math display">\[\Delta q(x,t) = \frac{\sigma^2}{2} \frac{d^2q(x,t)}{dx^2}\]</span></p>
<p>This is a diffusion equation, so that migration is acting to smooth out allele frequency differences with a diffusion constant of <span class="math inline">\(\tfrac{\sigma^2}{2}\)</span>. This is exactly analogous to the equation describing how a gas diffuses out to equal density, as both particles in a gas and our individuals of type <span class="math inline">\(2\)</span> are performing Brownian motion (blurring our eyes and seeing time as continuous).<br />
We will now introduce fitness differences into our model and set the relative fitnesses of allele <span class="math inline">\(1\)</span> and <span class="math inline">\(2\)</span> at location <span class="math inline">\(x \)</span> to be <span class="math inline">\(1\)</span> and <span class="math inline">\(1+s\gamma(x)\)</span>. To make progress in this model we’ll have to assume that selection isn’t too strong i.e. <span class="math inline">\( s \gamma(x) \ll 1\)</span> for all <span class="math inline">\(x\)</span>. The change in frequency of allele <span class="math inline">\(2\)</span> obtained within a generation due to selection is</p>
<p><span class="math display">\[q^{\prime}(x,t) - q(x,t) \approx s\gamma(x) q(x,t) \big( 1 - q(x,t) \big)\]</span></p>
<p>i.e. logistic growth of our favoured allele at location <span class="math inline">\(x\)</span>. Putting our selection and migration terms together we find <span class="math display">\[q(x,t+1) -
q(x,t) = s\gamma(x) q(x,t) \big( 1 - q(x,t) \big)+\frac{\sigma^2}{2}
\frac{d^2q(x,t)}{dx^2} \label{eqn:fisherKPP}\]</span> in deriving this we have essentially assumed that migration acted upon our original frequencies before selection and in doing so have ignored terms of the order of <span class="math inline">\(\sigma s\)</span>.</p>
<figure>
<img src="figures/equilib_cline.png" alt="An equilibrium cline in allele frequency. Our individuals dispersal an average distance of \sigma=1km per generation, and our allele 2 has a relative fitness of 1+s and 1-s on either side of the environmental change at x=0." /><figcaption>An equilibrium cline in allele frequency. Our individuals dispersal an average distance of <span class="math inline">\(\sigma=1\)</span>km per generation, and our allele <span class="math inline">\(2\)</span> has a relative fitness of <span class="math inline">\(1+s\)</span> and <span class="math inline">\(1-s\)</span> on either side of the environmental change at <span class="math inline">\(x=0\)</span>.<span data-label="fig:cline"></span></figcaption>
</figure>
<h4 id="the-cline-in-allele-frequency-associated-with-a-sharp-environmental-transition.">The cline in allele frequency associated with a sharp environmental transition.</h4>
<p>To make progress lets consider a simple model of location adaptation where the environment abruptly changes. Specifically we assume that <span class="math inline">\(\gamma(x)= 1\)</span> for <span class="math inline">\(x&lt;0\)</span> and <span class="math inline">\(\gamma(x)= -1\)</span> for <span class="math inline">\(x \geq 0\)</span>, i.e. our allele <span class="math inline">\(2\)</span> has a selective advantage at locations to the left of zero, while this allele is at a disadvantage to the right of zero. In this case we can get an equilibrium distribution of our two alleles were to the left of <span class="math inline">\(zero\)</span> our allele <span class="math inline">\(2\)</span> is at higher frequency, while to the right of zero allele <span class="math inline">\(1\)</span> predominates. As we cross from the left to the right side of our range the frequency of our allele <span class="math inline">\(2\)</span> decreases in a smooth cline.<br />
Our equilibrium spatial distribution of allele frequencies can be found by setting the LHS of eqn. to zero to arrive at <span class="math display">\[s\gamma(x) q(x) \left( 1 - q(x) \right) = - \frac{\sigma^2}{2} \frac{d^2q(x)}{dx^2}\]</span> We then could solve this differential equation with appropriate boundary conditions (<span class="math inline">\(q(-\infty)=1\)</span> and <span class="math inline">\(q(\infty) = 0\)</span>) to arrive at the appropriate functional form to our cline. While we won’t go into the solution of this equation here, we can note that by dividing our distance <span class="math inline">\(x\)</span> by <span class="math inline">\(\ell=\sigma/\sqrt{s}\)</span> we can remove the effect of our parameters from the above equation. This compound parameter <span class="math inline">\(\ell\)</span> is the characteristic length of our cline, and it is this parameter which determines over what geographic scale we change from allele <span class="math inline">\(2\)</span> predominating to allele <span class="math inline">\(1\)</span> predominating as we move across our environmental shift.<br />
The width of our cline, i.e. over what distance do we make this shift from allele <span class="math inline">\(2\)</span> predominating to allele <span class="math inline">\(1\)</span>, can be defined in a number of different ways. One simple way to define the cline width, which is easy to define but perhaps hard to measure accurately, is the slope (i.e. the tangent) of <span class="math inline">\(q(x)\)</span> at <span class="math inline">\(x=0\)</span>. Under this definition the cline width is approximately <span class="math inline">\(0.6
\sigma/\sqrt{s}\)</span>.<br />
</p>
<h4 id="the-rate-of-spatial-spread-of-a-beneficial-allele.">The rate of spatial spread of a beneficial allele.</h4>
<p>Consider a beneficial mutation that has arisen in a specific spatial location and has begun to spread geographically.</p>
<h1 id="stochasticity-and-genetic-drift-in-allele-frequencies">Stochasticity and Genetic Drift in allele frequencies</h1>
<h2 id="stochastic-loss-of-strongly-selected-alleles">Stochastic loss of strongly selected alleles</h2>
<p>Even strongly selected alleles can be lost from the population when they are sufficiently rare. This is because the number of offspring left by individuals to the next generation is fundamentally stochastic. A selection coefficient of s=<span class="math inline">\(1\%\)</span> is a strong selection coefficient, which can drive an allele through the population in a few hundred generations once the allele is established. However, if individuals have on average a small number of offspring per generation the first individual to carry our allele who has on average <span class="math inline">\(1\%\)</span> more children could easily have zero offspring, leading to the loss of our allele before it ever get a chance to spread.<br />
To take a first stab at this problem lets think of a very large haploid population, and in order for this population to stay constant in size we’ll assume that individuals without the selected mutation have on average one offspring per generation. While individuals with our selected allele have on average <span class="math inline">\(1+s\)</span> offspring per generation. We’ll assume that the distribution of offspring number of an individual is Poisson distributed with this mean, i.e. the probability that an individual with the selected allele has <span class="math inline">\(i\)</span> children is <span class="math display">\[P_i= \frac{(1+s)^i e^{-(1+s)}}{i!}\]</span></p>
<p>Consider starting from a single individual with the selected allele, and ask about the probability of eventual loss of our selected allele starting from this single copy (<span class="math inline">\(p_L\)</span>). To derive this we’ll make use of a simple argument (derived from branching processes). Our selected allele will be eventually lost from the population if every individual with the allele fails to leave descendants.</p>
<ol>
<li><p>In our first generation with probability <span class="math inline">\(P_0\)</span> our individual leaves no copies of itself to the next generation, in which case our allele is lost (Figure [fig:Proof_of_pL_2s]A).</p></li>
<li><p>Alternatively it could leave one copy of itself to the next generation (with probability <span class="math inline">\(P_1\)</span>), in which case with probability <span class="math inline">\(p_L\)</span> this copy eventually goes extinct (Figure [fig:Proof_of_pL_2s]B).</p></li>
<li><p>It could leave two copies of itself to the next generation (with probability <span class="math inline">\(P_2\)</span>), in which case with probability <span class="math inline">\(p_L^2\)</span> both of these copies eventually goes extinct (Figure [fig:Proof_of_pL_2s]C).</p></li>
<li><p>More generally it could leave could leave <span class="math inline">\(k\)</span> copies (<span class="math inline">\(k&gt;0\)</span>) of itself to the next generation (with probability <span class="math inline">\(P_k\)</span>), in which case with probability <span class="math inline">\(p_L^k\)</span> all of these copies eventually go extinct (e.g. Figure [fig:Proof_of_pL_2s]D).</p></li>
</ol>
<p>summing over this probabilities we see that <span class="math display">\[\begin{aligned}
p_L &amp;= \sum_{k=0}^{\infty} P_k p_L^{k}  \nonumber \\
&amp;=  \sum_{k=0}^{\infty} \frac{(1+s)^ke^{-(1+s)}}{k!} p_L^{k} \nonumber
\\
&amp;= e^{-(1+s)} \left( \sum_{k=0}^{\infty} \frac{\left(p_L(1+s) \right)^k}{k!}  \right)\end{aligned}\]</span> well the term in the brackets is itself an exponential expansion, so we can rewrite this as <span class="math display">\[p_L = e^{(1+s)(p_L-1)} \label{prob_loss}\]</span> solving this would give us our probability of loss for any selection coefficient. Lets rewrite this in terms of the the probability of escaping loss <span class="math inline">\(p_F = 1-p_L\)</span>. We can rewrite eqn as <span class="math display">\[1-p_F = e^{-p_F(1+s)}\]</span> to gain an approximation to this lets consider a small selection coefficient <span class="math inline">\(s \ll 1\)</span> such that <span class="math inline">\(p_F \ll 1\)</span> and then expanded out the exponential on the right hand side (ignoring terms of higher order than <span class="math inline">\(s^2\)</span> and <span class="math inline">\(p_F^2\)</span>) then <span class="math display">\[1-p_F \approx 1-p_F(1+s)+p_F^2(1+s)^2/2\]</span> solving this we find that <span class="math display">\[p_F = 2s.\]</span> Thus even an allele with a <span class="math inline">\(1\%\)</span> selection coefficient has a <span class="math inline">\(98\%\)</span> probability of being lost when it is first introduced into the population by mutation. This style of argument is modeling the fate of our allele using branching processes.<br />
</p>
<figure>
<img src="figures/Proof_of_pL_2s" alt="" /><figcaption><span data-label="fig:Proof_of_pL_2s"></span></figcaption>
</figure>
<p>We can also adapt this result to a diploid setting. Assuming that heterozygotes for the <span class="math inline">\(1\)</span> allele have <span class="math inline">\(1+(1-h)s\)</span> children, the probability of allele <span class="math inline">\(1\)</span> is not lost, starting from a single copy in the population, is <span class="math display">\[p_F = 2 (1-h)s \label{eqn:diploid_escape}\]</span> for <span class="math inline">\(h&gt;0\)</span>. Note this is a slightly different parameterization from our diploid model above.<br />
</p>
<p>Melanic squirrels suffer a higher rate of predation (due to hawks) than normally pigmented squirrels. Melanism is due to a dominant, autosomal mutation. The frequency of melanic squirrels at birth is <span class="math inline">\(4 \times 10^{-5}\)</span>.<br />
<span><strong>A)</strong></span> If the mutation rate to new melanic alleles is <span class="math inline">\(10^{-6}\)</span>, assuming the melanic allele is at mutation-selection equilibrium, what is the reduction in fitness of the heterozygote?<br />
<span><strong>B)</strong></span> Suddenly levels of pollution increase dramatically in our population, and predation by hawks now offers an equal (and opposite) advantage to the dark individuals as it once offered to the normally pigmented individuals. If the population size of our squirrels is a million individuals, what is the probability that the population adapts from the standing pool of melanic alleles?</p>
<h2 id="the-interaction-between-genetic-drift-and-weak-selection.">The interaction between genetic drift and weak selection.</h2>
<p>For strongly selected alleles, once the allele has escaped initial loss at low frequencies, their path will be determined deterministically by their selection coefficients. However, if selection is weak the stochasticity of reproduction can play a role in the trajectory an allele takes even when it is common in the population.<br />
To see this lets think of our simple Wright-Fisher model (see R exercise). Each generation we allow a deterministic change in our allele frequency, and then binomially sample two alleles for each of our offspring to construct our next generation.<br />
So the expected change in our allele frequency within a generation is given just by our deterministic formula. To make things easy on our self lets assume an additive model, i.e. <span class="math inline">\(h=1/2\)</span>, and that <span class="math inline">\(s \ll 1\)</span> so that <span class="math inline">\({\overline{w}}\approx 1\)</span>. This gives us <span class="math display">\[{\mathbb{E}}(\Delta p ) = \frac{s}{2} p(1-p) \label{eqn:WF_mean}\]</span> our variance in our allele frequency change is given by <span class="math display">\[Var(p^{\prime} - p) = Var(p^{\prime}) = \frac{p^{\prime}(1-p^{\prime})}{2N}\]</span> this variance in our allele frequency follows from the fact that we are binomially sampling <span class="math inline">\(2N\)</span> new alleles in the next generation from a frequency <span class="math inline">\(p^{\prime}\)</span>. Denoting our count of allele <span class="math inline">\(1\)</span> by <span class="math inline">\(i\)</span> our <span class="math display">\[Var(p^{\prime} - p) = Var(\frac{i}{2N} - p) =  Var(\frac{i}{2N} ) =\frac{Var(i)}{(2N)^2}\]</span> and from binomial sampling <span class="math inline">\(Var(i) = 2N p^{\prime}(1-p^{\prime})\)</span> and so we arrive at our answer. Assuming that <span class="math inline">\(s \ll 1\)</span>, <span class="math inline">\(p^{\prime}
\approx p\)</span>, then in practice we can use <span class="math display">\[Var(\Delta p)  =Var(p^{\prime} - p) \approx \frac{p(1-p)}{2N}. \label{eqn:WF_var}\]</span> To get our first look at the relative effects of selection vs drift we can simply look at when our change in allele frequency caused selection within a generate is reasonably faithfully passed across the generations. In particular if our expected change in frequency is much great than the variance around this change, genetic drift will play little role in the fate of our selected allele (once the allele is not too rare within the population). When does selected dominant genetic drift? This will happen if <span class="math inline">\({\mathbb{E}}(\Delta p) \gg Var(\Delta p)\)</span> when <span class="math inline">\(Ns \gg 1\)</span>. Conversely any hope of our selected allele following its deterministic path will be quickly undone if our change in allele frequencies due to selection is much less than the variance induced by drift. So if <span class="math inline">\(Ns \ll 1\)</span> then drift will dominate the fate of our allele.<br />
</p>
<figure>
<embed src="figures/prob_fix_diffusion.pdf" /><figcaption>The probability of the fixation of a new mutation with selection coefficient <span class="math inline">\(s\)</span> (<span class="math inline">\(h=1/2\)</span>) in a diploid population of effective size <span class="math inline">\(N_e\)</span>. The dashed line gives the infinite population solutub<span data-label="fig:prob_fix_diffusion"></span></figcaption>
</figure>
<p>To make further progress on understanding the fate of alleles with selection coefficients of the order <span class="math inline">\(1/N\)</span> requires more careful modeling. However, we can obtain the probability that under our diploid model, with an additive selection coefficient <span class="math inline">\(s\)</span>, the probability of allele <span class="math inline">\(1\)</span> fixing within the population starting from a frequency <span class="math inline">\(p\)</span> is given by <span class="math display">\[\pi(p) = \frac{1-e^{-2Ns p }}{1-e^{-2Ns}} \label{eqn:prob_fixed}\]</span> the proof of this is sketched out below (see Section [Section:fixation_weakly_sel]). A new allele will arrive in the population at frequency <span class="math inline">\(p=1/(2N)\)</span>, then its probability of reaching fixation is <span class="math display">\[\pi \left(\frac{1}{2N} \right) = \frac{1-e^{-s }}{1-e^{-2Ns}} \label{eqn:new_mut_prob_fixed}\]</span> if <span class="math inline">\(s \ll1\)</span> but <span class="math inline">\(Ns \gg 1\)</span> then <span class="math inline">\(\pi(\frac{1}{2N}) \approx s\)</span>, which nicely gives us back our result that we obtained above (eqn. ). Our probability of fixation (eqn. ) is plotted as a function of <span class="math inline">\(s\)</span> and <span class="math inline">\(N\)</span> in Figure [fig:prob_fix_diffusion].<br />
In the case where <span class="math inline">\(Ns\)</span> close to <span class="math inline">\(1\)</span> then <span class="math display">\[\pi(\frac{1}{2N}) \approx \frac{s}{1-e^{-2Ns}} \label{eqn:escape_from_intro}\]</span> this is greater than our result <span class="math inline">\(p_F=s\)</span> from the branching process argument (using our additive model of <span class="math inline">\(h=1/2\)</span>), increasingly so for smaller <span class="math inline">\(N\)</span>. Why is this? The reason why is that <span class="math inline">\(p_F\)</span> is really the probability of “never being lost” in an infinitely large population. So to persist indefinitely the allele has to escape loss permanently, by never being absorbed by the zero state. When the population size is finite, to fix we only need to reach a size 2N individuals. Weakly beneficial mutations (Ns 1) are slightly more likely to fix than the s probability, as they only have to reach 2N to never be lost.</p>
<h3 id="the-fixation-of-slightly-deleterious-alleles.">The fixation of slightly deleterious alleles.</h3>
<p>From Figure [fig:prob_fix_diffusion] we can see that weakly deleterious alleles can fix, especially in small populations. To understand how likely it is that deleterious alleles accidently reach fixation by genetic drift, lets assume a diploid model with additive selection (with a selection coefficient of <span class="math inline">\(-s\)</span> against our allele <span class="math inline">\(2\)</span>). If <span class="math inline">\(N s \gg 1\)</span> then our deleterious allele (allele <span class="math inline">\(2\)</span>) can not possibly reach fixation. However, if <span class="math inline">\(Ns\)</span> is not large then <span class="math display">\[\pi(\frac{1}{2N}) \approx \frac{s}{e^{2Ns}-1} \label{eqn:fix_deleterious}\]</span> for our deleterious allele. So deleterious alleles can fix within populations (albeit at a low rate) if <span class="math inline">\(Ns\)</span> is not too large. As above this is because while deleterious mutations will never escape loss in infinite population, but they can become fixed in finite population by reach 2N copies. This is captured by the denominator of the fixation probability under the diffusion model, which that this increases the fixation prob. of alleles with |Ns| 1. The absorption of alleles at 2N copies can also be modeled in finite individual models (i.e. not the diffusion limit), but we will not go into that here.</p>
<p>`Haldane’s sieve’ is the name for the idea that the mutations that contribute to adaptation are likely to be dominant or at least co-dominant.<br />
<strong><span>A)</span> Briefly explain this argument with a verbal model relating to the results we’ve developed in the last two chapters.<br />
<span><strong>B)</strong></span> Haldane’s sieve is thought to be less important for adaptation from previously deleterious standing variation, than adaptation from new mutation. Can you explain the intuition behind of this idea?<br />
<span><strong>C)</strong></span> Haldane’s sieve is likely to be less important in inbred, e.g. selfing, populations. Why is this?<br />
<img src="figures/haldanes_sieve.png" alt="image" /></strong></p>
<h3 id="Section:fixation_weakly_sel">A Sketch Proof of the probability of fixation of weakly selected alleles</h3>
<p>We’ll let <span class="math inline">\(P(\Delta p)\)</span> be the probability that our allele frequency shifts by <span class="math inline">\(\Delta p\)</span> in the next generation. Using this we can write our probability <span class="math inline">\(\pi(p)\)</span> in terms of the probability of achieving fixation averaged over the frequency in the next generation <span class="math display">\[\pi(p)  = \int \pi(p+\Delta p) P(\Delta p) d(\Delta p) \label{eqn:prob_fix_diff_step1}\]</span> This is very similar to the technique that we used deriving our probability of escaping loss in a very large population above.<br />
So we need an expression for <span class="math inline">\(\pi(p+\Delta p)\)</span>. To obtain this we’ll do a Taylor series expansion of <span class="math inline">\(\pi(p)\)</span> assuming that <span class="math inline">\(\Delta p \)</span> is small <span class="math display">\[\pi(p+\Delta p) \approx \pi(p) + \Delta p \frac{d\pi(p)}{dp} + (\Delta p)^2
\frac{d^2\pi(p)}{dp^2} (p)\]</span> ignoring higher order terms.<br />
Taking the expectation over <span class="math inline">\(\Delta p \)</span> on both sides, as in eqn. [eqn:prob_fix_diff_step1], we obtain <span class="math display">\[\pi(p) = \pi(p) + {\mathbb{E}}(\Delta p) \frac{d\pi (p)}{dp} + {\mathbb{E}}((\Delta p)^2)
\frac{d^2\pi(p)}{dp^2}\]</span></p>
<p>Well <span class="math inline">\({\mathbb{E}}(\Delta p) = \frac{s}{2}p(1-p)\)</span> and <span class="math inline">\(Var(\Delta p)= {\mathbb{E}}((\Delta
p)^2)-{\mathbb{E}}^2(\Delta p)\)</span>, so if <span class="math inline">\(s \ll 1\)</span> then <span class="math inline">\({\mathbb{E}}^2(\Delta p) \approx
0\)</span>, and <span class="math inline">\({\mathbb{E}}(\Delta p)^2 = \frac{p(1-p)}{2N}\)</span>. This leaves us with <span class="math display">\[0= \frac{s}{2}p(1-p)\frac{d\pi (p) }{dp} + \frac{p(1-p)}{2N}
\frac{d^2\pi (p) }{dp^2}\]</span> and we can specify the boundary conditions to be <span class="math inline">\(\pi(1)=1\)</span> and <span class="math inline">\(\pi(0)=0\)</span>. Solving this differential equation is somewhat involved process but in doing so we find that <span class="math display">\[\pi(p) = \frac{1-e^{-2Ns p }}{1-e^{-2Ns}}\]</span> This proof can be extended to alleles with arbitrary dominance, however, this does not lead to a analytically tractable expression so we do not pursue this here.</p>
<h1 id="the-effect-of-linked-selection-on-patterns-of-neutral-diversity">The effect of linked selection on patterns of neutral diversity</h1>
<p>A newly derived allele with an additive selection coefficient <span class="math inline">\(s\)</span> will take a time <span class="math inline">\(\tau \approx 2\log(2N)/s\)</span> generations to reach to fixation within our population. This short time window offers very little time for recombination between the selected site and linked neutral sites.<br />
First lets imagine examining variation at a locus fully linked to our selected locus, just after our sweep reached fixation. A pair of neutral alleles sampled at this locus must both trace their ancestral lineages back through to the neutral allele on whose background the selected allele initially arose. As that neutral allele, which existed <span class="math inline">\(\tau\)</span> generations ago is the ancestor of the entire population at this locus. Our individuals who carry the beneficial allele are, from the perspective of these two alleles, exactly like a rapidly expanding population. Therefore, our pair of neutral alleles sampled at our locus will be forced to coalesce <span class="math inline">\(\approx \tau\)</span> generations ago. This is a very short-time scale compared to the average neutral coalescent tie of <span class="math inline">\(2N\)</span> generations of a pair of alleles.<br />
If we now allow recombination into our model we can think about a pair of alleles sampled at a neutral locus a recombination distance <span class="math inline">\(r\)</span> away from our selected site. Our pair of alleles will be forced to coalesce <span class="math inline">\(\approx \tau\)</span> generations if neither of them reside on haplotypes that the selected allele recombined onto during the sweep. This is equivalent to saying that neither of our neutral alleles recombine off of the beneficial allele’s background moving backward in time.<br />
The probability that our lineage fail recombines off our beneficial allele’s background and onto the ancestral background in the <span class="math inline">\(j^{th}\)</span> generation back is <span class="math display">\[r (1-X(j))\]</span> so the probability (<span class="math inline">\(p_{NR}\)</span>) that our lineage fails to recombine off in the <span class="math inline">\(\tau\)</span> generations it takes our selected allele to move through the population is <span class="math display">\[p_{NR}=\prod_{j=1}^{\tau} \big(1- r(1-X(j))\big)\]</span> assuming that <span class="math inline">\(r\)</span> is small then <span class="math inline">\( \left(1- r(1-X(j))\right) \approx
e^{-r(1-X(j))}\)</span>, such that <span class="math display">\[p_{NR}=\prod_{j=1}^{\tau} \left(1- r(1-X(j))\right) \approx \exp
\left( -r\sum_{j=1}^{\tau}
1- X(j) \right) =\exp
\left( -r \tau (1-\widehat{X}) \right)\]</span> where <span class="math inline">\(\widehat{X}\)</span> is the average frequency of the derived allele across the trajectory <span class="math inline">\(\widehat{X} = \frac{1}{\tau}  \sum_{j=1}^{\tau}
 X(j)\)</span>. As our allele is additive its trajectory for frequencies <span class="math inline">\(&lt;0.5\)</span> is the mirror image of its trajectory for frequency <span class="math inline">\(&gt;0.5\)</span>, therefore it average frequency <span class="math inline">\(\widehat{X} =0.5\)</span>. So <span class="math display">\[p_{NR} = e^{-r \tau/2 }.\]</span> The probability that both of our lineages fail to recombine off the sweep and hence are forced to coalesce is <span class="math inline">\(p_{NR}^2\)</span>, assuming that they coalesce at a time close to <span class="math inline">\(\tau\)</span> so that they recombine independently of each other for times <span class="math inline">\(&lt; \tau\)</span>.<br />
If one or other of our lineages recombine off the sweep it will take them on average <span class="math inline">\(\approx 2N\)</span> generations to find a common ancestor as we are back our neutral coalescent. Thus the expected time till our pair of lineages find a common ancestor is <span class="math display">\[{\mathbb{E}}(T_2)  = \tau \times p_{NR}^2 +(1-p_{NR}^2) (\tau +2N) \approx
\left(1-p_{NR}^2 \right) 2N\]</span> where this last approximation assumes that <span class="math inline">\(\tau \ll 2N\)</span>. So the expected pairwise diversity for neutral alleles at a recombination distance <span class="math inline">\(r\)</span> away from the selected sweep (<span class="math inline">\(\pi_r\)</span>) is <span class="math display">\[{\mathbb{E}}(\pi_r) = 2\mu {\mathbb{E}}(T_2)  \approx \theta \left(1-e^{-r\tau} \right) \label{eqn:pi_HH}\]</span> so diversity increases as we move away from the selected site, slowly exponentially plauteuing to its neutral expectation <span class="math inline">\(\theta=4N\mu\)</span>.<br />
To get a sense of the physical scale over which diversity is reduced consider a region where recombination occurs at a rate <span class="math inline">\(r_{BP}\)</span> per base pair per generation, and our locus is <span class="math inline">\( \ell \)</span> base pairs away from the selected site <span class="math inline">\(r=r_{BP } \ell \)</span> (where <span class="math inline">\(r_{BP}  \ell  \ll 1\)</span> so we don’t need to worry about more than one recombination event occurring per generation). Typical recombination rates are on the order of <span class="math inline">\(r_{BP} = 10^{-8}\)</span>, in Figure [fig:hitchhiking_reduction] we show the reduction in diversity, given by eqn. , for two different selection coefficients.<br />
For our expected diversity levels to recover to <span class="math inline">\(50\%\)</span> of its neutral expectation <span class="math inline">\({\mathbb{E}}(\pi_r)/\theta=0.5\)</span>, requires a physical distance <span class="math inline">\(\ell^{*}\)</span> such that <span class="math inline">\(\log(0.5) = -r_{BP} \ell ^*\tau\)</span> as using our expression for <span class="math inline">\(\tau\)</span> then <span class="math inline">\( \ell^* = \frac{s}{r_{BP} \log (2N) }\)</span>. The width of our trough of reduced diversity depends on <span class="math inline">\(s/r_{BP}\)</span>, so else being equal we expect stronger sweeps or sweeps in regions of low recombination to have a larger hitchhiking effect. So that a selection coefficient of <span class="math inline">\(s=0.1\%\)</span> would reduce diversity over 10’s of kb, while a sweep of <span class="math inline">\(s=1\%\)</span> would affect <span class="math inline">\(\sim\)</span>100kb.<br />
</p>
<figure>
<img src="figures/hitchhiking_reduction.png" alt="Reduction in diversity compared to its neutral expectation as a function of the distance away from a site where a selected allele has just gone to fixation. The recombination rate is r_{BP}= 1\times
10^{-8}." /><figcaption>Reduction in diversity compared to its neutral expectation as a function of the distance away from a site where a selected allele has just gone to fixation. The recombination rate is <span class="math inline">\(r_{BP}= 1\times
10^{-8}\)</span>.<span data-label="fig:hitchhiking_reduction"></span></figcaption>
</figure>
<h2 id="a-simple-recurrent-model-of-selective-sweeps">A simple recurrent model of selective sweeps</h2>
<p>We sample a pair of neutral alleles at a locus a genetic distance <span class="math inline">\(r\)</span> away from a locus where sweeps are initiated within the population at some very low rate <span class="math inline">\(\nu\)</span> per generation. The waiting time between sweeps at our locus is exponential <span class="math inline">\(\sim Exp(\nu)\)</span>. Each sweep rapidly transits through the population in <span class="math inline">\(\tau\)</span> generations, such that each sweep is finished long before the next sweep (<span class="math inline">\(\tau \ll 1/\nu\)</span>).<br />
As before our chance that our neutral lineage fails to recombine off the sweep is <span class="math inline">\(p_{NR}\)</span>, such that the probability that our pair of lineages are forced to coalesce by a sweep <span class="math inline">\(e^{-r \tau}\)</span>. Our lineages therefore have a very low probability <span class="math display">\[\nu e^{-r \tau}\]</span> of being forced to coalesce by a sweep per generation. In addition of lineages can coalesce at a neutral rate of <span class="math inline">\(1/(2N)\)</span>. Thus the average waiting time till a coalescent event between our neutral pair of lineages due to either a sweep or a neutral coalescent event is <span class="math display">\[{\mathbb{E}}(T_2) = \frac{1}{\nu e^{-r \tau} + 1/(2N)}\]</span></p>
<p>Now imagine that the sweeps don’t occur at a fixed location with respect to our locus of interest, but now occur uniformly at random across our sequence. The sweeps are initiated at a very low rate of <span class="math inline">\(\nu_{BP}\)</span> per basepair per generation. The rate of coalescent due to sweeps at a locus <span class="math inline">\(\ell\)</span> basepairs away from our neutral loci is <span class="math inline">\(\nu_{BP} e^{-r_{BP} \ell \tau}\)</span>. If our neutral locus is in the middle of a chromosome that stretches <span class="math inline">\(L\)</span> basepairs in either direction the total rate of sweeps per generation that force our pair of lineages to coalesce is <span class="math display">\[2\int_0^{L} \nu_{BP} e^{-r_{BP} \ell \tau} d \ell =
\frac{2\nu_{BP}}{r_{BP} \tau} \left(1-e^{-r_{BP} \tau L} \right)\]</span> so that if <span class="math inline">\(L\)</span> is very large (<span class="math inline">\(r_{BP} \tau L \gg 1\)</span>) the rate of coalesce per generation due to sweeps is <span class="math inline">\(\frac{2\nu_{BP}}{r_{BP} \tau}\)</span>. The total rate of coalescence for a pair of lineages per generation is then <span class="math display">\[\frac{2\nu_{BP}}{r_{BP} \tau}+\frac{1}{2N}\]</span> So our average time till a pair of lineages coalesce is <span class="math display">\[{\mathbb{E}}(T_2) = \frac{1}{\frac{2\nu_{BP}}{r_{BP} \tau}+\frac{1}{2N}} = \frac{r_{BP}2N}{\frac{4N\nu_{BP}}{ \tau}+r_{BP}}\]</span> such that our expected pairwise diversity (<span class="math inline">\(\pi=2\mu{\mathbb{E}}(T_2)\)</span>) in a region of recombination rate <span class="math inline">\(r_{BP}\)</span> that experiences sweeps at rate <span class="math inline">\(\nu_{BP}\)</span> is <span class="math display">\[{\mathbb{E}}(\pi) = \theta \frac{r_{BP}}{\frac{4N\nu_{BP}}{ \tau}+r_{BP}} \label{eqn:pi_GW_HH}\]</span></p>
<figure>
<img src="figures/Genomewide_HH.png" alt="The relationship between (sex-averaged) recombination rate and synonymous site pairwise diversity (\pi) in Drosophila melanogaster using the data of Shapiro et al. 07 (kindly provided by Peter Andolfatto, see Sella et al. 09 for details). The curve is the predicted relationship between \pi and recombination rate obtained by fitting equation to this data using non-linear least squares via the nls() function in R." /><figcaption>The relationship between (sex-averaged) recombination rate and synonymous site pairwise diversity (<span class="math inline">\(\pi\)</span>) in <span><em>Drosophila melanogaster</em></span> using the data of Shapiro et al. 07 (kindly provided by Peter Andolfatto, see Sella et al. 09 for details). The curve is the predicted relationship between <span class="math inline">\(\pi\)</span> and recombination rate obtained by fitting equation to this data using non-linear least squares via the <span>nls()</span> function in <span>R</span>.<span data-label="fig:GW_hitchhiking_reduction"></span></figcaption>
</figure>
</body>
</html>
